// igniteui-webcomponents/components/tabs/themes/light/tab.bootstrap.css.js
import { css } from 'lit';
export const styles = css `:host {
  --item-text-color: var(--ig-tabs-item-text-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --item-background: var(--ig-tabs-item-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --item-hover-background: var(--ig-tabs-item-hover-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --item-hover-color: var(--ig-tabs-item-hover-color, hsla(var(--ig-primary-700), var(--ig-primary-a)));
  --item-active-color: var(--ig-tabs-item-active-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --item-icon-color: var(--ig-tabs-item-icon-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --item-active-icon-color: var(--ig-tabs-item-active-icon-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --item-hover-icon-color: var(--ig-tabs-item-hover-icon-color, hsla(var(--ig-primary-700), var(--ig-primary-a)));
  --item-disabled-icon-color: var(--ig-tabs-item-disabled-icon-color, hsla(var(--ig-primary-500), 0.5));
  --item-active-background: var(--ig-tabs-item-active-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --item-disabled-color: var(--ig-tabs-item-disabled-color, hsla(var(--ig-primary-500), 0.5));
  --indicator-color: var(--ig-tabs-indicator-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --button-color: var(--ig-tabs-button-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --button-background: var(--ig-tabs-button-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --button-hover-background: var(--ig-tabs-button-hover-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --button-hover-color: var(--ig-tabs-button-hover-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --button-disabled-color: var(--ig-tabs-button-disabled-color, hsla(var(--ig-primary-500), 0.5));
  --tab-ripple-color: var(--ig-tabs-tab-ripple-color, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --button-ripple-color: var(--ig-tabs-button-ripple-color, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --border-color: var(--ig-tabs-border-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --border-color--hover: var(--ig-tabs-border-color--hover, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --border-radius: var(--ig-tabs-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.6666666667) * 0.375rem), 0.375rem));
  -webkit-transition: none;
  transition: none;
  background: var(--item-background);
  color: var(--item-text-color);
  border-radius: var(--border-radius);
  border-end-end-radius: 0;
  border-end-start-radius: 0;
}

:host(:hover) {
  background: var(--item-hover-background);
  color: var(--item-hover-color);
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--border-color--hover);
          box-shadow: inset 0 0 0 0.0625rem var(--border-color--hover);
}

:host([selected]) {
  color: var(--item-active-color);
  position: relative;
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--border-color);
          box-shadow: inset 0 0 0 0.0625rem var(--border-color);
}
:host([selected]) ::slotted(igc-icon) {
  color: var(--item-active-icon-color);
}
:host([selected])::after {
  content: "";
  position: absolute;
  bottom: 0;
  inset-inline-start: 0;
  width: 100%;
  height: 0.0625rem;
  background: var(--item-background);
  z-index: 1;
}

:host([selected]:hover) {
  color: var(--item-active-color);
  border-top-color: var(--border-color);
  border-inline-color: var(--border-color);
  border-bottom-color: var(--item-background);
}

:host(:focus-within),
:host([selected]:focus-within) {
  border-radius: var(--border-radius);
  color: var(--item-hover-color);
  background: var(--item-background);
  -webkit-box-shadow: inset 0 0 0 0.125rem var(--item-hover-color);
          box-shadow: inset 0 0 0 0.125rem var(--item-hover-color);
  z-index: 1;
}
:host(:focus-within)::after,
:host([selected]:focus-within)::after {
  display: none;
}

:host([selected]:focus-within) {
  color: var(--item-active-color);
}

:host([disabled]) {
  color: var(--item-disabled-color);
}`;
