// igniteui-webcomponents/components/tabs/themes/light/tabs.bootstrap.css.js
import { css } from 'lit';
export const styles = css `:host {
  --item-text-color: var(--ig-tabs-item-text-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --item-background: var(--ig-tabs-item-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --item-hover-background: var(--ig-tabs-item-hover-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --item-hover-color: var(--ig-tabs-item-hover-color, hsla(var(--ig-primary-700), var(--ig-primary-a)));
  --item-active-color: var(--ig-tabs-item-active-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --item-icon-color: var(--ig-tabs-item-icon-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --item-active-icon-color: var(--ig-tabs-item-active-icon-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --item-hover-icon-color: var(--ig-tabs-item-hover-icon-color, hsla(var(--ig-primary-700), var(--ig-primary-a)));
  --item-disabled-icon-color: var(--ig-tabs-item-disabled-icon-color, hsla(var(--ig-primary-500), 0.5));
  --item-active-background: var(--ig-tabs-item-active-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --item-disabled-color: var(--ig-tabs-item-disabled-color, hsla(var(--ig-primary-500), 0.5));
  --indicator-color: var(--ig-tabs-indicator-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --button-color: var(--ig-tabs-button-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --button-background: var(--ig-tabs-button-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --button-hover-background: var(--ig-tabs-button-hover-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --button-hover-color: var(--ig-tabs-button-hover-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --button-disabled-color: var(--ig-tabs-button-disabled-color, hsla(var(--ig-primary-500), 0.5));
  --tab-ripple-color: var(--ig-tabs-tab-ripple-color, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --button-ripple-color: var(--ig-tabs-button-ripple-color, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --border-color: var(--ig-tabs-border-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --border-color--hover: var(--ig-tabs-border-color--hover, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --border-radius: var(--ig-tabs-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.6666666667) * 0.375rem), 0.375rem));
}

[part=headers] {
  background: var(--item-background);
}

[part=headers-wrapper]::after {
  content: "";
  position: absolute;
  bottom: 0;
  inset-inline-start: 0;
  width: 100%;
  height: 0.0625rem;
  background: var(--border-color);
  z-index: 0;
}

[part=start-scroll-button],
[part=end-scroll-button] {
  background: var(--button-background);
}
[part=start-scroll-button]:hover, [part=start-scroll-button]:focus,
[part=end-scroll-button]:hover,
[part=end-scroll-button]:focus {
  background: var(--button-hover-background);
}

igc-icon-button::part(base) {
  color: var(--button-color);
}
igc-icon-button::part(base):hover, igc-icon-button::part(base):focus-within {
  color: var(--button-hover-color);
}

igc-icon-button[disabled]::part(base) {
  color: var(--button-disabled-color);
}

[part=selected-indicator] {
  display: none;
}`;
