// igniteui-webcomponents/components/textarea/themes/light/textarea.indigo.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium: min(
      clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
      clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
  );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

:host([hidden]),
[hidden] {
  display: none !important;
}
:host {
  --search-resting-elevation: var(--ig-textarea-search-resting-elevation, var(--ig-elevation-1));
  --search-hover-elevation: var(--ig-textarea-search-hover-elevation, var(--ig-elevation-1));
  --search-disabled-elevation: var(--ig-textarea-search-disabled-elevation, var(--ig-elevation-1));
  --helper-text-color: var(--ig-textarea-helper-text-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --input-prefix-background: var(--ig-textarea-input-prefix-background, transparent);
  --input-prefix-color: var(--ig-textarea-input-prefix-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --input-prefix-background--filled: var(--ig-textarea-input-prefix-background--filled, inherit);
  --input-prefix-color--filled: var(--ig-textarea-input-prefix-color--filled, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --input-prefix-background--focused: var(--ig-textarea-input-prefix-background--focused, inherit);
  --input-prefix-color--focused: var(--ig-textarea-input-prefix-color--focused, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --input-suffix-background: var(--ig-textarea-input-suffix-background, transparent);
  --input-suffix-color: var(--ig-textarea-input-suffix-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --input-suffix-background--filled: var(--ig-textarea-input-suffix-background--filled, inherit);
  --input-suffix-color--filled: var(--ig-textarea-input-suffix-color--filled, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --input-suffix-background--focused: var(--ig-textarea-input-suffix-background--focused, inherit);
  --input-suffix-color--focused: var(--ig-textarea-input-suffix-color--focused, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --idle-text-color: var(--ig-textarea-idle-text-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --filled-text-color: var(--ig-textarea-filled-text-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --focused-text-color: var(--ig-textarea-focused-text-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --disabled-text-color: var(--ig-textarea-disabled-text-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --idle-secondary-color: var(--ig-textarea-idle-secondary-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --filled-secondary-color: var(--ig-textarea-filled-secondary-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --focused-secondary-color: var(--ig-textarea-focused-secondary-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --idle-bottom-line-color: var(--ig-textarea-idle-bottom-line-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --hover-bottom-line-color: var(--ig-textarea-hover-bottom-line-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --focused-bottom-line-color: var(--ig-textarea-focused-bottom-line-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --interim-bottom-line-color: var(--ig-textarea-interim-bottom-line-color, hsla(var(--ig-primary-500), 0.12));
  --disabled-bottom-line-color: var(--ig-textarea-disabled-bottom-line-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --disabled-border-color: var(--ig-textarea-disabled-border-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --border-color: var(--ig-textarea-border-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --hover-border-color: var(--ig-textarea-hover-border-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --focused-border-color: var(--ig-textarea-focused-border-color, hsla(var(--ig-primary-400), var(--ig-primary-a)));
  --border-background: var(--ig-textarea-border-background, transparent);
  --border-disabled-background: var(--ig-textarea-border-disabled-background, transparent);
  --search-background: var(--ig-textarea-search-background, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --search-disabled-background: var(--ig-textarea-search-disabled-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --success-secondary-color: var(--ig-textarea-success-secondary-color, hsla(var(--ig-success-500), var(--ig-success-a)));
  --warning-secondary-color: var(--ig-textarea-warning-secondary-color, hsla(var(--ig-warn-500), var(--ig-warn-a)));
  --error-secondary-color: var(--ig-textarea-error-secondary-color, hsla(var(--ig-error-500), var(--ig-error-a)));
  --placeholder-color: var(--ig-textarea-placeholder-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --disabled-placeholder-color: var(--ig-textarea-disabled-placeholder-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --box-background: var(--ig-textarea-box-background, transparent);
  --box-background-hover: var(--ig-textarea-box-background-hover, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --box-background-focus: var(--ig-textarea-box-background-focus, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --box-disabled-background: var(--ig-textarea-box-disabled-background, transparent);
  --box-border-radius: var(--ig-textarea-box-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0) * 1.25rem), 1.25rem));
  --border-border-radius: var(--ig-textarea-border-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0) * 1.25rem), 1.25rem));
  --search-border-radius: var(--ig-textarea-search-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.1) * 1.25rem), 1.25rem));
  --success-shadow-color: var(--ig-textarea-success-shadow-color, hsla(var(--ig-success-500), 0.38));
  --error-shadow-color: var(--ig-textarea-error-shadow-color, hsla(var(--ig-error-500), 0.38));
  --size: var(--ig-textarea-size, max(var(--is-large, 1) * max(3.5rem, -1 * 3.5rem), var(--is-medium, 1) * max(3rem, -1 * 3rem), var(--is-small, 1) * max(2.5rem, -1 * 2.5rem)));
  font-family: var(--ig-body-1-font-family, var(--ig-font-family));
  font-size: var(--ig-body-1-font-size);
  font-weight: var(--ig-body-1-font-weight);
  font-style: var(--ig-body-1-font-style);
  line-height: var(--ig-body-1-line-height);
  letter-spacing: var(--ig-body-1-letter-spacing);
  text-transform: var(--ig-body-1-text-transform);
  margin-top: var(--ig-body-1-margin-top);
  margin-bottom: var(--ig-body-1-margin-bottom);
}

:host(:hover) [part~=container] {
  background: var(--box-background-hover);
}

:host(:focus-within) [part~=label] {
  color: var(--focused-secondary-color);
}
:host(:focus-within) [part~=container] {
  background: var(--box-background-hover);
}
:host(:focus-within) [part~=container]::after {
  border-block-end-width: 0.125rem;
  border-block-end-color: var(--focused-bottom-line-color);
  -webkit-transition: all 0.25s cubic-bezier(0.215, 0.61, 0.355, 1);
  transition: all 0.25s cubic-bezier(0.215, 0.61, 0.355, 1);
}

[part~=label] {
  display: block;
  font-size: 0.875rem;
  font-weight: 600;
  position: static;
  -webkit-transform: translateY(0);
      -ms-transform: translateY(0);
          transform: translateY(0);
  -webkit-transform-origin: top left;
      -ms-transform-origin: top left;
          transform-origin: top left;
  margin-top: 0;
  height: auto;
  -webkit-margin-after: 0.25rem;
          margin-block-end: 0.25rem;
  color: var(--idle-secondary-color);
}

[part~=container] {
  border-start-start-radius: var(--box-border-radius);
  border-start-end-radius: var(--box-border-radius);
  -webkit-transition: background 0.25s cubic-bezier(0.215, 0.61, 0.355, 1);
  transition: background 0.25s cubic-bezier(0.215, 0.61, 0.355, 1);
  overflow: hidden;
  -webkit-padding-start: 0;
          padding-inline-start: 0;
  height: calc(100% - 0.125rem);
  top: 0.0625rem;
}
[part~=container]::after {
  content: "";
  position: absolute;
  bottom: 0%;
  width: 100%;
  -webkit-border-after: 0.0625rem solid var(--idle-bottom-line-color);
          border-block-end: 0.0625rem solid var(--idle-bottom-line-color);
  -webkit-transition: all 0.25s cubic-bezier(0.215, 0.61, 0.355, 1);
  transition: all 0.25s cubic-bezier(0.215, 0.61, 0.355, 1);
}

textarea {
  padding-block: 0.5rem;
  padding-inline: max(var(--is-large, 1) * max(1rem, -1 * 1rem) * var(--ig-spacing-inline-large, var(--ig-spacing-inline, --ig-spacing)), var(--is-medium, 1) * max(0.75rem, -1 * 0.75rem) * var(--ig-spacing-inline-medium, var(--ig-spacing-inline, --ig-spacing)), var(--is-small, 1) * max(0.5rem, -1 * 0.5rem) * var(--ig-spacing-inline-small, var(--ig-spacing-inline, --ig-spacing)));
  background: transparent;
  color: var(--filled-text-color);
}
textarea::-webkit-input-placeholder {
  color: var(--placeholder-color);
}
textarea::-moz-placeholder {
  color: var(--placeholder-color);
}
textarea:-ms-input-placeholder {
  color: var(--placeholder-color);
}
textarea::-ms-input-placeholder {
  color: var(--placeholder-color);
}
textarea::placeholder {
  color: var(--placeholder-color);
}

[part~=prefixed] textarea {
  -webkit-padding-start: 0.25rem;
          padding-inline-start: 0.25rem;
}

[part~=suffixed] textarea {
  -webkit-padding-end: 0.25rem;
          padding-inline-end: 0.25rem;
}

[part~=helper-text] {
  font-family: var(--ig-caption-font-family, var(--ig-font-family));
  font-size: var(--ig-caption-font-size);
  font-weight: var(--ig-caption-font-weight);
  font-style: var(--ig-caption-font-style);
  line-height: var(--ig-caption-line-height);
  letter-spacing: var(--ig-caption-letter-spacing);
  text-transform: var(--ig-caption-text-transform);
  margin-top: var(--ig-caption-margin-top);
  margin-bottom: var(--ig-caption-margin-bottom);
  color: var(--helper-text-color);
  -webkit-padding-before: 0.375rem;
          padding-block-start: 0.375rem;
  -webkit-padding-start: 0.75rem;
          padding-inline-start: 0.75rem;
}

[part~=prefixed] [part~=prefix] {
  color: var(--input-prefix-color);
  background: var(--input-prefix-background);
  border-start-start-radius: var(--border-border-radius);
}

[part~=suffixed] [part~=suffix] {
  color: var(--input-suffix-color);
  background: var(--input-suffix-background);
  border-start-end-radius: var(--border-border-radius);
}

[part~=prefixed] [part~=prefix],
[part~=suffixed] [part~=suffix] {
  height: calc(100% - 0.0625rem);
}

[part~=filled] [part=prefix],
[part~=filled] [part=suffix] {
  color: var(--input-prefix-color--filled);
  background: var(--input-prefix-background--filled);
}
[part~=filled] [part~=input] {
  color: var(--input-prefix-color--filled);
}

:host([invalid]:not(:disabled)) [part~=label],
:host([invalid]:not(:disabled)) [part~=helper-text],
:host([invalid]:not([disabled])) [part~=label],
:host([invalid]:not([disabled])) [part~=helper-text] {
  color: var(--error-secondary-color);
}
:host([invalid]:not(:disabled)) [part~=container]::after,
:host([invalid]:not([disabled])) [part~=container]::after {
  -webkit-border-after: 0.0625rem solid var(--error-secondary-color);
          border-block-end: 0.0625rem solid var(--error-secondary-color);
}

:host([invalid]:not(:disabled):focus-within) [part~=container]::after,
:host([invalid]:not([disabled]):focus-within) [part~=container]::after {
  border-block-end-width: 0.125rem;
}

:host(:disabled),
:host([disabled]) {
  color: var(--disabled-text-color);
}
:host(:disabled) [part=prefix],
:host(:disabled) [part=suffix],
:host([disabled]) [part=prefix],
:host([disabled]) [part=suffix] {
  color: var(--disabled-text-color);
  background: inherit;
}
:host(:disabled) [part~=helper-text],
:host([disabled]) [part~=helper-text] {
  color: var(--disabled-text-color);
}
:host(:disabled) [part~=container],
:host([disabled]) [part~=container] {
  background: transparent;
}
:host(:disabled) [part~=container]::after,
:host([disabled]) [part~=container]::after {
  border-block-end-color: var(--disabled-bottom-line-color);
}
:host(:disabled) textarea,
:host([disabled]) textarea {
  color: var(--disabled-text-color);
}
:host(:disabled) textarea::-webkit-input-placeholder, :host([disabled]) textarea::-webkit-input-placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) textarea::-moz-placeholder, :host([disabled]) textarea::-moz-placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) textarea:-ms-input-placeholder, :host([disabled]) textarea:-ms-input-placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) textarea::-ms-input-placeholder, :host([disabled]) textarea::-ms-input-placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) textarea::placeholder,
:host([disabled]) textarea::placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) [part~=label],
:host([disabled]) [part~=label] {
  color: var(--disabled-text-color);
}`;
