// igniteui-webcomponents/components/textarea/themes/light/textarea.fluent.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium: min(
      clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
      clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
  );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

:host([hidden]),
[hidden] {
  display: none !important;
}
:host {
  --search-resting-elevation: var(--ig-textarea-search-resting-elevation, var(--ig-elevation-0));
  --search-hover-elevation: var(--ig-textarea-search-hover-elevation, var(--ig-elevation-0));
  --search-disabled-elevation: var(--ig-textarea-search-disabled-elevation, var(--ig-elevation-0));
  --helper-text-color: var(--ig-textarea-helper-text-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --input-prefix-background: var(--ig-textarea-input-prefix-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --input-prefix-color: var(--ig-textarea-input-prefix-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --input-prefix-background--filled: var(--ig-textarea-input-prefix-background--filled, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --input-prefix-color--filled: var(--ig-textarea-input-prefix-color--filled, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --input-prefix-background--focused: var(--ig-textarea-input-prefix-background--focused, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --input-prefix-color--focused: var(--ig-textarea-input-prefix-color--focused, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --input-suffix-background: var(--ig-textarea-input-suffix-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --input-suffix-color: var(--ig-textarea-input-suffix-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --input-suffix-background--filled: var(--ig-textarea-input-suffix-background--filled, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --input-suffix-color--filled: var(--ig-textarea-input-suffix-color--filled, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --input-suffix-background--focused: var(--ig-textarea-input-suffix-background--focused, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --input-suffix-color--focused: var(--ig-textarea-input-suffix-color--focused, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --idle-text-color: var(--ig-textarea-idle-text-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --filled-text-color: var(--ig-textarea-filled-text-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --focused-text-color: var(--ig-textarea-focused-text-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --disabled-text-color: var(--ig-textarea-disabled-text-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --idle-secondary-color: var(--ig-textarea-idle-secondary-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --filled-secondary-color: var(--ig-textarea-filled-secondary-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --focused-secondary-color: var(--ig-textarea-focused-secondary-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --idle-bottom-line-color: var(--ig-textarea-idle-bottom-line-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --hover-bottom-line-color: var(--ig-textarea-hover-bottom-line-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --focused-bottom-line-color: var(--ig-textarea-focused-bottom-line-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --interim-bottom-line-color: var(--ig-textarea-interim-bottom-line-color, hsla(var(--ig-primary-500), 0.12));
  --disabled-bottom-line-color: var(--ig-textarea-disabled-bottom-line-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --disabled-border-color: var(--ig-textarea-disabled-border-color, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --border-color: var(--ig-textarea-border-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --hover-border-color: var(--ig-textarea-hover-border-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --focused-border-color: var(--ig-textarea-focused-border-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --border-background: var(--ig-textarea-border-background, transparent);
  --border-disabled-background: var(--ig-textarea-border-disabled-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --search-background: var(--ig-textarea-search-background, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --search-disabled-background: var(--ig-textarea-search-disabled-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --success-secondary-color: var(--ig-textarea-success-secondary-color, hsla(var(--ig-success-500), var(--ig-success-a)));
  --warning-secondary-color: var(--ig-textarea-warning-secondary-color, hsla(var(--ig-warn-500), var(--ig-warn-a)));
  --error-secondary-color: var(--ig-textarea-error-secondary-color, hsla(var(--ig-error-500), var(--ig-error-a)));
  --placeholder-color: var(--ig-textarea-placeholder-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --disabled-placeholder-color: var(--ig-textarea-disabled-placeholder-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --box-background: var(--ig-textarea-box-background, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --box-background-hover: var(--ig-textarea-box-background-hover, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --box-background-focus: var(--ig-textarea-box-background-focus, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --box-disabled-background: var(--ig-textarea-box-disabled-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --box-border-radius: var(--ig-textarea-box-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.1) * 1.25rem), 1.25rem));
  --border-border-radius: var(--ig-textarea-border-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.1) * 1.25rem), 1.25rem));
  --search-border-radius: var(--ig-textarea-search-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.1) * 1.25rem), 1.25rem));
  --success-shadow-color: var(--ig-textarea-success-shadow-color, hsla(var(--ig-success-500), 0.38));
  --error-shadow-color: var(--ig-textarea-error-shadow-color, hsla(var(--ig-error-500), 0.38));
  --size: var(--ig-textarea-size, max(var(--is-large, 1) * max(3rem, -1 * 3rem), var(--is-medium, 1) * max(2.5rem, -1 * 2.5rem), var(--is-small, 1) * max(2rem, -1 * 2rem)));
  font-family: var(--ig-body-2-font-family, var(--ig-font-family));
  font-size: var(--ig-body-2-font-size);
  font-weight: var(--ig-body-2-font-weight);
  font-style: var(--ig-body-2-font-style);
  line-height: var(--ig-body-2-line-height);
  letter-spacing: var(--ig-body-2-letter-spacing);
  text-transform: var(--ig-body-2-text-transform);
  margin-top: var(--ig-body-2-margin-top);
  margin-bottom: var(--ig-body-2-margin-bottom);
}

:host(:hover) [part~=container] {
  border-color: var(--hover-border-color);
}

:host(:focus-within) [part~=container] {
  border-color: var(--focused-border-color);
  border-width: 0.125rem;
}
:host(:focus-within) [part~=prefix]:not([hidden]) {
  -webkit-margin-start: -0.0625rem;
          margin-inline-start: -0.0625rem;
}
:host(:focus-within) [part~=suffix]:not([hidden]) {
  -webkit-margin-end: -0.0625rem;
          margin-inline-end: -0.0625rem;
}
:host(:focus-within) textarea {
  margin-block: -0.0625rem;
  -webkit-margin-start: -0.0625rem;
          margin-inline-start: -0.0625rem;
}
:host(:focus-within) [part~=prefix]:not([hidden]) ~ textarea {
  margin-inline: initial;
}

[part~=container] {
  border: 0.0625rem solid var(--border-color);
  background: var(--border-background);
  -webkit-box-align: stretch;
      -ms-flex-align: stretch;
          align-items: stretch;
  overflow: hidden;
  border-radius: var(--border-border-radius);
}

[part~=label] {
  display: block;
  font-size: 0.875rem;
  font-weight: 600;
  position: static;
  -webkit-transform: translateY(0);
      -ms-transform: translateY(0);
          transform: translateY(0);
  -webkit-transform-origin: top left;
      -ms-transform-origin: top left;
          transform-origin: top left;
  margin-top: 0;
  height: auto;
  -webkit-margin-after: 0.25rem;
          margin-block-end: 0.25rem;
  color: var(--idle-secondary-color);
}

:host([invalid]) [part~=label] {
  color: var(--idle-text-color);
}

textarea {
  font-family: var(--ig-body-2-font-family, var(--ig-font-family));
  font-size: var(--ig-body-2-font-size);
  font-weight: var(--ig-body-2-font-weight);
  font-style: var(--ig-body-2-font-style);
  line-height: var(--ig-body-2-line-height);
  letter-spacing: var(--ig-body-2-letter-spacing);
  text-transform: var(--ig-body-2-text-transform);
  margin-top: var(--ig-body-2-margin-top);
  margin-bottom: var(--ig-body-2-margin-bottom);
  padding: 0.5rem;
  background: transparent;
  color: var(--filled-text-color);
}
textarea::-webkit-input-placeholder {
  color: var(--placeholder-color);
}
textarea::-moz-placeholder {
  color: var(--placeholder-color);
}
textarea:-ms-input-placeholder {
  color: var(--placeholder-color);
}
textarea::-ms-input-placeholder {
  color: var(--placeholder-color);
}
textarea::placeholder {
  color: var(--placeholder-color);
}

[part~=prefixed] [part~=prefix] {
  color: var(--input-prefix-color);
  background: var(--input-prefix-background);
}

[part~=suffixed] [part~=suffix] {
  color: var(--input-suffix-color);
  background: var(--input-suffix-background);
}

[part~=filled] [part=prefix],
[part~=filled] [part=suffix] {
  color: var(--input-prefix-color--filled);
  background: var(--input-prefix-background--filled);
}
[part~=filled] [part~=input] {
  color: var(--input-prefix-color--filled);
}

[part~=helper-text] {
  font-family: var(--ig-caption-font-family, var(--ig-font-family));
  font-size: var(--ig-caption-font-size);
  font-weight: var(--ig-caption-font-weight);
  font-style: var(--ig-caption-font-style);
  line-height: var(--ig-caption-line-height);
  letter-spacing: var(--ig-caption-letter-spacing);
  text-transform: var(--ig-caption-text-transform);
  margin-top: var(--ig-caption-margin-top);
  margin-bottom: var(--ig-caption-margin-bottom);
  color: var(--helper-text-color);
  -webkit-padding-before: 0.3125rem;
          padding-block-start: 0.3125rem;
  -webkit-padding-start: 0;
          padding-inline-start: 0;
}

:host([required]:not(:disabled)) [part~=label]::after,
:host([required]:not([disabled])) [part~=label]::after {
  color: var(--error-secondary-color);
}

:host([invalid]:not(:disabled)) [part~=container],
:host([invalid]:not([disabled])) [part~=container] {
  border-color: var(--error-secondary-color);
}
:host([invalid]:not(:disabled)) [part~=helper-text],
:host([invalid]:not([disabled])) [part~=helper-text] {
  color: var(--error-secondary-color);
}

:host(:disabled),
:host([disabled]) {
  color: var(--disabled-text-color);
}
:host(:disabled) [part=prefix],
:host(:disabled) [part=suffix],
:host([disabled]) [part=prefix],
:host([disabled]) [part=suffix] {
  color: var(--disabled-text-color);
  background: inherit;
}
:host(:disabled) [part~=helper-text],
:host([disabled]) [part~=helper-text] {
  color: var(--disabled-text-color);
}
:host(:disabled) textarea,
:host([disabled]) textarea {
  color: var(--disabled-text-color);
}
:host(:disabled) textarea::-webkit-input-placeholder, :host([disabled]) textarea::-webkit-input-placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) textarea::-moz-placeholder, :host([disabled]) textarea::-moz-placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) textarea:-ms-input-placeholder, :host([disabled]) textarea:-ms-input-placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) textarea::-ms-input-placeholder, :host([disabled]) textarea::-ms-input-placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) textarea::placeholder,
:host([disabled]) textarea::placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) [part~=label],
:host([disabled]) [part~=label] {
  color: var(--disabled-text-color);
}
:host(:disabled) [part~=container],
:host([disabled]) [part~=container] {
  border-color: var(--disabled-border-color);
  background: var(--border-disabled-background);
}`;
