// igniteui-webcomponents/components/textarea/themes/light/textarea.material.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium: min(
      clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
      clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
  );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

:host([hidden]),
[hidden] {
  display: none !important;
}
:host {
  --search-resting-elevation: var(--ig-textarea-search-resting-elevation, var(--ig-elevation-1));
  --search-hover-elevation: var(--ig-textarea-search-hover-elevation, var(--ig-elevation-2));
  --search-disabled-elevation: var(--ig-textarea-search-disabled-elevation, var(--ig-elevation-0));
  --helper-text-color: var(--ig-textarea-helper-text-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --input-prefix-background: var(--ig-textarea-input-prefix-background, transparent);
  --input-prefix-color: var(--ig-textarea-input-prefix-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --input-prefix-background--filled: var(--ig-textarea-input-prefix-background--filled, inherit);
  --input-prefix-color--filled: var(--ig-textarea-input-prefix-color--filled, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --input-prefix-background--focused: var(--ig-textarea-input-prefix-background--focused, inherit);
  --input-prefix-color--focused: var(--ig-textarea-input-prefix-color--focused, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --input-suffix-background: var(--ig-textarea-input-suffix-background, transparent);
  --input-suffix-color: var(--ig-textarea-input-suffix-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --input-suffix-background--filled: var(--ig-textarea-input-suffix-background--filled, inherit);
  --input-suffix-color--filled: var(--ig-textarea-input-suffix-color--filled, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --input-suffix-background--focused: var(--ig-textarea-input-suffix-background--focused, inherit);
  --input-suffix-color--focused: var(--ig-textarea-input-suffix-color--focused, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --idle-text-color: var(--ig-textarea-idle-text-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --filled-text-color: var(--ig-textarea-filled-text-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --focused-text-color: var(--ig-textarea-focused-text-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --disabled-text-color: var(--ig-textarea-disabled-text-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --idle-secondary-color: var(--ig-textarea-idle-secondary-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --filled-secondary-color: var(--ig-textarea-filled-secondary-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --focused-secondary-color: var(--ig-textarea-focused-secondary-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --idle-bottom-line-color: var(--ig-textarea-idle-bottom-line-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --hover-bottom-line-color: var(--ig-textarea-hover-bottom-line-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --focused-bottom-line-color: var(--ig-textarea-focused-bottom-line-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --interim-bottom-line-color: var(--ig-textarea-interim-bottom-line-color, hsla(var(--ig-primary-500), 0.12));
  --disabled-bottom-line-color: var(--ig-textarea-disabled-bottom-line-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --disabled-border-color: var(--ig-textarea-disabled-border-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --border-color: var(--ig-textarea-border-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --hover-border-color: var(--ig-textarea-hover-border-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --focused-border-color: var(--ig-textarea-focused-border-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --border-background: var(--ig-textarea-border-background, transparent);
  --border-disabled-background: var(--ig-textarea-border-disabled-background, transparent);
  --search-background: var(--ig-textarea-search-background, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --search-disabled-background: var(--ig-textarea-search-disabled-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --success-secondary-color: var(--ig-textarea-success-secondary-color, hsla(var(--ig-success-500), var(--ig-success-a)));
  --warning-secondary-color: var(--ig-textarea-warning-secondary-color, hsla(var(--ig-warn-500), var(--ig-warn-a)));
  --error-secondary-color: var(--ig-textarea-error-secondary-color, hsla(var(--ig-error-500), var(--ig-error-a)));
  --placeholder-color: var(--ig-textarea-placeholder-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --disabled-placeholder-color: var(--ig-textarea-disabled-placeholder-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --box-background: var(--ig-textarea-box-background, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --box-background-hover: var(--ig-textarea-box-background-hover, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --box-background-focus: var(--ig-textarea-box-background-focus, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --box-disabled-background: var(--ig-textarea-box-disabled-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --box-border-radius: var(--ig-textarea-box-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.2) * 1.25rem), 1.25rem));
  --border-border-radius: var(--ig-textarea-border-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.2) * 1.25rem), 1.25rem));
  --search-border-radius: var(--ig-textarea-search-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.2) * 1.25rem), 1.25rem));
  --success-shadow-color: var(--ig-textarea-success-shadow-color, hsla(var(--ig-success-500), 0.38));
  --error-shadow-color: var(--ig-textarea-error-shadow-color, hsla(var(--ig-error-500), 0.38));
  --size: var(--ig-textarea-size, max(var(--is-large, 1) * max(3.5rem, -1 * 3.5rem), var(--is-medium, 1) * max(3rem, -1 * 3rem), var(--is-small, 1) * max(2.5rem, -1 * 2.5rem)));
  font-family: var(--ig-subtitle-1-font-family, var(--ig-font-family));
  font-size: var(--ig-subtitle-1-font-size);
  font-weight: var(--ig-subtitle-1-font-weight);
  font-style: var(--ig-subtitle-1-font-style);
  line-height: var(--ig-subtitle-1-line-height);
  letter-spacing: var(--ig-subtitle-1-letter-spacing);
  text-transform: var(--ig-subtitle-1-text-transform);
  margin-top: var(--ig-subtitle-1-margin-top);
  margin-bottom: var(--ig-subtitle-1-margin-bottom);
}

textarea {
  border: none;
  padding: 0;
  overflow: auto;
  -webkit-margin-before: 1.25rem;
          margin-block-start: 1.25rem;
  grid-area: 1/2/span 1/span 2;
  -webkit-box-flex: 1;
      -ms-flex-positive: 1;
          flex-grow: 1;
  position: relative;
  max-width: inherit;
  padding-inline: 0.25rem;
  background: transparent;
  color: var(--filled-text-color);
}
textarea::-webkit-input-placeholder {
  color: var(--placeholder-color);
}
textarea::-moz-placeholder {
  color: var(--placeholder-color);
}
textarea:-ms-input-placeholder {
  color: var(--placeholder-color);
}
textarea::-ms-input-placeholder {
  color: var(--placeholder-color);
}
textarea::placeholder {
  color: var(--placeholder-color);
}

:host(:not([outlined])) [part~=container] {
  overflow: hidden;
  border-start-start-radius: var(--box-border-radius);
  border-start-end-radius: var(--box-border-radius);
  background: var(--box-background);
}
:host(:not([outlined])) [part~=container]::after {
  content: "";
  position: absolute;
  bottom: 0%;
  width: 100%;
  -webkit-border-after: 0.0625rem solid var(--idle-bottom-line-color);
          border-block-end: 0.0625rem solid var(--idle-bottom-line-color);
  -webkit-transition: all 0.25s cubic-bezier(0.215, 0.61, 0.355, 1);
  transition: all 0.25s cubic-bezier(0.215, 0.61, 0.355, 1);
}

:host(:not([outlined]):hover) [part~=container] {
  background: var(--box-background-hover);
}
:host(:not([outlined]):hover) [part~=container]::after {
  border-block-end-width: 0.0625rem;
  border-block-end-color: var(--hover-bottom-line-color);
}

:host(:disabled),
:host([disabled]) {
  color: var(--disabled-text-color);
}
:host(:disabled) [part=prefix],
:host(:disabled) [part=suffix],
:host([disabled]) [part=prefix],
:host([disabled]) [part=suffix] {
  color: var(--disabled-text-color);
  background: inherit;
}
:host(:disabled) [part~=helper-text],
:host([disabled]) [part~=helper-text] {
  color: var(--disabled-text-color);
}
:host(:disabled) textarea,
:host([disabled]) textarea {
  color: var(--disabled-text-color);
}
:host(:disabled) textarea::-webkit-input-placeholder, :host([disabled]) textarea::-webkit-input-placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) textarea::-moz-placeholder, :host([disabled]) textarea::-moz-placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) textarea:-ms-input-placeholder, :host([disabled]) textarea:-ms-input-placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) textarea::-ms-input-placeholder, :host([disabled]) textarea::-ms-input-placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) textarea::placeholder,
:host([disabled]) textarea::placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) [part~=label],
:host([disabled]) [part~=label] {
  color: var(--disabled-text-color);
}

:host(:disabled:not([outlined])) [part~=container],
:host([disabled]:not([outlined])) [part~=container] {
  background: var(--box-disabled-background);
}
:host(:disabled:not([outlined])) [part~=container]::after,
:host([disabled]:not([outlined])) [part~=container]::after {
  border-block-end-color: var(--disabled-bottom-line-color);
  border-block-end-style: dashed;
}

:host([invalid]:not(:disabled)) [part~=container]::after,
:host([invalid]:not([disabled])) [part~=container]::after {
  border-block-end-color: var(--error-secondary-color);
}
:host([invalid]:not(:disabled)) [part~=label],
:host([invalid]:not([disabled])) [part~=label] {
  color: var(--error-secondary-color);
}
:host([invalid]:not(:disabled)) [part~=helper-text],
:host([invalid]:not([disabled])) [part~=helper-text] {
  color: var(--error-secondary-color);
}

:host(:focus-within) [part~=label] {
  color: var(--focused-secondary-color);
}

:host(:not([outlined]):focus-within) {
  color: var(--input-prefix-color--focused);
}
:host(:not([outlined]):focus-within) [part~=container] {
  background: var(--box-background-focus);
}
:host(:not([outlined]):focus-within) [part~=container]::after {
  border-block-end-width: 0.125rem;
  border-block-end-color: var(--focused-bottom-line-color);
}
:host(:not([outlined]):focus-within) [part~=prefix] {
  color: var(--input-prefix-color--focused);
  background: var(--input-prefix-background--focused);
}
:host(:not([outlined]):focus-within) [part~=suffix] {
  color: var(--input-suffix-color--focused);
  background: var(--input-suffix-background--focused);
}

:host([invalid]:not([outlined]):focus-within) [part~=container]::after {
  border-block-end-color: var(--error-secondary-color);
}

[part~=label] {
  position: absolute;
  max-width: 100%;
  -webkit-padding-end: 0.25rem;
          padding-inline-end: 0.25rem;
  line-height: normal;
  -webkit-backface-visibility: hidden;
          backface-visibility: hidden;
  will-change: transform;
  -webkit-transform-origin: top left;
      -ms-transform-origin: top left;
          transform-origin: top left;
  color: var(--idle-secondary-color);
  -webkit-transform: translateY(0);
      -ms-transform: translateY(0);
          transform: translateY(0);
  top: calc(1.25rem / 4);
  font-size: 0.75rem;
}

[part=start],
[part=end] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  min-width: max(var(--is-large, 1) * max(0.875rem, -1 * 0.875rem) * var(--ig-spacing-large, var(--ig-spacing, --ig-spacing)), var(--is-medium, 1) * max(0.75rem, -1 * 0.75rem) * var(--ig-spacing-medium, var(--ig-spacing, --ig-spacing)), var(--is-small, 1) * max(0.625rem, -1 * 0.625rem) * var(--ig-spacing-small, var(--ig-spacing, --ig-spacing)));
  min-height: 100%;
  overflow: hidden;
}

[part=start] {
  grid-area: 1/1;
}

[part=notch] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  width: auto;
  min-width: 0;
  height: 100%;
  position: relative;
  grid-area: 1/2;
  padding-inline: 0.25rem;
}

[part=filler] {
  grid-area: 1/3;
}

[part=end] {
  grid-area: 1/4;
}

[part~=prefixed] [part~=prefix] {
  color: var(--input-prefix-color);
  background: var(--input-prefix-background);
  padding-inline: max(var(--is-large, 1) * max(1rem, -1 * 1rem) * var(--ig-spacing-inline-large, var(--ig-spacing-inline, --ig-spacing)), var(--is-medium, 1) * max(0.875rem, -1 * 0.875rem) * var(--ig-spacing-inline-medium, var(--ig-spacing-inline, --ig-spacing)), var(--is-small, 1) * max(0.75rem, -1 * 0.75rem) * var(--ig-spacing-inline-small, var(--ig-spacing-inline, --ig-spacing)));
}

[part~=suffixed] [part~=suffix] {
  color: var(--input-suffix-color);
  background: var(--input-suffix-background);
  padding-inline: max(var(--is-large, 1) * max(1rem, -1 * 1rem) * var(--ig-spacing-inline-large, var(--ig-spacing-inline, --ig-spacing)), var(--is-medium, 1) * max(0.875rem, -1 * 0.875rem) * var(--ig-spacing-inline-medium, var(--ig-spacing-inline, --ig-spacing)), var(--is-small, 1) * max(0.75rem, -1 * 0.75rem) * var(--ig-spacing-inline-small, var(--ig-spacing-inline, --ig-spacing)));
}

[part~=filled] {
  color: var(--input-prefix-color--filled);
}
[part~=filled] [part~=prefix] {
  color: var(--input-prefix-color--filled);
  background: var(--input-prefix-background--filled);
}
[part~=filled] [part~=suffix] {
  color: var(--input-suffix-color--filled);
  background: var(--input-suffix-background--filled);
}

[part~=helper-text] {
  font-family: var(--ig-caption-font-family, var(--ig-font-family));
  font-size: var(--ig-caption-font-size);
  font-weight: var(--ig-caption-font-weight);
  font-style: var(--ig-caption-font-style);
  line-height: var(--ig-caption-line-height);
  letter-spacing: var(--ig-caption-letter-spacing);
  text-transform: var(--ig-caption-text-transform);
  margin-top: var(--ig-caption-margin-top);
  margin-bottom: var(--ig-caption-margin-bottom);
  color: var(--helper-text-color);
  -webkit-padding-before: 0.3125rem;
          padding-block-start: 0.3125rem;
  -webkit-padding-start: 1rem;
          padding-inline-start: 1rem;
}

:host([outlined]) {
  -webkit-box-align: initial;
      -ms-flex-align: initial;
          align-items: initial;
  padding: 0;
  -webkit-box-shadow: none;
          box-shadow: none;
  border-radius: var(--border-border-radius);
  background: var(--border-background);
  overflow: initial;
}
:host([outlined]) [part~=container] {
  grid-template-columns: auto auto 1fr auto;
  display: grid;
}
:host([outlined]) [part=start] {
  width: auto;
  border-color: var(--border-color);
  border-style: solid;
  border-inline-start-width: 0.0625rem;
  border-inline-end-width: 0;
  border-block-start-width: 0.0625rem;
  border-block-end-width: 0.0625rem;
  border-start-start-radius: var(--border-border-radius);
  border-end-start-radius: var(--border-border-radius);
}
:host([outlined]) [part~=label] {
  top: 0;
  -webkit-transform: translateY(-50%);
      -ms-transform: translateY(-50%);
          transform: translateY(-50%);
  -webkit-margin-after: auto;
          margin-block-end: auto;
  -webkit-padding-end: 0;
          padding-inline-end: 0;
  display: inline-block;
  position: relative;
  background: transparent;
}
:host([outlined]) [part=filler] {
  border-width: 0.0625rem;
  border-style: solid;
  border-color: var(--border-color);
  border-left: none;
  border-right: none;
}
:host([outlined]) [part=notch] {
  border-block-start-color: transparent;
  -webkit-border-after: 0.0625rem solid var(--border-color);
          border-block-end: 0.0625rem solid var(--border-color);
  padding: 0 0.25rem;
}
:host([outlined]) [part=notch]:empty {
  display: none;
}
:host([outlined]) [part=end] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
      -ms-flex-pack: end;
          justify-content: flex-end;
  -webkit-box-flex: 1;
      -ms-flex-positive: 1;
          flex-grow: 1;
  height: 100%;
  grid-area: 1/4;
  border-color: var(--border-color);
  border-style: solid;
  border-inline-start-width: 0;
  border-inline-end-width: 0.0625rem;
  border-block-start-width: 0.0625rem;
  border-block-end-width: 0.0625rem;
  border-start-end-radius: var(--border-border-radius);
  border-end-end-radius: var(--border-border-radius);
}
:host([outlined]) textarea {
  bottom: 0.125rem;
}

:host([outlined]:focus-within) [part=filler],
:host([outlined]:focus-within) [part=notch] {
  border-block-width: 0.125rem;
}
:host([outlined]:focus-within) [part=filler] {
  border-block-color: var(--focused-border-color);
}
:host([outlined]:focus-within) [part=notch] {
  border-block-end-color: var(--focused-border-color);
}
:host([outlined]:focus-within) [part=start] {
  border-color: var(--focused-border-color);
  border-inline-start-width: 0.125rem;
  border-inline-end-width: 0;
  border-block-start-width: 0.125rem;
  border-block-end-width: 0.125rem;
}
:host([outlined]:focus-within) [part=end] {
  border-color: var(--focused-border-color);
  border-inline-start-width: 0;
  border-inline-end-width: 0.125rem;
  border-block-start-width: 0.125rem;
  border-block-end-width: 0.125rem;
}
:host([outlined]:focus-within) [part~=prefix] {
  -webkit-margin-start: -0.0625rem;
          margin-inline-start: -0.0625rem;
}
:host([outlined]:focus-within) [part~=suffix] {
  -webkit-margin-end: -0.0625rem;
          margin-inline-end: -0.0625rem;
}

:host([invalid][outlined]:not(:disabled)) [part=start],
:host([invalid][outlined]:not(:disabled)) [part=filler],
:host([invalid][outlined]:not(:disabled)) [part=notch],
:host([invalid][outlined]:not(:disabled)) [part=end],
:host([invalid][outlined]:not(:disabled):focus-within) [part=start],
:host([invalid][outlined]:not(:disabled):focus-within) [part=filler],
:host([invalid][outlined]:not(:disabled):focus-within) [part=notch],
:host([invalid][outlined]:not(:disabled):focus-within) [part=end],
:host([invalid][outlined]:not([disabled])) [part=start],
:host([invalid][outlined]:not([disabled])) [part=filler],
:host([invalid][outlined]:not([disabled])) [part=notch],
:host([invalid][outlined]:not([disabled])) [part=end],
:host([invalid][outlined]:not([disabled]):focus-within) [part=start],
:host([invalid][outlined]:not([disabled]):focus-within) [part=filler],
:host([invalid][outlined]:not([disabled]):focus-within) [part=notch],
:host([invalid][outlined]:not([disabled]):focus-within) [part=end] {
  border-color: var(--error-secondary-color);
}

:host(:disabled[outlined]) [part~=container],
:host([disabled][outlined]) [part~=container] {
  background: var(--border-disabled-background);
}
:host(:disabled[outlined]) [part=start],
:host(:disabled[outlined]) [part=filler],
:host(:disabled[outlined]) [part=notch],
:host(:disabled[outlined]) [part=end],
:host([disabled][outlined]) [part=start],
:host([disabled][outlined]) [part=filler],
:host([disabled][outlined]) [part=notch],
:host([disabled][outlined]) [part=end] {
  border-color: var(--disabled-border-color);
}`;
