// igniteui-webcomponents/components/card/card.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { html, LitElement } from 'lit';
import { property } from 'lit/decorators.js';
import { themes } from '../../theming/theming-decorator.js';
import { styles } from './themes/light/card.base.css.js';
import { styles as material } from './themes/light/card.material.css.js';
import { styles as bootstrap } from './themes/light/card.bootstrap.css.js';
import { styles as fluent } from './themes/light/card.fluent.css.js';
import { styles as indigo } from './themes/light/card.indigo.css.js';
import { registerComponent } from '../common/definitions/register.js';
import IgcCardActionsComponent from './card.actions.js';
import IgcCardContentComponent from './card.content.js';
import IgcCardHeaderComponent from './card.header.js';
import IgcCardMediaComponent from './card.media.js';
let IgcCardComponent = class IgcCardComponent extends LitElement {
    constructor() {
        super(...arguments);
        this.elevated = false;
    }
    static register() {
        registerComponent(this, IgcCardActionsComponent, IgcCardContentComponent, IgcCardHeaderComponent, IgcCardMediaComponent);
    }
    render() {
        return html ` <slot></slot> `;
    }
};
IgcCardComponent.tagName = 'igc-card';
IgcCardComponent.styles = styles;
__decorate([
    property({ type: Boolean, reflect: true })
], IgcCardComponent.prototype, "elevated", void 0);
IgcCardComponent = __decorate([
    themes({
        light: {
            material,
            bootstrap,
            fluent,
            indigo,
        },
        dark: {
            material,
            bootstrap,
            fluent,
            indigo,
        },
    })
], IgcCardComponent);
export default IgcCardComponent;
