// igniteui-webcomponents/components/chip/themes/light/chip.indigo.css.js
import { css } from 'lit';
export const styles = css `:host {
  --ghost-elevation: var(--ig-chip-ghost-elevation, var(--ig-elevation-1));
  --remove-icon-color: var(--ig-chip-remove-icon-color, 'currentColor');
  --focus-outline-color: var(--ig-chip-focus-outline-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --focus-selected-outline-color: var(--ig-chip-focus-selected-outline-color, hsla(var(--ig-primary-200), var(--ig-primary-a)));
  --remove-icon-color-focus: var(--ig-chip-remove-icon-color-focus, 'currentColor');
  --text-color: var(--ig-chip-text-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --background: var(--ig-chip-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --border-color: var(--ig-chip-border-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --ghost-background: var(--ig-chip-ghost-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --hover-text-color: var(--ig-chip-hover-text-color, var(--ig-gray-400-contrast));
  --hover-background: var(--ig-chip-hover-background, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --hover-border-color: var(--ig-chip-hover-border-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --focus-text-color: var(--ig-chip-focus-text-color, var(--ig-gray-500-contrast));
  --focus-background: var(--ig-chip-focus-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --focus-border-color: var(--ig-chip-focus-border-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --selected-text-color: var(--ig-chip-selected-text-color, var(--ig-primary-500-contrast));
  --selected-background: var(--ig-chip-selected-background, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --selected-border-color: var(--ig-chip-selected-border-color, transparent);
  --hover-selected-text-color: var(--ig-chip-hover-selected-text-color, var(--ig-primary-400-contrast));
  --hover-selected-background: var(--ig-chip-hover-selected-background, hsla(var(--ig-primary-400), var(--ig-primary-a)));
  --hover-selected-border-color: var(--ig-chip-hover-selected-border-color, transparent);
  --focus-selected-text-color: var(--ig-chip-focus-selected-text-color, var(--ig-primary-500-contrast));
  --focus-selected-background: var(--ig-chip-focus-selected-background, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --focus-selected-border-color: var(--ig-chip-focus-selected-border-color, transparent);
  --border-radius: var(--ig-chip-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 1) * 1rem), 1rem));
  --disabled-text-color: var(--ig-chip-disabled-text-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --disabled-background: var(--ig-chip-disabled-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --disabled-border-color: var(--ig-chip-disabled-border-color, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --size: var(--ig-chip-size, max(var(--is-large, 1) * max(2rem, -1 * 2rem), var(--is-medium, 1) * max(1.5rem, -1 * 1.5rem), var(--is-small, 1) * max(1.25rem, -1 * 1.25rem)));
}

:host button {
  border-radius: var(--border-radius);
  border: 0.0625rem solid var(--border-color);
  background: var(--background);
  color: var(--text-color);
  height: var(--size);
}
:host button:hover {
  background: var(--hover-background);
  color: var(--hover-text-color);
}
:host button:focus {
  background: var(--focus-background);
  color: var(--focus-text-color);
  outline: 0.1875rem solid var(--focus-outline-color);
}
:host button[disabled] {
  border-color: var(--disabled-border-color);
  background: var(--disabled-background);
  color: var(--disabled-text-color);
}

:host([selected]) button:not([disabled]) {
  border-color: transparent;
  background: var(--selected-background);
  color: var(--selected-text-color);
}
:host([selected]) button:not([disabled]):hover {
  background: var(--hover-selected-background);
}
:host([selected]) button:not([disabled]):focus {
  outline: 0.1875rem solid var(--focus-selected-outline-color);
}

:host([variant=primary]) button,
:host([selected][variant=primary]) button:not([disabled]) {
  border-color: transparent;
}
:host([variant=primary]) button:hover,
:host([selected][variant=primary]) button:not([disabled]):hover {
  background: hsla(var(--ig-primary-400), var(--ig-primary-a));
  color: var(--ig-primary-400-contrast);
}
:host([variant=primary]) button:focus,
:host([selected][variant=primary]) button:not([disabled]):focus {
  background: hsla(var(--ig-primary-500), var(--ig-primary-a));
  color: var(--ig-primary-500-contrast);
  outline: 0.1875rem solid var(--focus-selected-outline-color);
}

:host([variant=primary]) button[disabled] {
  background: hsla(var(--ig-primary-200), var(--ig-primary-a));
  color: var(--ig-primary-400-contrast);
}

:host([variant=info]) button,
:host([selected][variant=info]) button:not([disabled]) {
  border-color: transparent;
}
:host([variant=info]) button:hover,
:host([selected][variant=info]) button:not([disabled]):hover {
  background: hsla(var(--ig-info-400), var(--ig-info-a));
  color: var(--ig-info-400-contrast);
}
:host([variant=info]) button:focus,
:host([selected][variant=info]) button:not([disabled]):focus {
  background: hsla(var(--ig-info-500), var(--ig-info-a));
  color: var(--ig-info-500-contrast);
  outline: 0.1875rem solid hsla(var(--ig-info-200), var(--ig-info-a));
}

:host([variant=info]) button[disabled] {
  background: hsla(var(--ig-info-200), var(--ig-info-a));
  color: var(--ig-info-400-contrast);
}

:host([variant=success]) button,
:host([selected][variant=success]) button:not([disabled]) {
  border-color: transparent;
}
:host([variant=success]) button:hover,
:host([selected][variant=success]) button:not([disabled]):hover {
  background: hsla(var(--ig-success-400), var(--ig-success-a));
  color: var(--ig-success-400-contrast);
}
:host([variant=success]) button:focus,
:host([selected][variant=success]) button:not([disabled]):focus {
  background: hsla(var(--ig-success-500), var(--ig-success-a));
  color: var(--ig-success-500-contrast);
  outline: 0.1875rem solid hsla(var(--ig-success-200), var(--ig-success-a));
}

:host([variant=success]) button[disabled] {
  background: hsla(var(--ig-success-200), var(--ig-success-a));
  color: var(--ig-success-800-contrast);
}

:host([variant=warning]) button,
:host([selected][variant=warning]) button:not([disabled]) {
  border-color: transparent;
}
:host([variant=warning]) button:hover,
:host([selected][variant=warning]) button:not([disabled]):hover {
  background: hsla(var(--ig-warn-400), var(--ig-warn-a));
  color: var(--ig-warn-400-contrast);
}
:host([variant=warning]) button:focus,
:host([selected][variant=warning]) button:not([disabled]):focus {
  background: hsla(var(--ig-warn-500), var(--ig-warn-a));
  color: var(--ig-warn-500-contrast);
  outline: 0.1875rem solid hsla(var(--ig-warn-200), var(--ig-warn-a));
}

:host([variant=warning]) button[disabled] {
  background: hsla(var(--ig-warn-200), var(--ig-warn-a));
  color: var(--ig-warn-900-contrast);
}

:host([variant=danger]) button,
:host([selected][variant=danger]) button:not([disabled]) {
  border-color: transparent;
}
:host([variant=danger]) button:hover,
:host([selected][variant=danger]) button:not([disabled]):hover {
  background: hsla(var(--ig-error-400), var(--ig-error-a));
  color: var(--ig-error-400-contrast);
}
:host([variant=danger]) button:focus,
:host([selected][variant=danger]) button:not([disabled]):focus {
  background: hsla(var(--ig-error-500), var(--ig-error-a));
  color: var(--ig-error-500-contrast);
  outline: 0.1875rem solid hsla(var(--ig-error-200), var(--ig-error-a));
}

:host([variant=danger]) button[disabled] {
  background: hsla(var(--ig-error-200), var(--ig-error-a));
  color: var(--ig-error-400-contrast);
}`;
