// igniteui-webcomponents/components/card/themes/light/card.fluent.css.js
import { css } from 'lit';
export const styles = css `:host {
  --resting-elevation: var(--ig-card-resting-elevation, var(--ig-elevation-4));
  --hover-elevation: var(--ig-card-hover-elevation, var(--ig-elevation-6));
  --background: var(--ig-card-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --header-text-color: var(--ig-card-header-text-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --subtitle-text-color: var(--ig-card-subtitle-text-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --content-text-color: var(--ig-card-content-text-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --actions-text-color: var(--ig-card-actions-text-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --outline-color: var(--ig-card-outline-color, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --border-radius: var(--ig-card-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.0833333333) * 1.5rem), 1.5rem));
  border: 0.0625rem solid var(--outline-color);
  background: var(--background);
  border-radius: var(--border-radius);
}

:host([elevated]) {
  -webkit-box-shadow: var(--resting-elevation);
          box-shadow: var(--resting-elevation);
}

:host([elevated]:hover),
:host([elevated]:focus-within) {
  -webkit-box-shadow: var(--hover-elevation);
          box-shadow: var(--hover-elevation);
}
:host([elevated]:hover)::after,
:host([elevated]:focus-within)::after {
  position: absolute;
  content: "";
  width: 100%;
  height: 100%;
  -webkit-transition: -webkit-box-shadow 0.1s ease-out;
  transition: -webkit-box-shadow 0.1s ease-out;
  transition: box-shadow 0.1s ease-out;
  transition: box-shadow 0.1s ease-out, -webkit-box-shadow 0.1s ease-out;
  -webkit-box-shadow: inset 0 0 0 0.125rem var(--outline-color);
          box-shadow: inset 0 0 0 0.125rem var(--outline-color);
  pointer-events: none;
}`;
