// igniteui-webcomponents/components/nav-drawer/themes/light/nav-drawer.base.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium: min(
      clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
      clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
  );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

:host([hidden]),
[hidden] {
  display: none !important;
}
:host {
  --elevation: var(--ig-nav-drawer-elevation, var(--ig-elevation-16));
  --background: var(--ig-nav-drawer-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --border-color: var(--ig-nav-drawer-border-color, hsla(var(--ig-gray-900), 0.14));
  --item-header-text-color: var(--ig-nav-drawer-item-header-text-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --item-text-color: var(--ig-nav-drawer-item-text-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --item-icon-color: var(--ig-nav-drawer-item-icon-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --item-active-text-color: var(--ig-nav-drawer-item-active-text-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --item-active-background: var(--ig-nav-drawer-item-active-background, hsla(var(--ig-primary-500), 0.12));
  --item-hover-background: var(--ig-nav-drawer-item-hover-background, hsla(var(--ig-gray-200), 0.8));
  --item-hover-text-color: var(--ig-nav-drawer-item-hover-text-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --item-active-icon-color: var(--ig-nav-drawer-item-active-icon-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --item-hover-icon-color: var(--ig-nav-drawer-item-hover-icon-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --border-radius: var(--ig-nav-drawer-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0) * 2.25rem), 2.25rem));
  --item-border-radius: var(--ig-nav-drawer-item-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.1666666667) * 1.5rem), 1.5rem));
  --menu-full-width: 16rem;
  --menu-mini-width: 3.75rem;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-negative: 0;
      flex-shrink: 0;
  font-family: var(--ig-font-family);
}
:host [part=base],
:host [part~=mini] {
  --ig-scrollbar-size: rem(8px);
  position: fixed;
  height: 100%;
  min-height: 100%;
  overflow-x: hidden;
  inset-block: 0;
  z-index: 999;
  -webkit-transition: 0.3s, -webkit-transform ease-out 0.3s;
  transition: 0.3s, -webkit-transform ease-out 0.3s;
  transition: transform ease-out 0.3s, 0.3s;
  transition: transform ease-out 0.3s, 0.3s, -webkit-transform ease-out 0.3s;
}
:host [part~=mini] {
  min-width: var(--menu-mini-width);
}
:host [part=main] {
  width: var(--menu-full-width);
}
:host [part=overlay] {
  position: fixed;
  inset: 0;
  z-index: 2;
  opacity: 1;
  -webkit-transition: opacity ease-out 0.3s, 0.3s;
  transition: opacity ease-out 0.3s, 0.3s;
}

:host([position=relative]) [part=base] {
  position: relative;
  width: var(--menu-full-width);
  -webkit-box-shadow: none;
          box-shadow: none;
  z-index: initial;
  border-inline-end-style: solid;
  border-inline-end-width: 0.0625rem;
}
:host([position=relative]) [part~=mini] {
  position: relative;
  border-inline-end-style: solid;
  border-inline-end-width: 0.0625rem;
}
:host([position=relative]) [part=overlay] {
  display: none;
}

:host([position=top]) [part=main],
:host([position=bottom]) [part=main] {
  width: 100%;
}
:host([position=top]) [part=base],
:host([position=bottom]) [part=base] {
  height: 50vh;
  min-height: initial;
  width: 100%;
  inset-inline: 0;
}

:host([position=top]) [part=base] {
  top: 0;
  bottom: auto;
}

:host([position=bottom]) [part=base] {
  top: auto;
  bottom: 0;
}

:host([position=start]) [part=base],
:host([position=start]) [part~=mini] {
  inset-inline-start: 0;
  border-inline-end-style: solid;
  border-inline-end-width: 0.0625rem;
}

:host([position=end]) [part=base],
:host([position=end]) [part~=mini] {
  inset-inline-end: 0;
  border-inline-start-style: solid;
  border-inline-start-width: 0.0625rem;
}

:host(:not([open])[position=start]),
:host([dir=rtl]:not([open])[position=end]) {
  --dir: calc(var(--ig-dir, 1) * -1);
}

:host(:not([open])[position=end]),
:host([dir=rtl]:not([open])[position=start]) {
  --dir: calc(var(--ig-dir, 1) * 1);
}

:host(:not([open])[position=relative]) [part=base] {
  -webkit-margin-start: calc(var(--menu-full-width) * -1);
          margin-inline-start: calc(var(--menu-full-width) * -1);
}

:host(:not([open])[position=top]) [part=base],
:host(:not([open])[position=bottom]) [part=base] {
  -webkit-transform: translateY(-60vh);
      -ms-transform: translateY(-60vh);
          transform: translateY(-60vh);
}
:host(:not([open])[position=top]) [part=mini],
:host(:not([open])[position=bottom]) [part=mini] {
  display: none;
}

:host(:not([open])[position=bottom]) [part=base] {
  -webkit-transform: translateY(60vh);
      -ms-transform: translateY(60vh);
          transform: translateY(60vh);
}

:host([open][position=end]) [part=base] {
  -webkit-border-end: none;
          border-inline-end: none;
  border-inline-start-style: solid;
  border-inline-start-width: 0.0625rem;
}

:host([open]) [part=mini] {
  display: none;
}

:host([open]:not([position=relative])) [part=base] {
  border-inline-end-style: solid;
  border-inline-end-width: 0.0625rem;
}

:host(:not([open])) [part=base] {
  -webkit-transform: translateX(calc(100% * var(--dir)));
      -ms-transform: translateX(calc(100% * var(--dir)));
          transform: translateX(calc(100% * var(--dir)));
}
:host(:not([open])) [part=overlay] {
  pointer-events: none;
  opacity: 0;
}`;
