// igniteui-webcomponents/components/checkbox/themes/light/checkbox.bootstrap.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium: min(
      clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
      clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
  );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

:host([hidden]),
[hidden] {
  display: none !important;
}
:host {
  --tick-color: var(--ig-checkbox-tick-color, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --tick-color-hover: var(--ig-checkbox-tick-color-hover, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --label-color: var(--ig-checkbox-label-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --empty-color: var(--ig-checkbox-empty-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --fill-color: var(--ig-checkbox-fill-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --fill-color-hover: var(--ig-checkbox-fill-color-hover, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --disabled-color: var(--ig-checkbox-disabled-color, hsla(var(--ig-primary-200), var(--ig-primary-a)));
  --disabled-indeterminate-color: var(--ig-checkbox-disabled-indeterminate-color, hsla(var(--ig-secondary-100), var(--ig-secondary-a)));
  --disabled-color-label: var(--ig-checkbox-disabled-color-label, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --error-color: var(--ig-checkbox-error-color, hsla(var(--ig-error-500), var(--ig-error-a)));
  --error-color-hover: var(--ig-checkbox-error-color-hover, hsla(var(--ig-error-100), var(--ig-error-a)));
  --border-radius: var(--ig-checkbox-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.4) * 0.625rem), 0.625rem));
  --border-radius-ripple: var(--ig-checkbox-border-radius-ripple, clamp(0rem, calc(var(--ig-radius-factor, 0.1666666667) * 1.5rem), 1.5rem));
  --focus-outline-color: var(--ig-checkbox-focus-outline-color, hsla(var(--ig-primary-200), var(--ig-primary-a)));
}

[part~=control] {
  --size: 1rem;
  border-radius: var(--border-radius-ripple);
  width: var(--size);
  height: var(--size);
  min-width: var(--size);
  -webkit-transition: background-color 0.15s ease-in-out;
  transition: background-color 0.15s ease-in-out;
  -webkit-transform: translate3d(0, 0, 0);
          transform: translate3d(0, 0, 0);
}
[part~=control]::after {
  border-radius: var(--border-radius);
  width: inherit;
  height: inherit;
  min-width: inherit;
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--empty-color);
          box-shadow: inset 0 0 0 0.0625rem var(--empty-color);
  -webkit-transition: none;
  transition: none;
}

[part~=indicator] {
  stroke-width: 3;
  -webkit-transform: scale(0.7);
      -ms-transform: scale(0.7);
          transform: scale(0.7);
  -webkit-transition: none;
  transition: none;
  stroke: white;
}

[part~=label] {
  font-family: var(--ig-body-2-font-family, var(--ig-font-family));
  font-size: var(--ig-body-2-font-size);
  font-weight: var(--ig-body-2-font-weight);
  font-style: var(--ig-body-2-font-style);
  line-height: var(--ig-body-2-line-height);
  letter-spacing: var(--ig-body-2-letter-spacing);
  text-transform: var(--ig-body-2-text-transform);
  margin-top: var(--ig-body-2-margin-top);
  margin-bottom: var(--ig-body-2-margin-bottom);
  -webkit-transition: none;
  transition: none;
  color: var(--label-color);
}

[part="control checked"]::after {
  background: var(--fill-color);
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--fill-color);
          box-shadow: inset 0 0 0 0.0625rem var(--fill-color);
}

:host([indeterminate]) [part~=control]::after {
  background: var(--fill-color);
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--fill-color);
          box-shadow: inset 0 0 0 0.0625rem var(--fill-color);
}
:host([indeterminate]) [part~=indicator] {
  stroke-dashoffset: 41;
  opacity: 1;
  -webkit-transform: rotate(45deg) translateX(-0.075rem) scale(0.7);
      -ms-transform: rotate(45deg) translateX(-0.075rem) scale(0.7);
          transform: rotate(45deg) translateX(-0.075rem) scale(0.7);
}

:host(:not([disabled])[invalid]) [part~=label],
:host(:not(:disabled)[invalid]) [part~=label] {
  color: var(--error-color);
}
:host(:not([disabled])[invalid]) [part~=control]::after,
:host(:not(:disabled)[invalid]) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--error-color);
          box-shadow: inset 0 0 0 0.0625rem var(--error-color);
  border-color: var(--error-color);
}
:host(:not([disabled])[invalid]) [part="control checked"]::after,
:host(:not(:disabled)[invalid]) [part="control checked"]::after {
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--error-color);
          box-shadow: inset 0 0 0 0.0625rem var(--error-color);
  background: var(--error-color);
}
:host(:not([disabled])[invalid]):focus-within [part~=control],
:host(:not([disabled])[invalid]):focus-within [part="control checked"],
:host(:not(:disabled)[invalid]):focus-within [part~=control],
:host(:not(:disabled)[invalid]):focus-within [part="control checked"] {
  -webkit-box-shadow: 0 0 0 0.25rem var(--error-color);
          box-shadow: 0 0 0 0.25rem var(--error-color);
}

:host(:not([disabled])[invalid][indeterminate]) [part~=control]::after,
:host(:not(:disabled)[invalid][indeterminate]) [part~=control]::after {
  background: var(--error-color);
}

:host(:focus-within) [part~=control],
[part~=focused] [part~=control] {
  -webkit-box-shadow: 0 0 0 0.25rem var(--focus-outline-color);
          box-shadow: 0 0 0 0.25rem var(--focus-outline-color);
}
:host(:focus-within) [part~=control]::after,
[part~=focused] [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--fill-color);
          box-shadow: inset 0 0 0 0.0625rem var(--fill-color);
}

:host(:focus-within:not([disabled])[invalid]) [part~=control],
:host(:focus-within:not(:disabled)[invalid]) [part~=control],
:host(:not([disabled])[invalid]) [part~=focused] [part~=control],
:host(:not(:disabled)[invalid]) [part~=focused] [part~=control] {
  -webkit-box-shadow: 0 0 0 0.25rem var(--error-color-hover);
          box-shadow: 0 0 0 0.25rem var(--error-color-hover);
}

:host([disabled]) [part~=label],
:host(:disabled) [part~=label] {
  color: var(--disabled-color-label);
}
:host([disabled]) [part~=control]::after,
:host(:disabled) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--disabled-color-label);
          box-shadow: inset 0 0 0 0.0625rem var(--disabled-color-label);
  border-color: var(--disabled-color-label);
}
:host([disabled]) [part="control checked"]::after,
:host(:disabled) [part="control checked"]::after {
  background: var(--disabled-color);
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--disabled-color);
          box-shadow: inset 0 0 0 0.0625rem var(--disabled-color);
}

:host([disabled][indeterminate]) [part=control]::after,
:host(:disabled[indeterminate]) [part=control]::after {
  background: var(--disabled-color);
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--disabled-color);
          box-shadow: inset 0 0 0 0.0625rem var(--disabled-color);
}`;
