// igniteui-webcomponents/components/checkbox/themes/light/checkbox.material.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium: min(
      clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
      clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
  );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

:host([hidden]),
[hidden] {
  display: none !important;
}
:host {
  --tick-color: var(--ig-checkbox-tick-color, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --tick-color-hover: var(--ig-checkbox-tick-color-hover, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --label-color: var(--ig-checkbox-label-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --empty-color: var(--ig-checkbox-empty-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --fill-color: var(--ig-checkbox-fill-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --fill-color-hover: var(--ig-checkbox-fill-color-hover, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --disabled-color: var(--ig-checkbox-disabled-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --disabled-indeterminate-color: var(--ig-checkbox-disabled-indeterminate-color, hsla(var(--ig-secondary-100), var(--ig-secondary-a)));
  --disabled-color-label: var(--ig-checkbox-disabled-color-label, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --error-color: var(--ig-checkbox-error-color, hsla(var(--ig-error-500), var(--ig-error-a)));
  --error-color-hover: var(--ig-checkbox-error-color-hover, hsla(var(--ig-error-500), var(--ig-error-a)));
  --border-radius: var(--ig-checkbox-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.2) * 0.625rem), 0.625rem));
  --border-radius-ripple: var(--ig-checkbox-border-radius-ripple, clamp(0rem, calc(var(--ig-radius-factor, 1) * 1.5rem), 1.5rem));
}

[part~=label] {
  font-family: var(--ig-subtitle-1-font-family, var(--ig-font-family));
  font-size: var(--ig-subtitle-1-font-size);
  font-weight: var(--ig-subtitle-1-font-weight);
  font-style: var(--ig-subtitle-1-font-style);
  line-height: var(--ig-subtitle-1-line-height);
  letter-spacing: var(--ig-subtitle-1-letter-spacing);
  text-transform: var(--ig-subtitle-1-text-transform);
  margin-top: var(--ig-subtitle-1-margin-top);
  margin-bottom: var(--ig-subtitle-1-margin-bottom);
  color: var(--label-color);
}

[part~=indicator] {
  stroke: var(--tick-color);
}

[part~=control] {
  --size: 1.25rem;
  width: calc(var(--size) * 2);
  height: calc(var(--size) * 2);
  min-width: calc(var(--size) * 2);
  -webkit-transition: background 0.15s ease-in;
  transition: background 0.15s ease-in;
}
[part~=control]::after {
  border-radius: var(--border-radius);
  border: 0.125rem solid var(--empty-color);
}
[part~=control]::before {
  content: "";
  position: absolute;
  inset-block-start: 0;
  inset-inline-start: 0;
  width: 100%;
  height: 100%;
  border-radius: var(--border-radius-ripple);
}

[part="control checked"]::after {
  border-color: var(--fill-color);
  background: var(--fill-color);
}

[part~=base]:hover [part~=control]::before {
  background: var(--empty-color);
  opacity: 0.06;
}

[part="base checked"]:hover [part~=control]::before {
  background: var(--fill-color);
  opacity: 0.06;
}

[part=base]:focus-within:hover [part~=control]::before,
[part=base]:focus-within [part~=control]::before,
[part="base focused"]:hover [part~=control]::before,
[part="base focused"] [part~=control]::before {
  background: var(--empty-color);
  opacity: 0.12;
}

[part="base checked"]:focus-within:hover [part~=control]::before,
[part="base checked"]:focus-within [part~=control]::before,
[part="base checked focused"]:hover [part~=control]::before,
[part="base checked focused"] [part~=control]::before {
  background: var(--fill-color);
  opacity: 0.12;
}

:host([indeterminate]) [part~=control]::after {
  border-color: var(--fill-color);
  background: var(--fill-color);
}
:host([indeterminate]) [part=base]:hover [part~=control]::before {
  background: var(--fill-color);
  opacity: 0.12;
}
:host([indeterminate]) [part=base]:focus-within [part~=control]::before,
:host([indeterminate]) [part="base focused"] [part~=control]::before {
  background: var(--fill-color);
  opacity: 0.12;
}

:host(:not([disabled])[invalid]) [part~=label],
:host(:not(:disabled)[invalid]) [part~=label] {
  color: var(--error-color);
}
:host(:not([disabled])[invalid]) [part~=control]::after,
:host(:not(:disabled)[invalid]) [part~=control]::after {
  border-color: var(--error-color);
}
:host(:not([disabled])[invalid]) [part="control checked"]::after,
:host(:not(:disabled)[invalid]) [part="control checked"]::after {
  background: var(--error-color);
}
:host(:not([disabled])[invalid]) [part~=base]:hover [part~=control]::before,
:host(:not(:disabled)[invalid]) [part~=base]:hover [part~=control]::before {
  background: var(--error-color-hover);
}
:host(:not([disabled])[invalid]) [part="base focused"] [part~=control]::before,
:host(:not(:disabled)[invalid]) [part="base focused"] [part~=control]::before {
  background: var(--error-color-hover);
}

:host(:not([disabled])[invalid][indeterminate]) [part~=control]::after,
:host(:not(:disabled)[invalid][indeterminate]) [part~=control]::after {
  border-color: var(--error-color);
  background: var(--error-color);
}
:host(:not([disabled])[invalid][indeterminate]) [part~=base]:hover [part~=control]::before,
:host(:not(:disabled)[invalid][indeterminate]) [part~=base]:hover [part~=control]::before {
  background: var(--error-color-hover);
}
:host(:not([disabled])[invalid][indeterminate]) [part="base focused"] [part~=control]::before,
:host(:not(:disabled)[invalid][indeterminate]) [part="base focused"] [part~=control]::before {
  background: var(--error-color-hover);
}

:host([disabled]) [part~=label],
:host(:disabled) [part~=label] {
  color: var(--disabled-color-label);
}
:host([disabled]) [part~=control]::after,
:host(:disabled) [part~=control]::after {
  border-color: var(--disabled-color);
}
:host([disabled]) [part="control checked"]::after,
:host(:disabled) [part="control checked"]::after {
  background: var(--disabled-color);
}

:host([disabled][indeterminate]) [part~=control]::after,
:host(:disabled[indeterminate]) [part~=control]::after {
  background: var(--disabled-indeterminate-color);
  border-color: var(--disabled-indeterminate-color);
}`;
