// igniteui-webcomponents/components/checkbox/themes/light/checkbox.fluent.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium: min(
      clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
      clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
  );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

:host([hidden]),
[hidden] {
  display: none !important;
}
:host {
  --tick-color: var(--ig-checkbox-tick-color, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --tick-color-hover: var(--ig-checkbox-tick-color-hover, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --label-color: var(--ig-checkbox-label-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --empty-color: var(--ig-checkbox-empty-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --fill-color: var(--ig-checkbox-fill-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --fill-color-hover: var(--ig-checkbox-fill-color-hover, hsla(var(--ig-primary-900), var(--ig-primary-a)));
  --disabled-color: var(--ig-checkbox-disabled-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --disabled-indeterminate-color: var(--ig-checkbox-disabled-indeterminate-color, hsla(var(--ig-secondary-100), var(--ig-secondary-a)));
  --disabled-color-label: var(--ig-checkbox-disabled-color-label, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --error-color: var(--ig-checkbox-error-color, hsla(var(--ig-error-500), var(--ig-error-a)));
  --error-color-hover: var(--ig-checkbox-error-color-hover, hsla(var(--ig-error-900), var(--ig-error-a)));
  --border-radius: var(--ig-checkbox-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.2) * 0.625rem), 0.625rem));
  --border-radius-ripple: var(--ig-checkbox-border-radius-ripple, clamp(0rem, calc(var(--ig-radius-factor, 0.0833333333) * 1.5rem), 1.5rem));
  --focus-outline-color: var(--ig-checkbox-focus-outline-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
}

[part~=control] {
  width: var(--size, 1.25rem);
  height: var(--size, 1.25rem);
  min-width: var(--size, 1.25rem);
  -webkit-transform: translate3d(0, 0, 0);
          transform: translate3d(0, 0, 0);
}
[part~=control]::after {
  border-radius: var(--border-radius);
  background: transparent;
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--empty-color);
          box-shadow: inset 0 0 0 0.0625rem var(--empty-color);
}

[part="control checked"]::after {
  background: var(--fill-color);
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--fill-color);
          box-shadow: inset 0 0 0 0.0625rem var(--fill-color);
}

[part~=label] {
  font-family: var(--ig-body-2-font-family, var(--ig-font-family));
  font-size: var(--ig-body-2-font-size);
  font-weight: var(--ig-body-2-font-weight);
  font-style: var(--ig-body-2-font-style);
  line-height: var(--ig-body-2-line-height);
  letter-spacing: var(--ig-body-2-letter-spacing);
  text-transform: var(--ig-body-2-text-transform);
  margin-top: var(--ig-body-2-margin-top);
  margin-bottom: var(--ig-body-2-margin-bottom);
  color: var(--label-color);
}

[part~=indicator] {
  stroke: var(--tick-color);
  stroke-width: 1.5;
  -webkit-transform: scale(0.8);
      -ms-transform: scale(0.8);
          transform: scale(0.8);
}

:host(:hover) [part~=indicator] {
  opacity: 1;
  stroke: var(--tick-color-hover);
  stroke-dashoffset: 0;
  -webkit-transition: opacity 0.1s ease-out;
  transition: opacity 0.1s ease-out;
}
:host(:hover) [part="indicator checked"] {
  stroke: var(--tick-color);
}

:host(:hover) [part="control checked"]::after,
:host(:focus-within) [part="control checked"]::after {
  background: var(--fill-color-hover);
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--fill-color-hover);
          box-shadow: inset 0 0 0 0.0625rem var(--fill-color-hover);
}

:host([indeterminate]) [part~=indicator] {
  display: none;
}
:host([indeterminate]) [part~=control]::before {
  border-radius: var(--border-radius);
  display: block;
  width: inherit;
  height: inherit;
  content: "";
  -webkit-transform: scale(0.5);
      -ms-transform: scale(0.5);
          transform: scale(0.5);
  -webkit-transition-duration: 0.1s;
          transition-duration: 0.1s;
  background: var(--fill-color);
  z-index: 1;
}
:host([indeterminate]) [part="control checked"]::after {
  background: transparent;
}
:host([indeterminate]) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--fill-color);
          box-shadow: inset 0 0 0 0.0625rem var(--fill-color);
}

:host([indeterminate]:hover) [part~=control]::before,
:host([indeterminate]:focus-within) [part~=control]::before {
  background: var(--fill-color-hover);
}
:host([indeterminate]:hover) [part~=control]::after,
:host([indeterminate]:focus-within) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--fill-color-hover);
          box-shadow: inset 0 0 0 0.0625rem var(--fill-color-hover);
}
:host([indeterminate]:hover) [part~=control] [part="control checked"]::after,
:host([indeterminate]:focus-within) [part~=control] [part="control checked"]::after {
  background: transparent;
}

:host([invalid]:not([disabled])) [part~=label] {
  color: var(--error-color);
}
:host([invalid]:not([disabled])) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--error-color);
          box-shadow: inset 0 0 0 0.0625rem var(--error-color);
}
:host([invalid]:not([disabled])) [part="control checked"]::after {
  background: var(--error-color);
}

:host([invalid]:focus-within) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--error-color-hover);
          box-shadow: inset 0 0 0 0.0625rem var(--error-color-hover);
}
:host([invalid]:focus-within) [part="control checked"]::after {
  background: var(--error-color-hover);
}

:host([invalid]:hover) [part~=indicator] {
  stroke: var(--error-color);
}
:host([invalid]:hover) [part="indicator checked"] {
  stroke: var(--tick-color);
}
:host([invalid]:hover) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--error-color-hover);
          box-shadow: inset 0 0 0 0.0625rem var(--error-color-hover);
}
:host([invalid]:hover) [part="control checked"]::after {
  background: var(--error-color-hover);
}

:host(:not([disabled])[invalid][indeterminate]) [part~=control]::before,
:host(:not(:disabled)[invalid][indeterminate]) [part~=control]::before {
  background: var(--error-color);
}
:host(:not([disabled])[invalid][indeterminate]) [part~=control]::after,
:host(:not(:disabled)[invalid][indeterminate]) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--error-color);
          box-shadow: inset 0 0 0 0.0625rem var(--error-color);
  background: transparent;
}
:host(:not([disabled])[invalid][indeterminate]):hover [part~=control]::before,
:host(:not(:disabled)[invalid][indeterminate]):hover [part~=control]::before {
  background: var(--error-color-hover);
}
:host(:not([disabled])[invalid][indeterminate]):hover [part~=control]::after,
:host(:not(:disabled)[invalid][indeterminate]):hover [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--error-color-hover);
          box-shadow: inset 0 0 0 0.0625rem var(--error-color-hover);
  background: transparent;
}
:host(:not([disabled])[invalid][indeterminate]):focus-within [part~=control]::before,
:host(:not(:disabled)[invalid][indeterminate]):focus-within [part~=control]::before {
  background: var(--error-color-hover);
}
:host(:not([disabled])[invalid][indeterminate]):focus-within [part~=control]::after,
:host(:not(:disabled)[invalid][indeterminate]):focus-within [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--error-color-hover);
          box-shadow: inset 0 0 0 0.0625rem var(--error-color-hover);
  background: transparent;
}

:host([invalid][checked]) [part="control checked"]::after {
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--error-color);
          box-shadow: inset 0 0 0 0.0625rem var(--error-color);
  background: var(--error-color);
}

[part~=focused] {
  position: relative;
}
[part~=focused]::after {
  content: "";
  position: absolute;
  top: -0.125rem;
  left: -0.125rem;
  -webkit-box-shadow: 0 0 0 0.0625rem hsla(var(--ig-gray-700), var(--ig-gray-a));
          box-shadow: 0 0 0 0.0625rem hsla(var(--ig-gray-700), var(--ig-gray-a));
  width: calc(100% + 0.125rem * 2);
  height: calc(100% + 0.125rem * 2);
}

:host([disabled]) [part~=label],
:host(:disabled) [part~=label] {
  color: var(--disabled-color-label);
}
:host([disabled]) [part~=control]::after,
:host(:disabled) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--disabled-color);
          box-shadow: inset 0 0 0 0.0625rem var(--disabled-color);
}
:host([disabled]) [part="control checked"]::after,
:host(:disabled) [part="control checked"]::after {
  background: var(--disabled-color);
}

:host([disabled][indeterminate]) [part~=control]::before,
:host(:disabled[indeterminate]) [part~=control]::before {
  background: var(--disabled-color);
}
:host([disabled][indeterminate]) [part~=control]::after,
:host(:disabled[indeterminate]) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--disabled-color);
          box-shadow: inset 0 0 0 0.0625rem var(--disabled-color);
  background: transparent;
}

:host([disabled][invalid]) [part~=control]::after,
:host(:disabled[invalid]) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--disabled-color);
          box-shadow: inset 0 0 0 0.0625rem var(--disabled-color);
}
:host([disabled][invalid]) [part="control checked"]::after,
:host(:disabled[invalid]) [part="control checked"]::after {
  background: var(--disabled-color);
}

:host([disabled][invalid][indeterminate]) [part~=control]::before,
:host(:disabled[invalid][indeterminate]) [part~=control]::before {
  background: var(--disabled-color);
}
:host([disabled][invalid][indeterminate]) [part~=control]::after,
:host(:disabled[invalid][indeterminate]) [part~=control]::after {
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--disabled-color);
          box-shadow: inset 0 0 0 0.0625rem var(--disabled-color);
  background: transparent;
}`;
