// igniteui-webcomponents/components/checkbox/themes/light/switch.fluent.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium: min(
      clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
      clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
  );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

:host([hidden]),
[hidden] {
  display: none !important;
}
:host {
  --size: 2.5rem;
  --thumb-size: 0.75rem;
  --border-size: 0.125rem;
  --thumb-offset: calc(var(--thumb-size) / 2 - 0.125rem);
  --label-gap: 0.5rem;
  --resting-elevation: var(--ig-switch-resting-elevation, var(--ig-elevation-2));
  --hover-elevation: var(--ig-switch-hover-elevation, var(--ig-elevation-3));
  --disabled-elevation: var(--ig-switch-disabled-elevation, var(--ig-elevation-1));
  --border-color: var(--ig-switch-border-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --border-hover-color: var(--ig-switch-border-hover-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --border-disabled-color: var(--ig-switch-border-disabled-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --border-on-color: var(--ig-switch-border-on-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --border-on-hover-color: var(--ig-switch-border-on-hover-color, hsla(var(--ig-primary-800), var(--ig-primary-a)));
  --thumb-on-color: var(--ig-switch-thumb-on-color, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --track-on-color: var(--ig-switch-track-on-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --thumb-off-color: var(--ig-switch-thumb-off-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --track-disabled-color: var(--ig-switch-track-disabled-color, transparent);
  --track-on-disabled-color: var(--ig-switch-track-on-disabled-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --track-off-color: var(--ig-switch-track-off-color, transparent);
  --thumb-disabled-color: var(--ig-switch-thumb-disabled-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --thumb-on-disabled-color: var(--ig-switch-thumb-on-disabled-color, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --label-color: var(--ig-switch-label-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --label-disabled-color: var(--ig-switch-label-disabled-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --label-invalid-color: var(--ig-switch-label-invalid-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --track-error-color: var(--ig-switch-track-error-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --thumb-on-error-color: var(--ig-switch-thumb-on-error-color, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --error-color: var(--ig-switch-error-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --error-color-hover: var(--ig-switch-error-color-hover, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --border-radius-track: var(--ig-switch-border-radius-track, clamp(0rem, calc(var(--ig-radius-factor, 1) * 0.625rem), 0.625rem));
  --border-radius-thumb: var(--ig-switch-border-radius-thumb, clamp(0rem, calc(var(--ig-radius-factor, 1) * 0.625rem), 0.625rem));
  --border-radius-ripple: var(--ig-switch-border-radius-ripple, clamp(0rem, calc(var(--ig-radius-factor, 1) * 1.5rem), 1.5rem));
  --focus-outline-color: var(--ig-switch-focus-outline-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --track-on-hover-color: var(--ig-switch-track-on-hover-color, hsla(var(--ig-primary-800), var(--ig-primary-a)));
}

[part~=control] {
  border-radius: var(--border-radius-track);
  width: var(--size);
  height: 1.25rem;
  background: var(--track-off-color);
  border: 0.0625rem solid var(--border-color);
}

[part~=control][part~=checked] {
  background: var(--track-on-color);
  border-color: var(--border-on-color);
}

[part~=thumb] {
  border-radius: var(--border-radius-thumb);
  width: var(--thumb-size);
  height: var(--thumb-size);
  min-width: var(--thumb-size);
  -webkit-margin-start: var(--thumb-offset);
          margin-inline-start: var(--thumb-offset);
  background: var(--thumb-off-color);
}

[part~=thumb][part~=checked] {
  -webkit-margin-start: calc(100% - var(--thumb-size) - var(--thumb-offset));
          margin-inline-start: calc(100% - var(--thumb-size) - var(--thumb-offset));
  background: var(--thumb-on-color);
}

[part~=label] {
  font-family: var(--ig-body-2-font-family, var(--ig-font-family));
  font-size: var(--ig-body-2-font-size);
  font-weight: var(--ig-body-2-font-weight);
  font-style: var(--ig-body-2-font-style);
  line-height: var(--ig-body-2-line-height);
  letter-spacing: var(--ig-body-2-letter-spacing);
  text-transform: var(--ig-body-2-text-transform);
  margin-top: var(--ig-body-2-margin-top);
  margin-bottom: var(--ig-body-2-margin-bottom);
  color: var(--label-color);
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

[part~=focused] {
  position: relative;
}
[part~=focused]::after {
  content: "";
  position: absolute;
  top: -0.125rem;
  left: -0.125rem;
  -webkit-box-shadow: 0 0 0 0.0625rem var(--focus-outline-color);
          box-shadow: 0 0 0 0.0625rem var(--focus-outline-color);
  width: calc(100% + 0.125rem * 2);
  height: calc(100% + 0.125rem * 2);
}

:host(:not([disabled]):hover) [part~=control],
:host(:not(:disabled):hover) [part~=control] {
  border-color: var(--border-hover-color);
}
:host(:not([disabled]):hover) [part~=control][part~=checked],
:host(:not(:disabled):hover) [part~=control][part~=checked] {
  background: var(--track-on-hover-color);
  border-color: var(--border-on-hover-color);
}
:host(:not([disabled]):hover) [part~=thumb],
:host(:not(:disabled):hover) [part~=thumb] {
  background: var(--border-hover-color);
}
:host(:not([disabled]):hover) [part~=thumb][part~=checked],
:host(:not(:disabled):hover) [part~=thumb][part~=checked] {
  background: var(--thumb-on-color);
}

:host(:not([disabled])[invalid]) [part~=control][part~=checked],
:host(:not(:disabled)[invalid]) [part~=control][part~=checked] {
  background: var(--track-error-color);
}
:host(:not([disabled])[invalid]) [part~=control],
:host(:not([disabled])[invalid]) [part~=control][part~=checked],
:host(:not(:disabled)[invalid]) [part~=control],
:host(:not(:disabled)[invalid]) [part~=control][part~=checked] {
  border-color: var(--error-color);
}
:host(:not([disabled])[invalid]) [part~=thumb],
:host(:not(:disabled)[invalid]) [part~=thumb] {
  background: var(--error-color);
}
:host(:not([disabled])[invalid]) [part~=control][part~=focused],
:host(:not([disabled])[invalid]) [part~=control][part~=checked][part~=focused],
:host(:not(:disabled)[invalid]) [part~=control][part~=focused],
:host(:not(:disabled)[invalid]) [part~=control][part~=checked][part~=focused] {
  -webkit-box-shadow: 0 0 0 0.125rem var(--error-color-hover);
          box-shadow: 0 0 0 0.125rem var(--error-color-hover);
}
:host(:not([disabled])[invalid]) [part~=thumb][part~=checked],
:host(:not(:disabled)[invalid]) [part~=thumb][part~=checked] {
  background: var(--thumb-on-error-color);
}
:host(:not([disabled])[invalid]) [part~=label],
:host(:not(:disabled)[invalid]) [part~=label] {
  color: var(--label-invalid-color);
}

:host([disabled]) [part~=label],
:host(:disabled) [part~=label] {
  color: var(--label-disabled-color);
}
:host([disabled]) [part~=control],
:host(:disabled) [part~=control] {
  background: var(--track-disabled-color);
  border-color: var(--border-disabled-color);
}
:host([disabled]) [part~=control][part~=checked],
:host(:disabled) [part~=control][part~=checked] {
  background: var(--track-on-disabled-color);
}
:host([disabled]) [part~=thumb],
:host(:disabled) [part~=thumb] {
  background: var(--thumb-disabled-color);
}
:host([disabled]) [part~=thumb][part~=checked],
:host(:disabled) [part~=thumb][part~=checked] {
  background: var(--thumb-on-disabled-color);
}`;
