// igniteui-webcomponents/components/checkbox/themes/light/switch.material.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium: min(
      clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
      clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
  );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

:host([hidden]),
[hidden] {
  display: none !important;
}
:host {
  --size: 2.25rem;
  --thumb-size: 1.25rem;
  --thumb-offset: -0.0625rem;
  --label-gap: 0.5rem;
  --resting-elevation: var(--ig-switch-resting-elevation, var(--ig-elevation-2));
  --hover-elevation: var(--ig-switch-hover-elevation, var(--ig-elevation-3));
  --disabled-elevation: var(--ig-switch-disabled-elevation, var(--ig-elevation-1));
  --border-color: var(--ig-switch-border-color, transparent);
  --border-hover-color: var(--ig-switch-border-hover-color, transparent);
  --border-disabled-color: var(--ig-switch-border-disabled-color, transparent);
  --border-on-color: var(--ig-switch-border-on-color, transparent);
  --border-on-hover-color: var(--ig-switch-border-on-hover-color, transparent);
  --thumb-on-color: var(--ig-switch-thumb-on-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --track-on-color: var(--ig-switch-track-on-color, hsla(var(--ig-secondary-200), var(--ig-secondary-a)));
  --thumb-off-color: var(--ig-switch-thumb-off-color, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --track-disabled-color: var(--ig-switch-track-disabled-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --track-on-disabled-color: var(--ig-switch-track-on-disabled-color, hsla(var(--ig-secondary-50), var(--ig-secondary-a)));
  --track-off-color: var(--ig-switch-track-off-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --thumb-disabled-color: var(--ig-switch-thumb-disabled-color, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --thumb-on-disabled-color: var(--ig-switch-thumb-on-disabled-color, hsla(var(--ig-secondary-100), var(--ig-secondary-a)));
  --label-color: var(--ig-switch-label-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --label-disabled-color: var(--ig-switch-label-disabled-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --label-invalid-color: var(--ig-switch-label-invalid-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --track-error-color: var(--ig-switch-track-error-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --thumb-on-error-color: var(--ig-switch-thumb-on-error-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --error-color: var(--ig-switch-error-color, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --error-color-hover: var(--ig-switch-error-color-hover, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --border-radius-track: var(--ig-switch-border-radius-track, clamp(0rem, calc(var(--ig-radius-factor, 1) * 0.4375rem), 0.4375rem));
  --border-radius-thumb: var(--ig-switch-border-radius-thumb, clamp(0rem, calc(var(--ig-radius-factor, 1) * 0.625rem), 0.625rem));
  --border-radius-ripple: var(--ig-switch-border-radius-ripple, clamp(0rem, calc(var(--ig-radius-factor, 1) * 1.5rem), 1.5rem));
}

[part~=base]:hover [part~=thumb]::after {
  background: var(--track-off-color);
  opacity: 0.06;
}
[part~=base]:hover [part~=thumb][part~=checked]::after {
  background: var(--thumb-on-color);
  opacity: 0.06;
}
[part~=base]:hover [part~=control] {
  border-color: var(--border-hover-color);
}
[part~=base]:hover [part~=control][part~=checked] {
  border-color: var(--border-on-hover-color);
}
[part~=base]:focus-within [part~=thumb]::after {
  background: var(--track-off-color);
  opacity: 0.12;
}
[part~=base]:focus-within [part~=thumb][part~=checked]::after {
  background: var(--thumb-on-color);
  opacity: 0.12;
}

[part~=control] {
  border-radius: var(--border-radius-track);
  width: var(--size);
  height: 0.875rem;
  background: var(--track-off-color);
  border: 0.0625rem solid var(--border-color);
}

[part~=control][part~=checked] {
  background: var(--track-on-color);
  border-color: var(--border-on-color);
}

[part~=thumb] {
  border-radius: var(--border-radius-thumb);
  -webkit-animation-name: thumb-off;
          animation-name: thumb-off;
  width: var(--thumb-size);
  height: var(--thumb-size);
  min-width: var(--thumb-size);
  background: var(--thumb-off-color);
  -webkit-box-shadow: var(--ig-elevation-2);
          box-shadow: var(--ig-elevation-2);
  -webkit-margin-start: var(--thumb-offset);
          margin-inline-start: var(--thumb-offset);
}
[part~=thumb]::after {
  border-radius: var(--border-radius-ripple);
  position: absolute;
  content: "";
  inset-block-start: calc(50% - 1.25rem);
  inset-inline-start: calc(50% - 1.25rem);
  width: 2.5rem;
  height: 2.5rem;
  overflow: hidden;
}
[part~=thumb]:hover {
  -webkit-box-shadow: var(--ig-elevation-3);
          box-shadow: var(--ig-elevation-3);
}

[part~=thumb][part~=checked] {
  background: var(--thumb-on-color);
  -webkit-margin-start: calc(var(--size) - var(--thumb-size) - var(--thumb-offset));
          margin-inline-start: calc(var(--size) - var(--thumb-size) - var(--thumb-offset));
  -webkit-animation-name: thumb-on;
          animation-name: thumb-on;
  -webkit-animation-duration: 0.2s;
          animation-duration: 0.2s;
  -webkit-animation-fill-mode: forwards;
          animation-fill-mode: forwards;
}

[part~=label] {
  font-family: var(--ig-subtitle-1-font-family, var(--ig-font-family));
  font-size: var(--ig-subtitle-1-font-size);
  font-weight: var(--ig-subtitle-1-font-weight);
  font-style: var(--ig-subtitle-1-font-style);
  line-height: var(--ig-subtitle-1-line-height);
  letter-spacing: var(--ig-subtitle-1-letter-spacing);
  text-transform: var(--ig-subtitle-1-text-transform);
  margin-top: var(--ig-subtitle-1-margin-top);
  margin-bottom: var(--ig-subtitle-1-margin-bottom);
  color: var(--label-color);
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

:host([invalid]) [part~=control] {
  border-color: var(--track-error-color);
  background: var(--track-error-color);
}
:host([invalid]) [part~=thumb] {
  background: var(--error-color);
}
:host([invalid]) [part~=thumb][part~=checked] {
  background: var(--thumb-on-error-color);
}
:host([invalid]) [part~=thumb][part~=focused]::after {
  background: var(--error-color-hover);
}
:host([invalid]):hover [part~=thumb]::after {
  background: var(--error-color-hover);
}

:host(:not([disabled])[invalid]) [part~=label],
:host(:not(:disabled)[invalid]) [part~=label] {
  color: var(--label-invalid-color);
}
:host(:not([disabled])[invalid]) [part~=control],
:host(:not(:disabled)[invalid]) [part~=control] {
  background: var(--track-error-color);
}

:host([disabled]) [part~=control],
:host(:disabled) [part~=control] {
  background: var(--track-disabled-color);
  border-color: var(--border-disabled-color);
}
:host([disabled]) [part~=control][part~=checked],
:host(:disabled) [part~=control][part~=checked] {
  background: var(--track-on-disabled-color);
}
:host([disabled]) [part~=thumb],
:host(:disabled) [part~=thumb] {
  -webkit-box-shadow: var(--ig-elevation-1);
          box-shadow: var(--ig-elevation-1);
  background: var(--thumb-disabled-color);
}
:host([disabled]) [part~=thumb][part~=checked],
:host(:disabled) [part~=thumb][part~=checked] {
  background: var(--thumb-on-disabled-color);
}
:host([disabled]) [part~=label],
:host(:disabled) [part~=label] {
  color: var(--label-disabled-color);
}`;
