// igniteui-webcomponents/components/radio/themes/light/radio.indigo.css.js
import { css } from 'lit';
export const styles = css `:host {
  --hover-color: var(--ig-radio-hover-color, transparent);
  --fill-hover-border-color: var(--ig-radio-fill-hover-border-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --label-color: var(--ig-radio-label-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --empty-color: var(--ig-radio-empty-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --fill-color: var(--ig-radio-fill-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --fill-color-hover: var(--ig-radio-fill-color-hover, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --disabled-color: var(--ig-radio-disabled-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --disabled-label-color: var(--ig-radio-disabled-label-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --error-color: var(--ig-radio-error-color, hsla(var(--ig-error-500), var(--ig-error-a)));
  --error-color-hover: var(--ig-radio-error-color-hover, hsla(var(--ig-error-200), var(--ig-error-a)));
  --focus-outline-color: var(--ig-radio-focus-outline-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --focus-outline-color-focused: var(--ig-radio-focus-outline-color-focused, hsla(var(--ig-primary-200), var(--ig-primary-a)));
  font-weight: 600;
}

[part~=base] {
  gap: 0.5rem;
}

[part~=label] {
  color: var(--label-color);
}

[part~=control] {
  border-radius: var(--size);
  width: var(--size, 1.25rem);
  height: var(--size, 1.25rem);
  min-width: var(--size, 1.25rem);
  -webkit-transition: none;
  transition: none;
  background: none;
}
[part~=control]::before, [part~=control]::after {
  border-radius: calc(var(--size) / 2);
}
[part~=control]::after {
  border: 0.125rem solid var(--empty-color);
}

[part="control checked"]::after {
  border-color: var(--fill-color);
}
[part="control checked"]::before {
  background: var(--fill-color-hover);
  border-color: var(--fill-hover-border-color);
}

[part~=focused] [part=control] {
  -webkit-box-shadow: 0 0 0 0.1875rem var(--focus-outline-color);
          box-shadow: 0 0 0 0.1875rem var(--focus-outline-color);
}
[part~=focused] [part="control checked"] {
  -webkit-box-shadow: 0 0 0 0.1875rem var(--focus-outline-color-focused);
          box-shadow: 0 0 0 0.1875rem var(--focus-outline-color-focused);
}

:host(:not([disabled])[invalid]) [part~=label],
:host(:not(:disabled)[invalid]) [part~=label] {
  color: var(--error-color);
}
:host(:not([disabled])[invalid]) [part=control]::after,
:host(:not(:disabled)[invalid]) [part=control]::after {
  border-color: var(--error-color);
}
:host(:not([disabled])[invalid]) [part="control checked"]::after,
:host(:not(:disabled)[invalid]) [part="control checked"]::after {
  border-color: var(--error-color);
}
:host(:not([disabled])[invalid]) [part="control checked"]::before,
:host(:not(:disabled)[invalid]) [part="control checked"]::before {
  background: var(--error-color);
}
:host(:not([disabled])[invalid]) [part~=focused] [part~=control],
:host(:not([disabled])[invalid]) [part~=focused] [part="control checked"],
:host(:not(:disabled)[invalid]) [part~=focused] [part~=control],
:host(:not(:disabled)[invalid]) [part~=focused] [part="control checked"] {
  -webkit-box-shadow: 0 0 0 0.1875rem var(--error-color-hover);
          box-shadow: 0 0 0 0.1875rem var(--error-color-hover);
}

:host([disabled]) [part~=label],
:host(:disabled) [part~=label] {
  color: var(--disabled-label-color);
}
:host([disabled]) [part=control]::after,
:host(:disabled) [part=control]::after {
  border-color: var(--disabled-color);
}
:host([disabled]) [part="control checked"]::after,
:host(:disabled) [part="control checked"]::after {
  border-color: var(--disabled-color);
}
:host([disabled]) [part="control checked"]::before,
:host(:disabled) [part="control checked"]::before {
  background: var(--disabled-color);
}`;
