// igniteui-webcomponents/components/combo/themes/light/combo.indigo.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium: min(
      clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
      clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
  );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

:host([hidden]),
[hidden] {
  display: none !important;
}
:host {
  --search-resting-elevation: var(--ig-input-search-resting-elevation, var(--ig-elevation-1));
  --search-hover-elevation: var(--ig-input-search-hover-elevation, var(--ig-elevation-1));
  --search-disabled-elevation: var(--ig-input-search-disabled-elevation, var(--ig-elevation-1));
  --helper-text-color: var(--ig-input-helper-text-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --input-prefix-background: var(--ig-input-input-prefix-background, transparent);
  --input-prefix-color: var(--ig-input-input-prefix-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --input-prefix-background--filled: var(--ig-input-input-prefix-background--filled, inherit);
  --input-prefix-color--filled: var(--ig-input-input-prefix-color--filled, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --input-prefix-background--focused: var(--ig-input-input-prefix-background--focused, inherit);
  --input-prefix-color--focused: var(--ig-input-input-prefix-color--focused, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --input-suffix-background: var(--ig-input-input-suffix-background, transparent);
  --input-suffix-color: var(--ig-input-input-suffix-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --input-suffix-background--filled: var(--ig-input-input-suffix-background--filled, inherit);
  --input-suffix-color--filled: var(--ig-input-input-suffix-color--filled, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --input-suffix-background--focused: var(--ig-input-input-suffix-background--focused, inherit);
  --input-suffix-color--focused: var(--ig-input-input-suffix-color--focused, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --idle-text-color: var(--ig-input-idle-text-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --filled-text-color: var(--ig-input-filled-text-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --focused-text-color: var(--ig-input-focused-text-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --disabled-text-color: var(--ig-input-disabled-text-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --idle-secondary-color: var(--ig-input-idle-secondary-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --filled-secondary-color: var(--ig-input-filled-secondary-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --focused-secondary-color: var(--ig-input-focused-secondary-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --idle-bottom-line-color: var(--ig-input-idle-bottom-line-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --hover-bottom-line-color: var(--ig-input-hover-bottom-line-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --focused-bottom-line-color: var(--ig-input-focused-bottom-line-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --interim-bottom-line-color: var(--ig-input-interim-bottom-line-color, hsla(var(--ig-primary-500), 0.12));
  --disabled-bottom-line-color: var(--ig-input-disabled-bottom-line-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --disabled-border-color: var(--ig-input-disabled-border-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --border-color: var(--ig-input-border-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --hover-border-color: var(--ig-input-hover-border-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --focused-border-color: var(--ig-input-focused-border-color, hsla(var(--ig-primary-400), var(--ig-primary-a)));
  --border-background: var(--ig-input-border-background, transparent);
  --border-disabled-background: var(--ig-input-border-disabled-background, transparent);
  --search-background: var(--ig-input-search-background, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --search-disabled-background: var(--ig-input-search-disabled-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --success-secondary-color: var(--ig-input-success-secondary-color, hsla(var(--ig-success-500), var(--ig-success-a)));
  --warning-secondary-color: var(--ig-input-warning-secondary-color, hsla(var(--ig-warn-500), var(--ig-warn-a)));
  --error-secondary-color: var(--ig-input-error-secondary-color, hsla(var(--ig-error-500), var(--ig-error-a)));
  --placeholder-color: var(--ig-input-placeholder-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --disabled-placeholder-color: var(--ig-input-disabled-placeholder-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --box-background: var(--ig-input-box-background, transparent);
  --box-background-hover: var(--ig-input-box-background-hover, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --box-background-focus: var(--ig-input-box-background-focus, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --box-disabled-background: var(--ig-input-box-disabled-background, transparent);
  --box-border-radius: var(--ig-input-box-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0) * 1.25rem), 1.25rem));
  --border-border-radius: var(--ig-input-border-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0) * 1.25rem), 1.25rem));
  --search-border-radius: var(--ig-input-search-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.1) * 1.25rem), 1.25rem));
  --success-shadow-color: var(--ig-input-success-shadow-color, hsla(var(--ig-success-500), 0.38));
  --error-shadow-color: var(--ig-input-error-shadow-color, hsla(var(--ig-error-500), 0.38));
  --size: var(--ig-input-size, max(var(--is-large, 1) * max(3.5rem, -1 * 3.5rem), var(--is-medium, 1) * max(3rem, -1 * 3rem), var(--is-small, 1) * max(2.5rem, -1 * 2.5rem)));
  --toggle-button-background-focus--border: var(--ig-combo-toggle-button-background-focus--border, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --case-icon-color: var(--ig-combo-case-icon-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --case-icon-color--active: var(--ig-combo-case-icon-color--active, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --empty-list-background: var(--ig-combo-empty-list-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --search-separator-border-color: var(--ig-combo-search-separator-border-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --empty-list-placeholder-color: var(--ig-combo-empty-list-placeholder-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --toggle-button-background: var(--ig-combo-toggle-button-background, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --toggle-button-background-focus: var(--ig-combo-toggle-button-background-focus, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --toggle-button-foreground: var(--ig-combo-toggle-button-foreground, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --toggle-button-foreground-focus: var(--ig-combo-toggle-button-foreground-focus, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --toggle-button-foreground-filled: var(--ig-combo-toggle-button-foreground-filled, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --clear-button-background: var(--ig-combo-clear-button-background, transparent);
  --clear-button-background-focus: var(--ig-combo-clear-button-background-focus, transparent);
  --clear-button-foreground: var(--ig-combo-clear-button-foreground, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --clear-button-foreground-focus: var(--ig-combo-clear-button-foreground-focus, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --toggle-button-background-disabled: var(--ig-combo-toggle-button-background-disabled, transparent);
  --toggle-button-foreground-disabled: var(--ig-combo-toggle-button-foreground-disabled, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --elevation: var(--ig-drop-down-elevation, var(--ig-elevation-3));
  --background-color: var(--ig-drop-down-background-color, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --header-text-color: var(--ig-drop-down-header-text-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --item-text-color: var(--ig-drop-down-item-text-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --hover-item-text-color: var(--ig-drop-down-hover-item-text-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --hover-item-background: var(--ig-drop-down-hover-item-background, hsla(var(--ig-primary-500), 0.15));
  --focused-item-background: var(--ig-drop-down-focused-item-background, hsla(var(--ig-primary-500), 0.15));
  --focused-item-text-color: var(--ig-drop-down-focused-item-text-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --selected-item-background: var(--ig-drop-down-selected-item-background, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --selected-item-text-color: var(--ig-drop-down-selected-item-text-color, var(--ig-primary-500-contrast));
  --selected-hover-item-background: var(--ig-drop-down-selected-hover-item-background, hsla(var(--ig-primary-400), var(--ig-primary-a)));
  --selected-hover-item-text-color: var(--ig-drop-down-selected-hover-item-text-color, var(--ig-primary-400-contrast));
  --selected-focus-item-background: var(--ig-drop-down-selected-focus-item-background, hsla(var(--ig-primary-400), var(--ig-primary-a)));
  --selected-focus-item-text-color: var(--ig-drop-down-selected-focus-item-text-color, var(--ig-primary-400-contrast));
  --disabled-item-background: var(--ig-drop-down-disabled-item-background, hsla(var(--ig-gray-50), 0));
  --disabled-item-text-color: var(--ig-drop-down-disabled-item-text-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --border-color: var(--ig-drop-down-border-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --border-width: var(--ig-drop-down-border-width, 0);
  --border-radius: var(--ig-drop-down-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.1) * 1.25rem), 1.25rem));
  --item-border-radius: var(--ig-drop-down-item-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.1) * 1.25rem), 1.25rem));
  --size: var(--ig-drop-down-size, max(var(--is-large, 1) * max(2.5rem, -1 * 2.5rem), var(--is-medium, 1) * max(2rem, -1 * 2rem), var(--is-small, 1) * max(1.75rem, -1 * 1.75rem)));
}
:host igc-input::part(container) {
  padding-inline: 0;
}
:host [part=helper-text] {
  margin-top: 0.375rem;
  padding-inline: max(var(--is-large, 1) * max(1rem, -1 * 1rem) * var(--ig-spacing-inline-large, var(--ig-spacing-inline, --ig-spacing)), var(--is-medium, 1) * max(0.75rem, -1 * 0.75rem) * var(--ig-spacing-inline-medium, var(--ig-spacing-inline, --ig-spacing)), var(--is-small, 1) * max(0.5rem, -1 * 0.5rem) * var(--ig-spacing-inline-small, var(--ig-spacing-inline, --ig-spacing)));
}
:host ::slotted([slot=helper-text]) {
  color: var(--helper-text-color);
}

[part=case-icon] {
  color: hsla(var(--ig-gray-600), var(--ig-gray-a));
}

[part="case-icon active"] {
  color: hsla(var(--ig-gray-900), var(--ig-gray-a));
}

[part=empty] {
  color: var(--empty-list-placeholder-color);
}

:host(:not([single-select])) > igc-input::part(input)::-moz-selection {
  background: transparent;
}

:host(:not([single-select])) > igc-input::part(input)::selection {
  background: transparent;
}

[part=list-wrapper] {
  border-radius: var(--border-radius);
}
[part=list-wrapper] igc-input::part(container) {
  background: transparent;
  border-inline-end-color: transparent;
}
[part=list-wrapper] igc-input::part(input) {
  border-inline-end-color: transparent;
}
[part=list-wrapper] igc-input::part(input)::-moz-selection {
  background: hsla(var(--ig-primary-100), var(--ig-primary-a));
}
[part=list-wrapper] igc-input::part(input)::selection {
  background: hsla(var(--ig-primary-100), var(--ig-primary-a));
}
[part=list-wrapper] [part="case-icon active"] {
  color: hsla(var(--ig-primary-500), var(--ig-primary-a));
}

[part~=toggle-icon] {
  background: var(--toggle-button-background);
  color: var(--toggle-button-foreground);
}
[part~=toggle-icon] igc-icon {
  color: inherit;
}

:host(:focus-within) igc-input::part(label) {
  color: var(--focused-secondary-color);
}
:host(:focus-within) [part~=toggle-icon] {
  background: var(--toggle-button-background-focus--border);
  color: var(--toggle-button-foreground-focus);
}
:host(:focus-within) [part~=toggle-icon] igc-icon {
  color: inherit;
}
:host(:focus-within) [part~=clear-icon] {
  background: var(--clear-button-background-focus);
  color: var(--clear-button-foreground-focus);
}
:host(:focus-within) [part~=clear-icon] igc-icon {
  color: inherit;
}

:host([invalid]) ::slotted([slot=helper-text]) {
  color: var(--error-secondary-color);
}

:host(:disabled) ::slotted([slot=prefix]),
:host(:disabled) ::slotted([slot=suffix]),
:host(:disabled) ::slotted([slot=helper-text]),
:host([disabled]) ::slotted([slot=prefix]),
:host([disabled]) ::slotted([slot=suffix]),
:host([disabled]) ::slotted([slot=helper-text]) {
  color: var(--disabled-text-color);
}
:host(:disabled) igc-input::part(input)::-moz-selection, :host([disabled]) igc-input::part(input)::-moz-selection {
  background: transparent;
}
:host(:disabled) igc-input::part(input)::selection,
:host([disabled]) igc-input::part(input)::selection {
  background: transparent;
}
:host(:disabled) [part~=clear-icon],
:host([disabled]) [part~=clear-icon] {
  background: inherit;
  color: var(--disabled-text-color);
}
:host(:disabled) [part~=clear-icon] igc-icon,
:host([disabled]) [part~=clear-icon] igc-icon {
  color: inherit;
}
:host(:disabled) [part~=toggle-icon],
:host([disabled]) [part~=toggle-icon] {
  background: var(--toggle-button-background-disabled);
  color: var(--toggle-button-foreground-disabled);
}
:host(:disabled) [part~=toggle-icon] igc-icon,
:host([disabled]) [part~=toggle-icon] igc-icon {
  color: inherit;
}`;
