// igniteui-webcomponents/components/button/button-base.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { html, LitElement } from 'lit';
import { property, query } from 'lit/decorators.js';
import { classMap } from 'lit/directives/class-map.js';
import { ifDefined } from 'lit/directives/if-defined.js';
import { EventEmitterMixin } from '../common//mixins/event-emitter.js';
import { alternateName } from '../common/decorators/alternateName.js';
import { blazorDeepImport } from '../common/decorators/blazorDeepImport.js';
import { SizableMixin } from '../common/mixins/sizable.js';
let IgcButtonBaseComponent = class IgcButtonBaseComponent extends SizableMixin(EventEmitterMixin(LitElement)) {
    get disabled() {
        return this._disabled;
    }
    set disabled(value) {
        const prev = this._disabled;
        this._disabled = value;
        this.toggleAttribute('disabled', Boolean(this._disabled));
        this.requestUpdate('disabled', prev);
    }
    get form() {
        return this.__internals.form;
    }
    constructor() {
        super();
        this._disabled = false;
        this.__internals = this.attachInternals();
    }
    focus(options) {
        this.nativeElement.focus(options);
    }
    click() {
        this.nativeElement.click();
    }
    blur() {
        this.nativeElement.blur();
    }
    handleFocus() {
        this.emitEvent('igcFocus');
    }
    handleBlur() {
        this.emitEvent('igcBlur');
    }
    get classes() {
        return {};
    }
    handleClick() {
        var _a, _b;
        switch (this.type) {
            case 'submit':
                return (_a = this.form) === null || _a === void 0 ? void 0 : _a.requestSubmit();
            case 'reset':
                return (_b = this.form) === null || _b === void 0 ? void 0 : _b.reset();
            default:
                return;
        }
    }
    formDisabledCallback(state) {
        this._disabled = state;
        this.requestUpdate();
    }
    renderButton() {
        return html `
      <button
        part="base"
        .ariaLabel=${this.ariaLabel}
        .disabled=${this.disabled}
        class=${classMap(this.classes)}
        type=${ifDefined(this.type)}
        @click=${this.handleClick}
        @focus=${this.handleFocus}
        @blur=${this.handleBlur}
      >
        ${this.renderContent()}
      </button>
    `;
    }
    renderLinkButton() {
        return html `
      <a
        part="base"
        role="button"
        href=${ifDefined(this.href)}
        target=${ifDefined(this.target)}
        download=${ifDefined(this.download)}
        rel=${ifDefined(this.rel)}
        aria-disabled=${this.disabled ? 'true' : 'false'}
        .ariaLabel=${this.ariaLabel}
        class=${classMap(this.classes)}
        @focus=${!this.disabled && this.handleFocus}
        @blur=${!this.disabled && this.handleBlur}
      >
        ${this.renderContent()}
      </a>
    `;
    }
    render() {
        const link = this.href !== undefined;
        return link ? this.renderLinkButton() : this.renderButton();
    }
};
IgcButtonBaseComponent.formAssociated = true;
IgcButtonBaseComponent.shadowRootOptions = {
    ...LitElement.shadowRootOptions,
    delegatesFocus: true,
};
__decorate([
    query('[part="base"]', true)
], IgcButtonBaseComponent.prototype, "nativeElement", void 0);
__decorate([
    alternateName('displayType'),
    property()
], IgcButtonBaseComponent.prototype, "type", void 0);
__decorate([
    property()
], IgcButtonBaseComponent.prototype, "href", void 0);
__decorate([
    property()
], IgcButtonBaseComponent.prototype, "download", void 0);
__decorate([
    property()
], IgcButtonBaseComponent.prototype, "target", void 0);
__decorate([
    property()
], IgcButtonBaseComponent.prototype, "rel", void 0);
__decorate([
    property({ type: Boolean, reflect: true })
], IgcButtonBaseComponent.prototype, "disabled", null);
__decorate([
    alternateName('focusComponent')
], IgcButtonBaseComponent.prototype, "focus", null);
__decorate([
    alternateName('blurComponent')
], IgcButtonBaseComponent.prototype, "blur", null);
IgcButtonBaseComponent = __decorate([
    blazorDeepImport
], IgcButtonBaseComponent);
export { IgcButtonBaseComponent };
