// igniteui-webcomponents/components/slider/themes/light/slider.fluent.css.js
import { css } from 'lit';
export const styles = css `:host {
  --track-color: var(--ig-slider-track-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --track-step-color: var(--ig-slider-track-step-color, white);
  --track-step-size: var(--ig-slider-track-step-size, 0.1875rem);
  --track-hover-color: var(--ig-slider-track-hover-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --thumb-color: var(--ig-slider-thumb-color, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --thumb-focus-color: var(--ig-slider-thumb-focus-color, hsla(var(--ig-primary-700), var(--ig-primary-a)));
  --thumb-border-color: var(--ig-slider-thumb-border-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --thumb-border-focus-color: var(--ig-slider-thumb-border-focus-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --thumb-disabled-border-color: var(--ig-slider-thumb-disabled-border-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --disabled-thumb-color: var(--ig-slider-disabled-thumb-color, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --label-background-color: var(--ig-slider-label-background-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --label-text-color: var(--ig-slider-label-text-color, var(--ig-gray-700-contrast));
  --base-track-color: var(--ig-slider-base-track-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --base-track-hover-color: var(--ig-slider-base-track-hover-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --disabled-base-track-color: var(--ig-slider-disabled-base-track-color, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --disabled-fill-track-color: var(--ig-slider-disabled-fill-track-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --tick-color: var(--ig-slider-tick-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --tick-label-color: var(--ig-slider-tick-label-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
}

:host(:hover) [part=fill] {
  background: var(--track-hover-color);
}
:host(:hover) [part~=thumb] {
  border-color: var(--thumb-focus-color);
}

[part~=focused]::after {
  position: absolute;
  content: "";
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%, -50%);
      -ms-transform: translate(-50%, -50%);
          transform: translate(-50%, -50%);
  -webkit-box-shadow: 0 0 0 0.0625rem var(--thumb-border-focus-color);
          box-shadow: 0 0 0 0.0625rem var(--thumb-border-focus-color);
  width: calc(1rem + 0.125rem * 2);
  height: calc(1rem + 0.125rem * 2);
}

[part=track],
[part=fill],
[part=inactive] {
  height: 0.25rem;
}

[part=inactive] {
  background: var(--base-track-color);
  top: initial;
}

[part=fill] {
  background: var(--track-color);
}

[part=steps] {
  color: var(--track-step-color);
}

[part=tick] {
  background: var(--tick-color);
}

[part=tick-label] {
  color: var(--tick-label-color);
}

[part~=thumb] {
  background: var(--thumb-color);
  border: 0.125rem solid var(--thumb-border-color);
  width: 1rem;
  height: 1rem;
  top: calc((1rem - 0.5rem) * -1);
  -webkit-margin-start: calc((1rem - 0.5rem) * -1);
          margin-inline-start: calc((1rem - 0.5rem) * -1);
}

[part=thumb-label-inner] {
  background: var(--label-background-color);
  color: var(--label-text-color);
}
[part=thumb-label-inner]::after {
  border-top: 0.625rem solid var(--label-background-color);
}

:host([disabled]) [part=inactive],
:host(:disabled) [part=inactive] {
  background: var(--disabled-base-track-color);
}
:host([disabled]) [part=fill],
:host(:disabled) [part=fill] {
  background: var(--disabled-fill-track-color);
}
:host([disabled]) [part~=thumb],
:host(:disabled) [part~=thumb] {
  background: var(--disabled-thumb-color);
  border-color: var(--thumb-disabled-border-color);
}
:host([disabled]) [part=tick-label],
:host(:disabled) [part=tick-label] {
  color: var(--disabled-fill-track-color);
}`;
