// igniteui-webcomponents/components/slider/themes/light/slider.indigo.css.js
import { css } from 'lit';
export const styles = css `:host {
  --track-color: var(--ig-slider-track-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --track-step-color: var(--ig-slider-track-step-color, white);
  --track-step-size: var(--ig-slider-track-step-size, 0.1875rem);
  --track-hover-color: var(--ig-slider-track-hover-color, hsla(var(--ig-primary-400), var(--ig-primary-a)));
  --thumb-color: var(--ig-slider-thumb-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --thumb-focus-color: var(--ig-slider-thumb-focus-color, hsla(var(--ig-primary-200), var(--ig-primary-a)));
  --thumb-border-color: var(--ig-slider-thumb-border-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --thumb-border-focus-color: var(--ig-slider-thumb-border-focus-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --thumb-disabled-border-color: var(--ig-slider-thumb-disabled-border-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --disabled-thumb-color: var(--ig-slider-disabled-thumb-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --label-background-color: var(--ig-slider-label-background-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --label-text-color: var(--ig-slider-label-text-color, var(--ig-gray-700-contrast));
  --base-track-color: var(--ig-slider-base-track-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --base-track-hover-color: var(--ig-slider-base-track-hover-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --disabled-base-track-color: var(--ig-slider-disabled-base-track-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --disabled-fill-track-color: var(--ig-slider-disabled-fill-track-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --tick-color: var(--ig-slider-tick-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --tick-label-color: var(--ig-slider-tick-label-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
}

[part=base]:hover [part=inactive] {
  background: var(--base-track-hover-color);
}
[part=base]:hover [part=fill] {
  background: var(--track-hover-color);
}

[part=fill] {
  background: var(--track-color);
}

[part=inactive] {
  background: var(--base-track-color);
}

[part=track],
[part=fill],
[part=inactive] {
  height: 0.125rem;
  top: initial;
  border-radius: initial;
}

[part=steps] {
  color: var(--track-step-color);
  height: 0.125rem;
}
[part=steps] line {
  -webkit-transform: none;
      -ms-transform: none;
          transform: none;
}

[part=tick] {
  background: var(--tick-color);
}

[part=tick-label] {
  color: var(--tick-label-color);
}

[part~=thumb] {
  background: var(--thumb-color);
  width: 0.875rem;
  height: 0.875rem;
  top: calc((0.875rem - 0.4375rem) * -1);
  -webkit-margin-start: calc((0.875rem - 0.4375rem) * -1);
          margin-inline-start: calc((0.875rem - 0.4375rem) * -1);
}
[part~=thumb]:focus {
  -webkit-box-shadow: 0 0 0 0.1875rem var(--thumb-focus-color);
          box-shadow: 0 0 0 0.1875rem var(--thumb-focus-color);
}
[part~=thumb]::after {
  display: none;
}

[part=thumb-label-inner] {
  background: var(--label-background-color);
  color: var(--label-text-color);
}
[part=thumb-label-inner]::after {
  border-top: 0.625rem solid var(--label-background-color);
}

:host([disabled]) [part=inactive],
:host(:disabled) [part=inactive] {
  background: var(--disabled-base-track-color);
}
:host([disabled]) [part=fill],
:host([disabled]) [part=tick],
:host(:disabled) [part=fill],
:host(:disabled) [part=tick] {
  background: var(--disabled-fill-track-color);
}
:host([disabled]) [part~=thumb],
:host(:disabled) [part~=thumb] {
  background: var(--disabled-thumb-color);
}
:host([disabled]) [part=tick-label],
:host(:disabled) [part=tick-label] {
  color: var(--disabled-fill-track-color);
}`;
