// igniteui-webcomponents/components/slider/themes/light/slider.material.css.js
import { css } from 'lit';
export const styles = css `:host {
  --track-color: var(--ig-slider-track-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --track-step-color: var(--ig-slider-track-step-color, hsla(var(--ig-secondary-200), var(--ig-secondary-a)));
  --track-step-size: var(--ig-slider-track-step-size, 0.1875rem);
  --track-hover-color: var(--ig-slider-track-hover-color, hsla(var(--ig-secondary-400), var(--ig-secondary-a)));
  --thumb-color: var(--ig-slider-thumb-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --thumb-focus-color: var(--ig-slider-thumb-focus-color, transparent);
  --thumb-border-color: var(--ig-slider-thumb-border-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --thumb-border-focus-color: var(--ig-slider-thumb-border-focus-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --thumb-disabled-border-color: var(--ig-slider-thumb-disabled-border-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --disabled-thumb-color: var(--ig-slider-disabled-thumb-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --label-background-color: var(--ig-slider-label-background-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --label-text-color: var(--ig-slider-label-text-color, var(--ig-gray-700-contrast));
  --base-track-color: var(--ig-slider-base-track-color, hsla(var(--ig-secondary-500), 0.24));
  --base-track-hover-color: var(--ig-slider-base-track-hover-color, hsla(var(--ig-secondary-500), 0.24));
  --disabled-base-track-color: var(--ig-slider-disabled-base-track-color, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --disabled-fill-track-color: var(--ig-slider-disabled-fill-track-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --tick-color: var(--ig-slider-tick-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --tick-label-color: var(--ig-slider-tick-label-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
}

[part=track] {
  height: 0.375rem;
}

[part=fill] {
  background: var(--track-color);
  height: 0.375rem;
}

[part=inactive] {
  background: var(--base-track-color);
  height: 0.25rem;
  top: 0.0625rem;
}

[part=steps] {
  color: var(--track-step-color);
}

[part~=thumb] {
  width: 1.25rem;
  height: 1.25rem;
  top: calc((1.25rem - 0.625rem) * -1);
  -webkit-margin-start: calc((1.25rem - 0.625rem) * -1);
          margin-inline-start: calc((1.25rem - 0.625rem) * -1);
  background: var(--thumb-color);
}
[part~=thumb]::after {
  border-radius: clamp(0rem, calc(var(--ig-radius-factor, 1) * 1.25rem), 1.25rem);
  position: absolute;
  content: "";
  width: 2.5rem;
  height: 2.5rem;
  background: var(--thumb-color);
  top: calc(50% - 1.25rem);
  inset-inline-start: calc(50% - 1.25rem);
  opacity: 0;
  -webkit-transform: scale(0);
      -ms-transform: scale(0);
          transform: scale(0);
  -webkit-transform-origin: center center;
      -ms-transform-origin: center center;
          transform-origin: center center;
  -webkit-transition: opacity 0.1s ease-out, -webkit-transform 0.1s ease-out;
  transition: opacity 0.1s ease-out, -webkit-transform 0.1s ease-out;
  transition: transform 0.1s ease-out, opacity 0.1s ease-out;
  transition: transform 0.1s ease-out, opacity 0.1s ease-out, -webkit-transform 0.1s ease-out;
  overflow: hidden;
}
[part~=thumb]:hover::after {
  opacity: 0.12;
  -webkit-transform: scale(1);
      -ms-transform: scale(1);
          transform: scale(1);
}
[part~=thumb]:focus::after {
  opacity: 0.18;
  -webkit-transform: scale(1);
      -ms-transform: scale(1);
          transform: scale(1);
}
[part~=thumb]:active::after {
  opacity: 0.24;
}

[part=tick] {
  background: var(--tick-color);
}

[part=tick-label] {
  color: var(--tick-label-color);
}

[part=thumb-label-inner] {
  color: var(--label-text-color);
  background: var(--label-background-color);
}
[part=thumb-label-inner]::after {
  border-top: 0.625rem solid var(--label-background-color);
}

:host([disabled]) [part=inactive],
:host(:disabled) [part=inactive] {
  background: var(--disabled-base-track-color);
}
:host([disabled]) [part=fill],
:host([disabled]) [part=tick],
:host(:disabled) [part=fill],
:host(:disabled) [part=tick] {
  background: var(--disabled-fill-track-color);
}
:host([disabled]) [part~=thumb],
:host(:disabled) [part~=thumb] {
  background: var(--disabled-thumb-color);
}
:host([disabled]) [part=tick-label],
:host(:disabled) [part=tick-label] {
  color: var(--disabled-fill-track-color);
}`;
