// igniteui-webcomponents/components/slider/themes/light/slider.bootstrap.css.js
import { css } from 'lit';
export const styles = css `:host {
  --track-color: var(--ig-slider-track-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --track-step-color: var(--ig-slider-track-step-color, white);
  --track-step-size: var(--ig-slider-track-step-size, 0.1875rem);
  --track-hover-color: var(--ig-slider-track-hover-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --thumb-color: var(--ig-slider-thumb-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --thumb-focus-color: var(--ig-slider-thumb-focus-color, hsla(var(--ig-primary-200), var(--ig-primary-a)));
  --thumb-border-color: var(--ig-slider-thumb-border-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --thumb-border-focus-color: var(--ig-slider-thumb-border-focus-color, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --thumb-disabled-border-color: var(--ig-slider-thumb-disabled-border-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --disabled-thumb-color: var(--ig-slider-disabled-thumb-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --label-background-color: var(--ig-slider-label-background-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --label-text-color: var(--ig-slider-label-text-color, var(--ig-gray-700-contrast));
  --base-track-color: var(--ig-slider-base-track-color, hsla(var(--ig-secondary-500), 0.24));
  --base-track-hover-color: var(--ig-slider-base-track-hover-color, hsla(var(--ig-secondary-500), 0.24));
  --disabled-base-track-color: var(--ig-slider-disabled-base-track-color, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --disabled-fill-track-color: var(--ig-slider-disabled-fill-track-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --tick-color: var(--ig-slider-tick-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --tick-label-color: var(--ig-slider-tick-label-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
}

[part=track],
[part=inactive] {
  height: 0.5rem;
}

[part=fill] {
  display: none;
}

[part=inactive] {
  background: var(--base-track-color);
}

[part=steps] {
  color: var(--track-step-color);
}

[part~=thumb] {
  width: 1rem;
  height: 1rem;
  background: var(--thumb-color);
  top: calc((1rem - calc(1rem / 2)) * -1);
  -webkit-margin-start: calc((1rem - calc(1rem / 2)) * -1);
          margin-inline-start: calc((1rem - calc(1rem / 2)) * -1);
}
[part~=thumb]:focus {
  border: 0.0625rem solid var(--thumb-border-focus-color);
  -webkit-box-shadow: 0 0 0 0.1875rem var(--thumb-focus-color);
          box-shadow: 0 0 0 0.1875rem var(--thumb-focus-color);
}

[part=tick] {
  background: var(--tick-color);
}

[part=tick-label] {
  color: var(--tick-label-color);
}

[part=thumb-label-inner] {
  color: var(--label-text-color);
  background: var(--label-background-color);
}
[part=thumb-label-inner]::after {
  border-top: 0.625rem solid var(--label-background-color);
}

:host([disabled]) [part=inactive],
:host([disabled]) [part=tick],
:host(:disabled) [part=inactive],
:host(:disabled) [part=tick] {
  background: var(--disabled-base-track-color);
}
:host([disabled]) [part~=thumb],
:host(:disabled) [part~=thumb] {
  background: var(--disabled-thumb-color);
}
:host([disabled]) [part=tick-label],
:host(:disabled) [part=tick-label] {
  color: var(--disabled-base-track-color);
}`;
