// igniteui-webcomponents/components/input/themes/light/input.fluent.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium: min(
      clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
      clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
  );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

:host([hidden]),
[hidden] {
  display: none !important;
}
:host {
  --search-resting-elevation: var(--ig-input-search-resting-elevation, var(--ig-elevation-0));
  --search-hover-elevation: var(--ig-input-search-hover-elevation, var(--ig-elevation-0));
  --search-disabled-elevation: var(--ig-input-search-disabled-elevation, var(--ig-elevation-0));
  --helper-text-color: var(--ig-input-helper-text-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --input-prefix-background: var(--ig-input-input-prefix-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --input-prefix-color: var(--ig-input-input-prefix-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --input-prefix-background--filled: var(--ig-input-input-prefix-background--filled, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --input-prefix-color--filled: var(--ig-input-input-prefix-color--filled, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --input-prefix-background--focused: var(--ig-input-input-prefix-background--focused, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --input-prefix-color--focused: var(--ig-input-input-prefix-color--focused, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --input-suffix-background: var(--ig-input-input-suffix-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --input-suffix-color: var(--ig-input-input-suffix-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --input-suffix-background--filled: var(--ig-input-input-suffix-background--filled, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --input-suffix-color--filled: var(--ig-input-input-suffix-color--filled, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --input-suffix-background--focused: var(--ig-input-input-suffix-background--focused, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --input-suffix-color--focused: var(--ig-input-input-suffix-color--focused, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --idle-text-color: var(--ig-input-idle-text-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --filled-text-color: var(--ig-input-filled-text-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --focused-text-color: var(--ig-input-focused-text-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --disabled-text-color: var(--ig-input-disabled-text-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --idle-secondary-color: var(--ig-input-idle-secondary-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --filled-secondary-color: var(--ig-input-filled-secondary-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --focused-secondary-color: var(--ig-input-focused-secondary-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --idle-bottom-line-color: var(--ig-input-idle-bottom-line-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --hover-bottom-line-color: var(--ig-input-hover-bottom-line-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --focused-bottom-line-color: var(--ig-input-focused-bottom-line-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --interim-bottom-line-color: var(--ig-input-interim-bottom-line-color, hsla(var(--ig-primary-500), 0.12));
  --disabled-bottom-line-color: var(--ig-input-disabled-bottom-line-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --disabled-border-color: var(--ig-input-disabled-border-color, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --border-color: var(--ig-input-border-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --hover-border-color: var(--ig-input-hover-border-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --focused-border-color: var(--ig-input-focused-border-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --border-background: var(--ig-input-border-background, transparent);
  --border-disabled-background: var(--ig-input-border-disabled-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --search-background: var(--ig-input-search-background, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --search-disabled-background: var(--ig-input-search-disabled-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --success-secondary-color: var(--ig-input-success-secondary-color, hsla(var(--ig-success-500), var(--ig-success-a)));
  --warning-secondary-color: var(--ig-input-warning-secondary-color, hsla(var(--ig-warn-500), var(--ig-warn-a)));
  --error-secondary-color: var(--ig-input-error-secondary-color, hsla(var(--ig-error-500), var(--ig-error-a)));
  --placeholder-color: var(--ig-input-placeholder-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --disabled-placeholder-color: var(--ig-input-disabled-placeholder-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --box-background: var(--ig-input-box-background, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --box-background-hover: var(--ig-input-box-background-hover, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --box-background-focus: var(--ig-input-box-background-focus, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --box-disabled-background: var(--ig-input-box-disabled-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --box-border-radius: var(--ig-input-box-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.1) * 1.25rem), 1.25rem));
  --border-border-radius: var(--ig-input-border-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.1) * 1.25rem), 1.25rem));
  --search-border-radius: var(--ig-input-search-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.1) * 1.25rem), 1.25rem));
  --success-shadow-color: var(--ig-input-success-shadow-color, hsla(var(--ig-success-500), 0.38));
  --error-shadow-color: var(--ig-input-error-shadow-color, hsla(var(--ig-error-500), 0.38));
  --size: var(--ig-input-size, max(var(--is-large, 1) * max(3rem, -1 * 3rem), var(--is-medium, 1) * max(2.5rem, -1 * 2.5rem), var(--is-small, 1) * max(2rem, -1 * 2rem)));
}
:host [part=prefix],
:host [part=suffix] {
  font-size: 0.875rem;
  cursor: default;
}
:host [part=prefix] ::slotted(*),
:host [part=suffix] ::slotted(*) {
  color: inherit;
}
:host [name=prefix]::slotted(*),
:host [name=suffix]::slotted(*) {
  display: -webkit-inline-box;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  height: 100%;
  padding-inline: max(var(--is-large, 1) * max(16px, -1 * 16px) * var(--ig-spacing-inline-large, var(--ig-spacing-inline, --ig-spacing)), var(--is-medium, 1) * max(12px, -1 * 12px) * var(--ig-spacing-inline-medium, var(--ig-spacing-inline, --ig-spacing)), var(--is-small, 1) * max(8px, -1 * 8px) * var(--ig-spacing-inline-small, var(--ig-spacing-inline, --ig-spacing)));
}
:host [part=prefix] {
  color: var(--input-prefix-color);
  background: var(--input-prefix-background);
  grid-area: 1/1;
}
:host [part=suffix] {
  color: var(--input-suffix-color);
  background: var(--input-suffix-background);
  grid-area: 1/3;
}
:host [part=label] {
  font-size: 0.875rem;
  font-weight: 600;
  color: var(--idle-secondary-color);
}
:host [part^=container] {
  height: var(--size);
  border-radius: var(--border-border-radius);
  border: 0.0625rem solid var(--border-color);
  -webkit-transition: none;
  transition: none;
  cursor: text;
  grid-template-columns: auto 1fr auto;
  overflow: hidden;
}
:host [part~=input] {
  color: var(--idle-text-color);
  height: calc(var(--size) - 0.125rem);
  background: initial;
  font-size: 0.875rem;
  padding-inline: 0.5rem;
  max-height: 100%;
  border: none;
  grid-area: 1/2;
}
:host [part=helper-text] {
  padding: 0;
  margin-top: 0.3125rem;
  color: var(--helper-text-color);
}

:host(:hover) [part^=container] {
  border-color: var(--hover-border-color);
}

[part~=filled] [part=prefix] {
  color: var(--input-prefix-color--filled);
  background: var(--input-prefix-background--filled);
}
[part~=filled] [part=suffix] {
  color: var(--input-suffix-color--filled);
  background: var(--input-suffix-background--filled);
}

:host(:focus-within) [part=label] {
  color: hsla(var(--ig-gray-800), var(--ig-gray-a));
}
:host(:focus-within) [part^=container] {
  border-color: var(--focused-border-color);
  border-width: 0.125rem;
}
:host(:focus-within) [part~=input] {
  height: calc(var(--size) - 0.25rem);
}
:host(:focus-within) [part=prefix] {
  color: var(--input-prefix-color--focused);
  background: var(--input-prefix-background--focused);
  -webkit-margin-start: -0.0625rem;
          margin-inline-start: -0.0625rem;
}
:host(:focus-within) [part=suffix] {
  color: var(--input-suffix-color--focused);
  background: var(--input-suffix-background--focused);
  -webkit-margin-end: -0.0625rem;
          margin-inline-end: -0.0625rem;
}
:host(:focus-within) [name=prefix]::slotted(*),
:host(:focus-within) [name=suffix]::slotted(*) {
  padding-block: max(var(--is-large, 1) * max(10px, -1 * 10px) * var(--ig-spacing-block-large, var(--ig-spacing-block, --ig-spacing)), var(--is-medium, 1) * max(8px, -1 * 8px) * var(--ig-spacing-block-medium, var(--ig-spacing-block, --ig-spacing)), var(--is-small, 1) * max(6px, -1 * 6px) * var(--ig-spacing-block-small, var(--ig-spacing-block, --ig-spacing)));
}

:host([required]) [part=label]::after {
  color: var(--error-secondary-color);
}

:host([invalid]) [part^=container] {
  border-color: var(--error-secondary-color);
}
:host([invalid]) [part=helper-text] {
  color: var(--error-secondary-color);
}

:host(:disabled) [part^=container],
:host([disabled]) [part^=container] {
  border-color: var(--disabled-border-color);
}
:host(:disabled) [part^=container],
:host(:disabled) [part=prefix],
:host(:disabled) [part=suffix],
:host([disabled]) [part^=container],
:host([disabled]) [part=prefix],
:host([disabled]) [part=suffix] {
  background: var(--border-disabled-background);
}
:host(:disabled) [part~=input]::-webkit-input-placeholder, :host([disabled]) [part~=input]::-webkit-input-placeholder {
  color: var(--disabled-text-color);
}
:host(:disabled) [part~=input]::-moz-placeholder, :host([disabled]) [part~=input]::-moz-placeholder {
  color: var(--disabled-text-color);
}
:host(:disabled) [part~=input]:-ms-input-placeholder, :host([disabled]) [part~=input]:-ms-input-placeholder {
  color: var(--disabled-text-color);
}
:host(:disabled) [part~=input]::-ms-input-placeholder, :host([disabled]) [part~=input]::-ms-input-placeholder {
  color: var(--disabled-text-color);
}
:host(:disabled) [part=prefix],
:host(:disabled) [part=suffix],
:host(:disabled) [part=label],
:host(:disabled) [part=label]::after,
:host(:disabled) [part=helper-text],
:host(:disabled) [part~=input],
:host(:disabled) [part~=input]::placeholder,
:host([disabled]) [part=prefix],
:host([disabled]) [part=suffix],
:host([disabled]) [part=label],
:host([disabled]) [part=label]::after,
:host([disabled]) [part=helper-text],
:host([disabled]) [part~=input],
:host([disabled]) [part~=input]::placeholder {
  color: var(--disabled-text-color);
}`;
