// igniteui-webcomponents/components/input/themes/light/input.bootstrap.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium: min(
      clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
      clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
  );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

:host([hidden]),
[hidden] {
  display: none !important;
}
:host {
  --search-resting-elevation: var(--ig-input-search-resting-elevation, var(--ig-elevation-1));
  --search-hover-elevation: var(--ig-input-search-hover-elevation, var(--ig-elevation-2));
  --search-disabled-elevation: var(--ig-input-search-disabled-elevation, var(--ig-elevation-0));
  --helper-text-color: var(--ig-input-helper-text-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --input-prefix-background: var(--ig-input-input-prefix-background, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --input-prefix-color: var(--ig-input-input-prefix-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --input-prefix-background--filled: var(--ig-input-input-prefix-background--filled, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --input-prefix-color--filled: var(--ig-input-input-prefix-color--filled, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --input-prefix-background--focused: var(--ig-input-input-prefix-background--focused, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --input-prefix-color--focused: var(--ig-input-input-prefix-color--focused, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --input-suffix-background: var(--ig-input-input-suffix-background, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --input-suffix-color: var(--ig-input-input-suffix-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --input-suffix-background--filled: var(--ig-input-input-suffix-background--filled, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --input-suffix-color--filled: var(--ig-input-input-suffix-color--filled, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --input-suffix-background--focused: var(--ig-input-input-suffix-background--focused, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --input-suffix-color--focused: var(--ig-input-input-suffix-color--focused, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --idle-text-color: var(--ig-input-idle-text-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --filled-text-color: var(--ig-input-filled-text-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --focused-text-color: var(--ig-input-focused-text-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --disabled-text-color: var(--ig-input-disabled-text-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --idle-secondary-color: var(--ig-input-idle-secondary-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --filled-secondary-color: var(--ig-input-filled-secondary-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --focused-secondary-color: var(--ig-input-focused-secondary-color, hsla(var(--ig-primary-500), 0.38));
  --idle-bottom-line-color: var(--ig-input-idle-bottom-line-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --hover-bottom-line-color: var(--ig-input-hover-bottom-line-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --focused-bottom-line-color: var(--ig-input-focused-bottom-line-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --interim-bottom-line-color: var(--ig-input-interim-bottom-line-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --disabled-bottom-line-color: var(--ig-input-disabled-bottom-line-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --disabled-border-color: var(--ig-input-disabled-border-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --border-color: var(--ig-input-border-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --hover-border-color: var(--ig-input-hover-border-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --focused-border-color: var(--ig-input-focused-border-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --border-background: var(--ig-input-border-background, transparent);
  --border-disabled-background: var(--ig-input-border-disabled-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --search-background: var(--ig-input-search-background, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --search-disabled-background: var(--ig-input-search-disabled-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --success-secondary-color: var(--ig-input-success-secondary-color, hsla(var(--ig-success-500), var(--ig-success-a)));
  --warning-secondary-color: var(--ig-input-warning-secondary-color, hsla(var(--ig-warn-500), var(--ig-warn-a)));
  --error-secondary-color: var(--ig-input-error-secondary-color, hsla(var(--ig-error-500), var(--ig-error-a)));
  --placeholder-color: var(--ig-input-placeholder-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --disabled-placeholder-color: var(--ig-input-disabled-placeholder-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --box-background: var(--ig-input-box-background, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --box-background-hover: var(--ig-input-box-background-hover, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --box-background-focus: var(--ig-input-box-background-focus, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --box-disabled-background: var(--ig-input-box-disabled-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --box-border-radius: var(--ig-input-box-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.2) * 1.25rem), 1.25rem));
  --border-border-radius: var(--ig-input-border-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.2) * 1.25rem), 1.25rem));
  --search-border-radius: var(--ig-input-search-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.2) * 1.25rem), 1.25rem));
  --success-shadow-color: var(--ig-input-success-shadow-color, hsla(var(--ig-success-500), 0.38));
  --error-shadow-color: var(--ig-input-error-shadow-color, hsla(var(--ig-error-500), 0.38));
  --size: var(--ig-input-size, max(var(--is-large, 1) * max(3rem, -1 * 3rem), var(--is-medium, 1) * max(2.375rem, -1 * 2.375rem), var(--is-small, 1) * max(1.9375rem, -1 * 1.9375rem)));
  --font: max(var(--is-large, 1) * max(1.25rem, -1 * 1.25rem), var(--is-medium, 1) * max(1rem, -1 * 1rem), var(--is-small, 1) * max(0.875rem, -1 * 0.875rem));
  --input-font: max(var(--is-large, 1) * max(1.25rem, -1 * 1.25rem), var(--is-medium, 1) * max(1rem, -1 * 1rem), var(--is-small, 1) * max(1rem, -1 * 1rem));
}

[part~=container] {
  border-radius: var(--border-border-radius);
  border-style: none;
  grid-template-columns: auto 1fr auto;
}
[part~=container]::after {
  display: none;
}

[part~=input] {
  border-radius: var(--border-border-radius);
  font-size: var(--input-font);
  padding-inline: max(var(--is-large, 1) * max(16px, -1 * 16px) * var(--ig-spacing-inline-large, var(--ig-spacing-inline, --ig-spacing)), var(--is-medium, 1) * max(12px, -1 * 12px) * var(--ig-spacing-inline-medium, var(--ig-spacing-inline, --ig-spacing)), var(--is-small, 1) * max(8px, -1 * 8px) * var(--ig-spacing-inline-small, var(--ig-spacing-inline, --ig-spacing)));
  padding-block: max(var(--is-large, 1) * max(8px, -1 * 8px) * var(--ig-spacing-block-large, var(--ig-spacing-block, --ig-spacing)), var(--is-medium, 1) * max(6px, -1 * 6px) * var(--ig-spacing-block-medium, var(--ig-spacing-block, --ig-spacing)), var(--is-small, 1) * max(4px, -1 * 4px) * var(--ig-spacing-block-small, var(--ig-spacing-block, --ig-spacing)));
  color: var(--idle-text-color);
  background: transparent;
  -webkit-box-shadow: none;
          box-shadow: none;
  z-index: 1;
  border: 1px solid var(--border-color);
  -webkit-transition: border 0.15s ease-out, -webkit-box-shadow 0.15s ease-out;
  transition: border 0.15s ease-out, -webkit-box-shadow 0.15s ease-out;
  transition: box-shadow 0.15s ease-out, border 0.15s ease-out;
  transition: box-shadow 0.15s ease-out, border 0.15s ease-out, -webkit-box-shadow 0.15s ease-out;
  grid-area: 1/2;
  background-clip: padding-box;
}

[part~=input],
[part~=container] {
  height: var(--size);
}

[part*="input prefixed"] {
  border-start-start-radius: 0;
  border-end-start-radius: 0;
}

[part*="input suffixed"] {
  border-start-end-radius: 0;
  border-end-end-radius: 0;
}

[part*="input prefixed suffixed"] {
  border-radius: 0;
}

[part~=label] {
  line-height: 1.5rem;
  margin-bottom: 0.25rem;
  color: var(--idle-secondary-color);
}
[part~=label]:empty {
  display: none;
}

[name=prefix]::slotted(*),
[name=suffix]::slotted(*) {
  display: -webkit-inline-box;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  height: 100%;
  padding-inline: max(var(--is-large, 1) * max(16px, -1 * 16px) * var(--ig-spacing-inline-large, var(--ig-spacing-inline, --ig-spacing)), var(--is-medium, 1) * max(12px, -1 * 12px) * var(--ig-spacing-inline-medium, var(--ig-spacing-inline, --ig-spacing)), var(--is-small, 1) * max(8px, -1 * 8px) * var(--ig-spacing-inline-small, var(--ig-spacing-inline, --ig-spacing)));
}

[part=prefix],
[part=suffix] {
  position: relative;
  border-radius: none;
  min-width: auto;
  -webkit-transition: border 0.15s ease-out, -webkit-box-shadow 0.15s ease-out;
  transition: border 0.15s ease-out, -webkit-box-shadow 0.15s ease-out;
  transition: box-shadow 0.15s ease-out, border 0.15s ease-out;
  transition: box-shadow 0.15s ease-out, border 0.15s ease-out, -webkit-box-shadow 0.15s ease-out;
  font-size: var(--font);
}
[part=prefix] ::slotted(*),
[part=suffix] ::slotted(*) {
  color: inherit;
}

[part=prefix] {
  border-start-start-radius: inherit;
  border-end-start-radius: inherit;
  color: var(--input-prefix-color);
  background: var(--input-prefix-background);
  grid-area: 1/1;
  border-style: solid;
  border-color: var(--border-color);
  border-inline-start-width: 1px;
  border-inline-end-width: 0;
  border-block-start-width: 1px;
  border-block-end-width: 1px;
}

[part=suffix] {
  border-start-end-radius: inherit;
  border-end-end-radius: inherit;
  color: var(--input-suffix-color);
  background: var(--input-suffix-background);
  grid-area: 1/3;
  border-style: solid;
  border-color: var(--border-color);
  border-inline-start-width: 0;
  border-inline-end-width: 1px;
  border-block-start-width: 1px;
  border-block-end-width: 1px;
}

[part=helper-text] {
  font-family: var(--ig-body-2-font-family, var(--ig-font-family));
  font-size: var(--ig-body-2-font-size);
  font-weight: var(--ig-body-2-font-weight);
  font-style: var(--ig-body-2-font-style);
  line-height: var(--ig-body-2-line-height);
  letter-spacing: var(--ig-body-2-letter-spacing);
  text-transform: var(--ig-body-2-text-transform);
  margin-top: var(--ig-body-2-margin-top);
  margin-bottom: var(--ig-body-2-margin-bottom);
  margin-top: 0.25rem;
  color: var(--helper-text-color);
}

:host(:focus-within) [part=prefix] {
  color: var(--input-prefix-color--focused);
  background: var(--input-prefix-background--focused);
}
:host(:focus-within) [part=suffix] {
  color: var(--input-suffix-color--focused);
  background: var(--input-suffix-background--focused);
}
:host(:focus-within) [part~=input] {
  color: var(--input-prefix-color--focused);
  border-color: var(--focused-border-color);
  -webkit-box-shadow: 0 0 0 0.25rem var(--focused-secondary-color);
          box-shadow: 0 0 0 0.25rem var(--focused-secondary-color);
}

[part~=filled] [part=prefix],
[part~=filled] [part=suffix] {
  color: var(--input-prefix-color--filled);
  background: var(--input-prefix-background--filled);
}
[part~=filled] [part~=input] {
  color: var(--input-prefix-color--filled);
}

:host([invalid]) [part=helper-text] {
  color: var(--error-secondary-color);
}

[part~=prefixed] [part=prefix] ::slotted(*),
[part~=suffixed] [part=suffix] ::slotted(*) {
  color: inherit;
}

:host(:not([disabled])[invalid]) [part~=input],
:host(:not([disabled])[invalid]:focus-within) [part~=input] {
  border-color: var(--error-secondary-color);
  -webkit-box-shadow: 0 0 0 0.25rem var(--error-shadow-color);
          box-shadow: 0 0 0 0.25rem var(--error-shadow-color);
}

:host(:disabled) [part=prefix],
:host(:disabled) [part=suffix],
:host(:disabled) [part~=input],
:host([disabled]) [part=prefix],
:host([disabled]) [part=suffix],
:host([disabled]) [part~=input] {
  color: var(--disabled-text-color);
  background: var(--border-disabled-background);
  border-block-color: var(--disabled-border-color);
}
:host(:disabled) [part=prefix],
:host([disabled]) [part=prefix] {
  border-inline-start-color: var(--disabled-border-color);
}
:host(:disabled) [part=suffix],
:host([disabled]) [part=suffix] {
  border-inline-end-color: var(--disabled-border-color);
}
:host(:disabled) [part~=input],
:host([disabled]) [part~=input] {
  border-color: var(--disabled-border-color);
}
:host(:disabled) [part=helper-text],
:host(:disabled) [part=label],
:host([disabled]) [part=helper-text],
:host([disabled]) [part=label] {
  color: var(--disabled-text-color);
}
:host(:disabled) [part~=input]::-webkit-input-placeholder, :host([disabled]) [part~=input]::-webkit-input-placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) [part~=input]::-moz-placeholder, :host([disabled]) [part~=input]::-moz-placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) [part~=input]:-ms-input-placeholder, :host([disabled]) [part~=input]:-ms-input-placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) [part~=input]::-ms-input-placeholder, :host([disabled]) [part~=input]::-ms-input-placeholder {
  color: var(--disabled-placeholder-color);
}
:host(:disabled) [part~=input]::placeholder,
:host([disabled]) [part~=input]::placeholder {
  color: var(--disabled-placeholder-color);
}`;
