// igniteui-webcomponents/components/input/themes/light/input.indigo.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium: min(
      clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
      clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
  );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

:host([hidden]),
[hidden] {
  display: none !important;
}
:host [part=suffix] ::slotted(*), :host [part=prefix] ::slotted(*) {
  display: -webkit-inline-box;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  height: calc(100% - 0.125rem);
  padding-inline: max(var(--is-large, 1) * max(16px, -1 * 16px) * var(--ig-spacing-inline-large, var(--ig-spacing-inline, --ig-spacing)), var(--is-medium, 1) * max(12px, -1 * 12px) * var(--ig-spacing-inline-medium, var(--ig-spacing-inline, --ig-spacing)), var(--is-small, 1) * max(8px, -1 * 8px) * var(--ig-spacing-inline-small, var(--ig-spacing-inline, --ig-spacing)));
}

:host {
  --search-resting-elevation: var(--ig-input-search-resting-elevation, var(--ig-elevation-1));
  --search-hover-elevation: var(--ig-input-search-hover-elevation, var(--ig-elevation-1));
  --search-disabled-elevation: var(--ig-input-search-disabled-elevation, var(--ig-elevation-1));
  --helper-text-color: var(--ig-input-helper-text-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --input-prefix-background: var(--ig-input-input-prefix-background, transparent);
  --input-prefix-color: var(--ig-input-input-prefix-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --input-prefix-background--filled: var(--ig-input-input-prefix-background--filled, inherit);
  --input-prefix-color--filled: var(--ig-input-input-prefix-color--filled, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --input-prefix-background--focused: var(--ig-input-input-prefix-background--focused, inherit);
  --input-prefix-color--focused: var(--ig-input-input-prefix-color--focused, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --input-suffix-background: var(--ig-input-input-suffix-background, transparent);
  --input-suffix-color: var(--ig-input-input-suffix-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --input-suffix-background--filled: var(--ig-input-input-suffix-background--filled, inherit);
  --input-suffix-color--filled: var(--ig-input-input-suffix-color--filled, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --input-suffix-background--focused: var(--ig-input-input-suffix-background--focused, inherit);
  --input-suffix-color--focused: var(--ig-input-input-suffix-color--focused, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --idle-text-color: var(--ig-input-idle-text-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --filled-text-color: var(--ig-input-filled-text-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --focused-text-color: var(--ig-input-focused-text-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --disabled-text-color: var(--ig-input-disabled-text-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --idle-secondary-color: var(--ig-input-idle-secondary-color, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --filled-secondary-color: var(--ig-input-filled-secondary-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --focused-secondary-color: var(--ig-input-focused-secondary-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --idle-bottom-line-color: var(--ig-input-idle-bottom-line-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --hover-bottom-line-color: var(--ig-input-hover-bottom-line-color, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --focused-bottom-line-color: var(--ig-input-focused-bottom-line-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --interim-bottom-line-color: var(--ig-input-interim-bottom-line-color, hsla(var(--ig-primary-500), 0.12));
  --disabled-bottom-line-color: var(--ig-input-disabled-bottom-line-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --disabled-border-color: var(--ig-input-disabled-border-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --border-color: var(--ig-input-border-color, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --hover-border-color: var(--ig-input-hover-border-color, hsla(var(--ig-gray-600), var(--ig-gray-a)));
  --focused-border-color: var(--ig-input-focused-border-color, hsla(var(--ig-primary-400), var(--ig-primary-a)));
  --border-background: var(--ig-input-border-background, transparent);
  --border-disabled-background: var(--ig-input-border-disabled-background, transparent);
  --search-background: var(--ig-input-search-background, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --search-disabled-background: var(--ig-input-search-disabled-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --success-secondary-color: var(--ig-input-success-secondary-color, hsla(var(--ig-success-500), var(--ig-success-a)));
  --warning-secondary-color: var(--ig-input-warning-secondary-color, hsla(var(--ig-warn-500), var(--ig-warn-a)));
  --error-secondary-color: var(--ig-input-error-secondary-color, hsla(var(--ig-error-500), var(--ig-error-a)));
  --placeholder-color: var(--ig-input-placeholder-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --disabled-placeholder-color: var(--ig-input-disabled-placeholder-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --box-background: var(--ig-input-box-background, transparent);
  --box-background-hover: var(--ig-input-box-background-hover, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --box-background-focus: var(--ig-input-box-background-focus, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --box-disabled-background: var(--ig-input-box-disabled-background, transparent);
  --box-border-radius: var(--ig-input-box-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0) * 1.25rem), 1.25rem));
  --border-border-radius: var(--ig-input-border-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0) * 1.25rem), 1.25rem));
  --search-border-radius: var(--ig-input-search-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.1) * 1.25rem), 1.25rem));
  --success-shadow-color: var(--ig-input-success-shadow-color, hsla(var(--ig-success-500), 0.38));
  --error-shadow-color: var(--ig-input-error-shadow-color, hsla(var(--ig-error-500), 0.38));
  --size: var(--ig-input-size, max(var(--is-large, 1) * max(3.5rem, -1 * 3.5rem), var(--is-medium, 1) * max(3rem, -1 * 3rem), var(--is-small, 1) * max(2.5rem, -1 * 2.5rem)));
}
:host [part=prefix],
:host [part=suffix] {
  -webkit-transition-duration: 0ms;
          transition-duration: 0ms;
}
:host [part=prefix] ::slotted(*),
:host [part=suffix] ::slotted(*) {
  height: 100%;
  color: inherit;
  -webkit-transition-duration: 0ms;
          transition-duration: 0ms;
}
:host [part=prefix] {
  border-start-start-radius: var(--border-border-radius);
  color: var(--input-prefix-color);
  background: var(--input-prefix-background);
  grid-area: 1/1;
}
:host [part=suffix] {
  border-start-end-radius: var(--border-border-radius);
  color: var(--input-suffix-color);
  background: var(--input-suffix-background);
  grid-area: 1/3;
}
:host [part~=label] {
  color: var(--idle-secondary-color);
  font-size: 0.875rem;
  font-weight: 600;
  line-height: 1;
}
:host [part^=container] {
  border-radius: var(--box-border-radius);
  border-end-start-radius: 0;
  border-end-end-radius: 0;
  border-bottom: 0.0625rem solid var(--idle-bottom-line-color);
  -webkit-transition: background 0.25s ease-out;
  transition: background 0.25s ease-out;
  grid-template-columns: auto 1fr auto;
  height: var(--size);
}
:host [part^=container]::after {
  position: absolute;
  content: "";
  width: 100%;
  height: 0.125rem;
  left: 0;
  right: 0;
  bottom: -1px;
  background: var(--focused-bottom-line-color);
  -webkit-transform: scaleY(0);
      -ms-transform: scaleY(0);
          transform: scaleY(0);
  -webkit-transition: opacity 180ms cubic-bezier(0.4, 0, 0.2, 1), -webkit-transform 180ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: opacity 180ms cubic-bezier(0.4, 0, 0.2, 1), -webkit-transform 180ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: transform 180ms cubic-bezier(0.4, 0, 0.2, 1), opacity 180ms cubic-bezier(0.4, 0, 0.2, 1);
  transition: transform 180ms cubic-bezier(0.4, 0, 0.2, 1), opacity 180ms cubic-bezier(0.4, 0, 0.2, 1), -webkit-transform 180ms cubic-bezier(0.4, 0, 0.2, 1);
  -webkit-transform-origin: bottom center;
      -ms-transform-origin: bottom center;
          transform-origin: bottom center;
}
:host [part~=input] {
  color: var(--idle-text-color);
  font-size: 1rem;
  background: initial;
  padding-block: 0.5rem;
  padding-inline: max(var(--is-large, 1) * max(1rem, -1 * 1rem) * var(--ig-spacing-inline-large, var(--ig-spacing-inline, --ig-spacing)), var(--is-medium, 1) * max(0.75rem, -1 * 0.75rem) * var(--ig-spacing-inline-medium, var(--ig-spacing-inline, --ig-spacing)), var(--is-small, 1) * max(0.5rem, -1 * 0.5rem) * var(--ig-spacing-inline-small, var(--ig-spacing-inline, --ig-spacing)));
  border: none;
  grid-area: 1/2;
  height: 100%;
}
:host [part~=prefixed] {
  -webkit-padding-start: 0.25rem;
          padding-inline-start: 0.25rem;
}
:host [part~=suffixed] {
  -webkit-padding-end: 0.25rem;
          padding-inline-end: 0.25rem;
}
:host [part=helper-text] {
  margin-top: 0.375rem;
  padding-inline: max(var(--is-large, 1) * max(1rem, -1 * 1rem) * var(--ig-spacing-inline-large, var(--ig-spacing-inline, --ig-spacing)), var(--is-medium, 1) * max(0.75rem, -1 * 0.75rem) * var(--ig-spacing-inline-medium, var(--ig-spacing-inline, --ig-spacing)), var(--is-small, 1) * max(0.5rem, -1 * 0.5rem) * var(--ig-spacing-inline-small, var(--ig-spacing-inline, --ig-spacing)));
  color: var(--helper-text-color);
}

:host(:focus-within) [part^=container],
:host(:hover) [part^=container] {
  background: var(--box-background-hover);
}

[part~=filled] [part=prefix] {
  color: var(--input-prefix-color--filled);
  background: var(--input-prefix-background--filled);
}
[part~=filled] [part=suffix] {
  color: var(--input-suffix-color--filled);
  background: var(--input-suffix-background--filled);
}
[part~=filled] [part~=input] {
  color: var(--filled-text-color);
}

:host(:focus-within) [part=prefix],
:host(:focus-within) [part=suffix] {
  -webkit-transition-duration: 0.25s;
          transition-duration: 0.25s;
}
:host(:focus-within) [part^=container]::after {
  -webkit-transform: scaleY(1);
      -ms-transform: scaleY(1);
          transform: scaleY(1);
}
:host(:focus-within) [part=prefix] {
  color: var(--input-prefix-color--focused);
  background: var(--input-prefix-background--focused);
}
:host(:focus-within) [part=suffix] {
  color: var(--input-suffix-color--focused);
  background: var(--input-suffix-background--focused);
}
:host(:focus-within) [part=label] {
  color: var(--focused-secondary-color);
}

:host([invalid]) [part^=container] {
  border-color: var(--error-secondary-color);
}
:host([invalid]) [part^=container]::after {
  background: var(--error-secondary-color);
}
:host([invalid]) [part=label] {
  color: var(--error-secondary-color);
}
:host([invalid]) [part=helper-text] {
  color: var(--error-secondary-color);
}

:host(:disabled) [part^=container],
:host([disabled]) [part^=container] {
  border-color: var(--disabled-bottom-line-color);
}
:host(:disabled) [part~=input]::-webkit-input-placeholder, :host([disabled]) [part~=input]::-webkit-input-placeholder {
  color: var(--disabled-text-color);
}
:host(:disabled) [part~=input]::-moz-placeholder, :host([disabled]) [part~=input]::-moz-placeholder {
  color: var(--disabled-text-color);
}
:host(:disabled) [part~=input]:-ms-input-placeholder, :host([disabled]) [part~=input]:-ms-input-placeholder {
  color: var(--disabled-text-color);
}
:host(:disabled) [part~=input]::-ms-input-placeholder, :host([disabled]) [part~=input]::-ms-input-placeholder {
  color: var(--disabled-text-color);
}
:host(:disabled) [part=prefix],
:host(:disabled) [part=suffix],
:host(:disabled) [part=label],
:host(:disabled) [part~=input],
:host(:disabled) [part=helper-text],
:host(:disabled) [part~=input]::placeholder,
:host([disabled]) [part=prefix],
:host([disabled]) [part=suffix],
:host([disabled]) [part=label],
:host([disabled]) [part~=input],
:host([disabled]) [part=helper-text],
:host([disabled]) [part~=input]::placeholder {
  color: var(--disabled-text-color);
}`;
