// igniteui-webcomponents/components/tree/themes/light/tree-item.fluent.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium: min(
      clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
      clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
  );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

:host([hidden]),
[hidden] {
  display: none !important;
}
:host {
  --background: var(--ig-tree-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --foreground: var(--ig-tree-foreground, var(--ig-surface-500-contrast));
  --background-selected: var(--ig-tree-background-selected, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --foreground-selected: var(--ig-tree-foreground-selected, var(--ig-gray-200-contrast));
  --background-active: var(--ig-tree-background-active, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --foreground-active: var(--ig-tree-foreground-active, var(--ig-gray-200-contrast));
  --background-active-selected: var(--ig-tree-background-active-selected, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --foreground-active-selected: var(--ig-tree-foreground-active-selected, var(--ig-gray-300-contrast));
  --border-color: var(--ig-tree-border-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --background-disabled: var(--ig-tree-background-disabled, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --foreground-disabled: var(--ig-tree-foreground-disabled, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --drop-area-color: var(--ig-tree-drop-area-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --hover-color: var(--ig-tree-hover-color, hsla(var(--ig-gray-900), 0.1));
  --size: var(--ig-tree-size, max(var(--is-large, 1) * max(3.125rem, -1 * 3.125rem), var(--is-medium, 1) * max(2.5rem, -1 * 2.5rem), var(--is-small, 1) * max(2rem, -1 * 2rem)));
}

[part~=wrapper] {
  min-height: var(--size);
  background: var(--background);
  color: var(--foreground);
}
[part~=wrapper]::after {
  background: transparent;
}
[part~=wrapper]:hover::after {
  background: var(--hover-color);
}

[part~=label] {
  max-height: var(--size);
}

[part~=active] {
  background: var(--background-active);
  color: var(--foreground-active);
}

[part~=selected] {
  background: var(--background-selected);
  color: var(--foreground-selected);
}

[part~=selected][part~=focused][part~=active],
[part~=selected][part~=focused],
[part~=selected][part~=active] {
  background: var(--background-active-selected);
  color: var(--foreground-active-selected);
}

[part~=select] {
  -webkit-margin-end: 0.5rem;
          margin-inline-end: 0.5rem;
}

[part~=focused] {
  -webkit-box-shadow: inset 0 0 0 1px var(--border-color);
          box-shadow: inset 0 0 0 1px var(--border-color);
}

:host([disabled]) [part~=label] {
  color: var(--foreground-disabled);
  background: var(--background-disabled);
}`;
