// igniteui-webcomponents/components/button/themes/button/light/button.fluent.css.js
import { css } from 'lit';
export const styles = css `:host([variant=flat]) {
  --shadow-color: var(--ig-button-shadow-color, transparent);
  --border-color: var(--ig-button-border-color, transparent);
  --hover-border-color: var(--ig-button-hover-border-color, transparent);
  --focus-border-color: var(--ig-button-focus-border-color, transparent);
  --focus-visible-border-color: var(--ig-button-focus-visible-border-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --active-border-color: var(--ig-button-active-border-color, transparent);
  --disabled-border-color: var(--ig-button-disabled-border-color, transparent);
  --disabled-background: var(--ig-button-disabled-background, transparent);
  --disabled-foreground: var(--ig-button-disabled-foreground, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --size: var(--ig-button-size, max(var(--is-large, 1) * max(2.375rem, -1 * 2.375rem), var(--is-medium, 1) * max(2rem, -1 * 2rem), var(--is-small, 1) * max(1.5rem, -1 * 1.5rem)));
  --resting-elevation: var(--ig-button-resting-elevation, var(--ig-elevation-0));
  --hover-elevation: var(--ig-button-hover-elevation, var(--ig-elevation-0));
  --focus-elevation: var(--ig-button-focus-elevation, var(--ig-elevation-0));
  --active-elevation: var(--ig-button-active-elevation, var(--ig-elevation-0));
  --background: var(--ig-button-background, transparent);
  --foreground: var(--ig-button-foreground, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --hover-background: var(--ig-button-hover-background, transparent);
  --hover-foreground: var(--ig-button-hover-foreground, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --focus-background: var(--ig-button-focus-background, transparent);
  --focus-foreground: var(--ig-button-focus-foreground, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --focus-visible-background: var(--ig-button-focus-visible-background, transparent);
  --focus-visible-foreground: var(--ig-button-focus-visible-foreground, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --active-background: var(--ig-button-active-background, transparent);
  --active-foreground: var(--ig-button-active-foreground, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --border-radius: var(--ig-button-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0) * 1.25rem), 1.25rem));
}
:host([variant=flat]) [part=base] {
  height: var(--size);
  border-radius: var(--border-radius);
}

:host([variant=outlined]) {
  --shadow-color: var(--ig-button-shadow-color, transparent);
  --border-color: var(--ig-button-border-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --hover-border-color: var(--ig-button-hover-border-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --focus-border-color: var(--ig-button-focus-border-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --focus-visible-border-color: var(--ig-button-focus-visible-border-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --active-border-color: var(--ig-button-active-border-color, hsla(var(--ig-gray-800), var(--ig-gray-a)));
  --disabled-border-color: var(--ig-button-disabled-border-color, transparent);
  --disabled-background: var(--ig-button-disabled-background, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --disabled-foreground: var(--ig-button-disabled-foreground, hsla(var(--ig-gray-400), 0.5));
  --size: var(--ig-button-size, max(var(--is-large, 1) * max(2.375rem, -1 * 2.375rem), var(--is-medium, 1) * max(2rem, -1 * 2rem), var(--is-small, 1) * max(1.5rem, -1 * 1.5rem)));
  --resting-elevation: var(--ig-button-resting-elevation, var(--ig-elevation-0));
  --hover-elevation: var(--ig-button-hover-elevation, var(--ig-elevation-0));
  --focus-elevation: var(--ig-button-focus-elevation, var(--ig-elevation-0));
  --active-elevation: var(--ig-button-active-elevation, var(--ig-elevation-0));
  --background: var(--ig-button-background, transparent);
  --foreground: var(--ig-button-foreground, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --hover-background: var(--ig-button-hover-background, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --hover-foreground: var(--ig-button-hover-foreground, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --focus-background: var(--ig-button-focus-background, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --focus-foreground: var(--ig-button-focus-foreground, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --focus-visible-background: var(--ig-button-focus-visible-background, transparent);
  --focus-visible-foreground: var(--ig-button-focus-visible-foreground, hsla(var(--ig-gray-900), var(--ig-gray-a)));
  --active-background: var(--ig-button-active-background, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --active-foreground: var(--ig-button-active-foreground, var(--ig-gray-300-contrast));
  --border-radius: var(--ig-button-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.1) * 1.25rem), 1.25rem));
}
:host([variant=outlined]) [part=base] {
  height: var(--size);
  border-radius: var(--border-radius);
}
:host([variant=outlined]) [part=base]::after {
  border-radius: calc(var(--border-radius) - 0.25rem);
}

:host([variant=contained]) {
  --shadow-color: var(--ig-button-shadow-color, transparent);
  --border-color: var(--ig-button-border-color, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --hover-border-color: var(--ig-button-hover-border-color, hsla(var(--ig-primary-600), var(--ig-primary-a)));
  --focus-border-color: var(--ig-button-focus-border-color, hsla(var(--ig-primary-600), var(--ig-primary-a)));
  --focus-visible-border-color: var(--ig-button-focus-visible-border-color, hsla(var(--ig-primary-600), var(--ig-primary-a)));
  --active-border-color: var(--ig-button-active-border-color, hsla(var(--ig-primary-800), var(--ig-primary-a)));
  --disabled-border-color: var(--ig-button-disabled-border-color, transparent);
  --disabled-background: var(--ig-button-disabled-background, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --disabled-foreground: var(--ig-button-disabled-foreground, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --size: var(--ig-button-size, max(var(--is-large, 1) * max(2.375rem, -1 * 2.375rem), var(--is-medium, 1) * max(2rem, -1 * 2rem), var(--is-small, 1) * max(1.5rem, -1 * 1.5rem)));
  --resting-elevation: var(--ig-button-resting-elevation, var(--ig-elevation-0));
  --hover-elevation: var(--ig-button-hover-elevation, var(--ig-elevation-0));
  --focus-elevation: var(--ig-button-focus-elevation, var(--ig-elevation-0));
  --active-elevation: var(--ig-button-active-elevation, var(--ig-elevation-0));
  --background: var(--ig-button-background, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --foreground: var(--ig-button-foreground, var(--ig-primary-900-contrast));
  --hover-background: var(--ig-button-hover-background, hsla(var(--ig-primary-600), var(--ig-primary-a)));
  --hover-foreground: var(--ig-button-hover-foreground, var(--ig-primary-900-contrast));
  --focus-background: var(--ig-button-focus-background, hsla(var(--ig-primary-600), var(--ig-primary-a)));
  --focus-foreground: var(--ig-button-focus-foreground, var(--ig-primary-900-contrast));
  --focus-visible-background: var(--ig-button-focus-visible-background, hsla(var(--ig-primary-600), var(--ig-primary-a)));
  --focus-visible-foreground: var(--ig-button-focus-visible-foreground, var(--ig-primary-900-contrast));
  --active-background: var(--ig-button-active-background, hsla(var(--ig-primary-800), var(--ig-primary-a)));
  --active-foreground: var(--ig-button-active-foreground, var(--ig-primary-900-contrast));
  --border-radius: var(--ig-button-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.1) * 1.25rem), 1.25rem));
}
:host([variant=contained]) [part=base] {
  height: var(--size);
  border-radius: var(--border-radius);
}
:host([variant=contained]) [part=base]::after {
  border-radius: calc(var(--border-radius) - 0.25rem);
}

:host([variant=fab]) {
  --shadow-color: var(--ig-button-shadow-color, transparent);
  --border-color: var(--ig-button-border-color, var(--ig-gray-900-contrast));
  --hover-border-color: var(--ig-button-hover-border-color, var(--ig-gray-900-contrast));
  --focus-border-color: var(--ig-button-focus-border-color, var(--ig-gray-900-contrast));
  --focus-visible-border-color: var(--ig-button-focus-visible-border-color, var(--ig-gray-900-contrast));
  --active-border-color: var(--ig-button-active-border-color, var(--ig-gray-900-contrast));
  --disabled-border-color: var(--ig-button-disabled-border-color, transparent);
  --disabled-background: var(--ig-button-disabled-background, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --disabled-foreground: var(--ig-button-disabled-foreground, hsla(var(--ig-gray-400), var(--ig-gray-a)));
  --size: var(--ig-button-size, max(var(--is-large, 1) * max(2.375rem, -1 * 2.375rem), var(--is-medium, 1) * max(2rem, -1 * 2rem), var(--is-small, 1) * max(1.5rem, -1 * 1.5rem)));
  --resting-elevation: var(--ig-button-resting-elevation, var(--ig-elevation-0));
  --hover-elevation: var(--ig-button-hover-elevation, var(--ig-elevation-0));
  --focus-elevation: var(--ig-button-focus-elevation, var(--ig-elevation-0));
  --active-elevation: var(--ig-button-active-elevation, var(--ig-elevation-0));
  --background: var(--ig-button-background, hsla(var(--ig-primary-500), var(--ig-primary-a)));
  --foreground: var(--ig-button-foreground, var(--ig-primary-900-contrast));
  --hover-background: var(--ig-button-hover-background, hsla(var(--ig-primary-600), var(--ig-primary-a)));
  --hover-foreground: var(--ig-button-hover-foreground, var(--ig-primary-900-contrast));
  --focus-background: var(--ig-button-focus-background, hsla(var(--ig-primary-600), var(--ig-primary-a)));
  --focus-foreground: var(--ig-button-focus-foreground, var(--ig-primary-900-contrast));
  --focus-visible-background: var(--ig-button-focus-visible-background, hsla(var(--ig-primary-600), var(--ig-primary-a)));
  --focus-visible-foreground: var(--ig-button-focus-visible-foreground, var(--ig-primary-900-contrast));
  --active-background: var(--ig-button-active-background, hsla(var(--ig-primary-800), var(--ig-primary-a)));
  --active-foreground: var(--ig-button-active-foreground, var(--ig-primary-900-contrast));
  --border-radius: var(--ig-button-border-radius, clamp(0.75rem, calc(var(--ig-radius-factor, 0.5) * 1.75rem), 1.75rem));
}
:host([variant=fab]) [part=base] {
  height: var(--size);
  border-radius: var(--border-radius);
}
:host([variant=fab]) [part=base]::after {
  border-radius: calc(var(--border-radius) - 0.25rem);
}

[part=base] {
  padding: 0 max(var(--is-large, 1) * max(16px, -1 * 16px) * var(--ig-spacing-large, var(--ig-spacing, --ig-spacing)), var(--is-medium, 1) * max(16px, -1 * 16px) * var(--ig-spacing-medium, var(--ig-spacing, --ig-spacing)), var(--is-small, 1) * max(8px, -1 * 8px) * var(--ig-spacing-small, var(--ig-spacing, --ig-spacing)));
  -webkit-transition: all 0.1s ease-in-out;
  transition: all 0.1s ease-in-out;
}

:host(:not([disabled])[variant=flat]) [part=base],
:host(:not(:disabled)[variant=flat]) [part=base] {
  color: var(--foreground);
  background: var(--background);
  border: 1px solid var(--border-color);
}
:host(:not([disabled])[variant=flat]) [part=base]:hover,
:host(:not(:disabled)[variant=flat]) [part=base]:hover {
  color: var(--hover-foreground);
  background: var(--hover-background);
  border-color: var(--hover-border-color);
}
:host(:not([disabled])[variant=flat]) [part=base]:focus,
:host(:not(:disabled)[variant=flat]) [part=base]:focus {
  color: var(--focus-foreground);
  background: var(--focus-background);
  border-color: var(--focus-border-color);
}
:host(:not([disabled])[variant=flat]) [part=base]:active,
:host(:not(:disabled)[variant=flat]) [part=base]:active {
  color: var(--active-foreground);
  background: var(--active-background);
  border-color: var(--active-border-color);
}
:host(:not([disabled])[variant=flat]) [part=base]:focus-visible,
:host(:not(:disabled)[variant=flat]) [part=base]:focus-visible {
  color: var(--focus-visible-foreground);
  background: var(--focus-visible-background);
  border-color: var(--focus-visible-border-color);
}

:host(:not([disabled])[variant=outlined]) [part=base],
:host(:not(:disabled)[variant=outlined]) [part=base] {
  color: var(--foreground);
  background: var(--background);
  border: 0.0625rem solid var(--border-color);
}
:host(:not([disabled])[variant=outlined]) [part=base]:hover,
:host(:not(:disabled)[variant=outlined]) [part=base]:hover {
  color: var(--hover-foreground);
  background: var(--hover-background);
  border-color: var(--hover-border-color);
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus,
:host(:not(:disabled)[variant=outlined]) [part=base]:focus {
  color: var(--focus-foreground);
  background: var(--focus-background);
  border-color: var(--focus-border-color);
}
:host(:not([disabled])[variant=outlined]) [part=base]:active,
:host(:not(:disabled)[variant=outlined]) [part=base]:active {
  color: var(--active-foreground);
  background: var(--active-background);
  border-color: var(--active-border-color);
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus-visible,
:host(:not(:disabled)[variant=outlined]) [part=base]:focus-visible {
  color: var(--focus-visible-foreground);
  background: var(--focus-visible-background);
  border-color: var(--focus-visible-border-color);
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus-visible::after,
:host(:not(:disabled)[variant=outlined]) [part=base]:focus-visible::after {
  content: "";
  position: absolute;
  inset-block-start: 0.125rem;
  inset-inline-start: 0.125rem;
  pointer-events: none;
  width: calc(100% - 0.125rem * 2);
  height: calc(100% - 0.125rem * 2);
  -webkit-box-shadow: 0 0 0 0.0625rem var(--focus-visible-foreground);
          box-shadow: 0 0 0 0.0625rem var(--focus-visible-foreground);
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus-visible:hover,
:host(:not(:disabled)[variant=outlined]) [part=base]:focus-visible:hover {
  color: var(--hover-foreground);
  background: var(--hover-background);
  border-color: var(--hover-border-color);
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus-visible:hover::after,
:host(:not(:disabled)[variant=outlined]) [part=base]:focus-visible:hover::after {
  -webkit-box-shadow: 0 0 0 0.0625rem var(--hover-foreground);
          box-shadow: 0 0 0 0.0625rem var(--hover-foreground);
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus-visible:active,
:host(:not(:disabled)[variant=outlined]) [part=base]:focus-visible:active {
  color: var(--active-foreground);
  background: var(--active-background);
  border-color: var(--active-border-color);
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus-visible:active::after,
:host(:not(:disabled)[variant=outlined]) [part=base]:focus-visible:active::after {
  -webkit-box-shadow: 0 0 0 0.0625rem var(--active-foreground);
          box-shadow: 0 0 0 0.0625rem var(--active-foreground);
}

:host(:not([disabled])[variant=contained]) [part=base],
:host(:not(:disabled)[variant=contained]) [part=base] {
  color: var(--foreground);
  background: var(--background);
  border: 0.0625rem solid var(--border-color);
}
:host(:not([disabled])[variant=contained]) [part=base]:hover,
:host(:not(:disabled)[variant=contained]) [part=base]:hover {
  color: var(--hover-foreground);
  background: var(--hover-background);
  border-color: var(--hover-border-color);
}
:host(:not([disabled])[variant=contained]) [part=base]:focus,
:host(:not(:disabled)[variant=contained]) [part=base]:focus {
  color: var(--focus-foreground);
  background: var(--focus-background);
  border-color: var(--focus-border-color);
}
:host(:not([disabled])[variant=contained]) [part=base]:active,
:host(:not(:disabled)[variant=contained]) [part=base]:active {
  color: var(--active-foreground);
  background: var(--active-background);
  border-color: var(--active-border-color);
}
:host(:not([disabled])[variant=contained]) [part=base]:focus-visible,
:host(:not(:disabled)[variant=contained]) [part=base]:focus-visible {
  color: var(--focus-visible-foreground);
  background: var(--focus-visible-background);
  border-color: var(--focus-visible-border-color);
  position: relative;
}
:host(:not([disabled])[variant=contained]) [part=base]:focus-visible::after,
:host(:not(:disabled)[variant=contained]) [part=base]:focus-visible::after {
  content: "";
  -webkit-box-shadow: 0 0 0 1px var(--ig-primary-600-contrast);
          box-shadow: 0 0 0 1px var(--ig-primary-600-contrast);
  position: absolute;
  inset-block-start: 3px;
  inset-inline-start: 3px;
  pointer-events: none;
  width: calc(100% - 6px);
  height: calc(100% - 6px);
}
:host(:not([disabled])[variant=contained]) [part=base]:focus-visible:hover,
:host(:not(:disabled)[variant=contained]) [part=base]:focus-visible:hover {
  color: var(--hover-foreground);
  background: var(--hover-background);
  border-color: var(--hover-border-color);
}
:host(:not([disabled])[variant=contained]) [part=base]:focus-visible:active,
:host(:not(:disabled)[variant=contained]) [part=base]:focus-visible:active {
  color: var(--active-foreground);
  background: var(--active-background);
  border-color: var(--active-border-color);
}

:host(:not([disabled])[variant=fab]) [part=base],
:host(:not(:disabled)[variant=fab]) [part=base] {
  color: var(--foreground);
  background: var(--background);
  border: 0.0625rem solid var(--border-color);
}
:host(:not([disabled])[variant=fab]) [part=base]:hover,
:host(:not(:disabled)[variant=fab]) [part=base]:hover {
  color: var(--hover-foreground);
  background: var(--hover-background);
  border-color: var(--hover-border-color);
}
:host(:not([disabled])[variant=fab]) [part=base]:focus,
:host(:not(:disabled)[variant=fab]) [part=base]:focus {
  color: var(--focus-foreground);
  background: var(--focus-background);
  border-color: var(--focus-border-color);
}
:host(:not([disabled])[variant=fab]) [part=base]:active,
:host(:not(:disabled)[variant=fab]) [part=base]:active {
  color: var(--active-foreground);
  background: var(--active-background);
  border-color: var(--active-border-color);
}
:host(:not([disabled])[variant=fab]) [part=base]:focus-visible,
:host(:not(:disabled)[variant=fab]) [part=base]:focus-visible {
  color: var(--focus-visible-foreground);
  background: var(--focus-visible-background);
  border-color: var(--focus-visible-border-color);
  position: relative;
}
:host(:not([disabled])[variant=fab]) [part=base]:focus-visible::after,
:host(:not(:disabled)[variant=fab]) [part=base]:focus-visible::after {
  content: "";
  -webkit-box-shadow: 0 0 0 1px var(--ig-primary-600-contrast);
          box-shadow: 0 0 0 1px var(--ig-primary-600-contrast);
  position: absolute;
  inset-block-start: 3px;
  inset-inline-start: 3px;
  pointer-events: none;
  width: calc(100% - 6px);
  height: calc(100% - 6px);
}
:host(:not([disabled])[variant=fab]) [part=base]:focus-visible:hover,
:host(:not(:disabled)[variant=fab]) [part=base]:focus-visible:hover {
  color: var(--hover-foreground);
  background: var(--hover-background);
  border-color: var(--hover-border-color);
}
:host(:not([disabled])[variant=fab]) [part=base]:focus-visible:active,
:host(:not(:disabled)[variant=fab]) [part=base]:focus-visible:active {
  color: var(--active-foreground);
  background: var(--active-background);
  border-color: var(--active-border-color);
}

:host([disabled][variant=flat]) [part=base],
:host(:disabled[variant=flat]) [part=base] {
  color: var(--disabled-foreground);
  background: var(--disabled-background);
}

:host([disabled][variant=outlined]) [part=base],
:host(:disabled[variant=outlined]) [part=base] {
  color: var(--disabled-foreground);
  background: var(--disabled-background);
  border-color: var(--disabled-border-color);
}

:host([disabled][variant=contained]) [part=base],
:host(:disabled[variant=contained]) [part=base] {
  color: var(--disabled-foreground);
  background: var(--disabled-background);
  border: 0.0625rem solid var(--disabled-border-color);
}

:host([disabled][variant=fab]) [part=base],
:host(:disabled[variant=fab]) [part=base] {
  color: var(--disabled-foreground);
  background: var(--disabled-background);
}`;
