// igniteui-webcomponents/components/button/themes/button/light/button.material.css.js
import { css } from 'lit';
export const styles = css `:host([variant=flat]) {
  --shadow-color: var(--ig-button-shadow-color, transparent);
  --border-color: var(--ig-button-border-color, transparent);
  --hover-border-color: var(--ig-button-hover-border-color, transparent);
  --focus-border-color: var(--ig-button-focus-border-color, transparent);
  --focus-visible-border-color: var(--ig-button-focus-visible-border-color, transparent);
  --active-border-color: var(--ig-button-active-border-color, transparent);
  --disabled-border-color: var(--ig-button-disabled-border-color, transparent);
  --disabled-background: var(--ig-button-disabled-background, transparent);
  --disabled-foreground: var(--ig-button-disabled-foreground, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --size: var(--ig-button-size, max(var(--is-large, 1) * max(2.25rem, -1 * 2.25rem), var(--is-medium, 1) * max(1.875rem, -1 * 1.875rem), var(--is-small, 1) * max(1.5rem, -1 * 1.5rem)));
  --resting-elevation: var(--ig-button-resting-elevation, var(--ig-elevation-0));
  --hover-elevation: var(--ig-button-hover-elevation, var(--ig-elevation-0));
  --focus-elevation: var(--ig-button-focus-elevation, var(--ig-elevation-0));
  --active-elevation: var(--ig-button-active-elevation, var(--ig-elevation-0));
  --background: var(--ig-button-background, transparent);
  --foreground: var(--ig-button-foreground, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --hover-background: var(--ig-button-hover-background, hsla(var(--ig-secondary-500), 0.05));
  --hover-foreground: var(--ig-button-hover-foreground, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --focus-background: var(--ig-button-focus-background, hsla(var(--ig-secondary-400), 0.12));
  --focus-foreground: var(--ig-button-focus-foreground, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --focus-visible-background: var(--ig-button-focus-visible-background, hsla(var(--ig-secondary-400), 0.12));
  --focus-visible-foreground: var(--ig-button-focus-visible-foreground, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --active-background: var(--ig-button-active-background, hsla(var(--ig-secondary-400), 0.12));
  --active-foreground: var(--ig-button-active-foreground, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --border-radius: var(--ig-button-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.2) * 1.25rem), 1.25rem));
}
:host([variant=flat]) [part=base] {
  height: var(--size);
}

:host([variant=outlined]) {
  --shadow-color: var(--ig-button-shadow-color, transparent);
  --border-color: var(--ig-button-border-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --hover-border-color: var(--ig-button-hover-border-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --focus-border-color: var(--ig-button-focus-border-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --focus-visible-border-color: var(--ig-button-focus-visible-border-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --active-border-color: var(--ig-button-active-border-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --disabled-border-color: var(--ig-button-disabled-border-color, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --disabled-background: var(--ig-button-disabled-background, transparent);
  --disabled-foreground: var(--ig-button-disabled-foreground, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --size: var(--ig-button-size, max(var(--is-large, 1) * max(2.25rem, -1 * 2.25rem), var(--is-medium, 1) * max(1.875rem, -1 * 1.875rem), var(--is-small, 1) * max(1.5rem, -1 * 1.5rem)));
  --resting-elevation: var(--ig-button-resting-elevation, var(--ig-elevation-0));
  --hover-elevation: var(--ig-button-hover-elevation, var(--ig-elevation-0));
  --focus-elevation: var(--ig-button-focus-elevation, var(--ig-elevation-0));
  --active-elevation: var(--ig-button-active-elevation, var(--ig-elevation-0));
  --background: var(--ig-button-background, transparent);
  --foreground: var(--ig-button-foreground, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --hover-background: var(--ig-button-hover-background, hsla(var(--ig-secondary-500), 0.05));
  --hover-foreground: var(--ig-button-hover-foreground, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --focus-background: var(--ig-button-focus-background, hsla(var(--ig-secondary-400), 0.12));
  --focus-foreground: var(--ig-button-focus-foreground, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --focus-visible-background: var(--ig-button-focus-visible-background, hsla(var(--ig-secondary-400), 0.12));
  --focus-visible-foreground: var(--ig-button-focus-visible-foreground, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --active-background: var(--ig-button-active-background, hsla(var(--ig-secondary-400), 0.12));
  --active-foreground: var(--ig-button-active-foreground, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --border-radius: var(--ig-button-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.2) * 1.25rem), 1.25rem));
}
:host([variant=outlined]) [part=base] {
  height: var(--size);
}

:host([variant=contained]) {
  --shadow-color: var(--ig-button-shadow-color, transparent);
  --border-color: var(--ig-button-border-color, transparent);
  --hover-border-color: var(--ig-button-hover-border-color, transparent);
  --focus-border-color: var(--ig-button-focus-border-color, transparent);
  --focus-visible-border-color: var(--ig-button-focus-visible-border-color, transparent);
  --active-border-color: var(--ig-button-active-border-color, transparent);
  --disabled-border-color: var(--ig-button-disabled-border-color, transparent);
  --disabled-background: var(--ig-button-disabled-background, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --disabled-foreground: var(--ig-button-disabled-foreground, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --size: var(--ig-button-size, max(var(--is-large, 1) * max(2.25rem, -1 * 2.25rem), var(--is-medium, 1) * max(1.875rem, -1 * 1.875rem), var(--is-small, 1) * max(1.5rem, -1 * 1.5rem)));
  --resting-elevation: var(--ig-button-resting-elevation, var(--ig-elevation-2));
  --hover-elevation: var(--ig-button-hover-elevation, var(--ig-elevation-4));
  --focus-elevation: var(--ig-button-focus-elevation, var(--ig-elevation-8));
  --active-elevation: var(--ig-button-active-elevation, var(--ig-elevation-8));
  --background: var(--ig-button-background, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --foreground: var(--ig-button-foreground, var(--ig-secondary-500-contrast));
  --hover-background: var(--ig-button-hover-background, hsla(var(--ig-secondary-300), var(--ig-secondary-a)));
  --hover-foreground: var(--ig-button-hover-foreground, var(--ig-secondary-300-contrast));
  --focus-background: var(--ig-button-focus-background, hsla(var(--ig-secondary-300), var(--ig-secondary-a)));
  --focus-foreground: var(--ig-button-focus-foreground, var(--ig-secondary-300-contrast));
  --focus-visible-background: var(--ig-button-focus-visible-background, hsla(var(--ig-secondary-300), var(--ig-secondary-a)));
  --focus-visible-foreground: var(--ig-button-focus-visible-foreground, var(--ig-secondary-300-contrast));
  --active-background: var(--ig-button-active-background, hsla(var(--ig-secondary-300), var(--ig-secondary-a)));
  --active-foreground: var(--ig-button-active-foreground, var(--ig-secondary-300-contrast));
  --border-radius: var(--ig-button-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0.2) * 1.25rem), 1.25rem));
}
:host([variant=contained]) [part=base] {
  height: var(--size);
}

:host([variant=fab]) {
  --shadow-color: var(--ig-button-shadow-color, transparent);
  --border-color: var(--ig-button-border-color, transparent);
  --hover-border-color: var(--ig-button-hover-border-color, transparent);
  --focus-border-color: var(--ig-button-focus-border-color, transparent);
  --focus-visible-border-color: var(--ig-button-focus-visible-border-color, transparent);
  --active-border-color: var(--ig-button-active-border-color, transparent);
  --disabled-border-color: var(--ig-button-disabled-border-color, transparent);
  --disabled-background: var(--ig-button-disabled-background, hsla(var(--ig-gray-300), var(--ig-gray-a)));
  --disabled-foreground: var(--ig-button-disabled-foreground, hsla(var(--ig-gray-500), var(--ig-gray-a)));
  --size: var(--ig-button-size, max(var(--is-large, 1) * max(3rem, -1 * 3rem), var(--is-medium, 1) * max(2.5rem, -1 * 2.5rem), var(--is-small, 1) * max(2rem, -1 * 2rem)));
  --resting-elevation: var(--ig-button-resting-elevation, var(--ig-elevation-6));
  --hover-elevation: var(--ig-button-hover-elevation, var(--ig-elevation-12));
  --focus-elevation: var(--ig-button-focus-elevation, var(--ig-elevation-12));
  --active-elevation: var(--ig-button-active-elevation, var(--ig-elevation-12));
  --background: var(--ig-button-background, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --foreground: var(--ig-button-foreground, var(--ig-secondary-500-contrast));
  --hover-background: var(--ig-button-hover-background, hsla(var(--ig-secondary-300), var(--ig-secondary-a)));
  --hover-foreground: var(--ig-button-hover-foreground, var(--ig-secondary-300-contrast));
  --focus-background: var(--ig-button-focus-background, hsla(var(--ig-secondary-300), var(--ig-secondary-a)));
  --focus-foreground: var(--ig-button-focus-foreground, var(--ig-secondary-300-contrast));
  --focus-visible-background: var(--ig-button-focus-visible-background, hsla(var(--ig-secondary-300), var(--ig-secondary-a)));
  --focus-visible-foreground: var(--ig-button-focus-visible-foreground, var(--ig-secondary-300-contrast));
  --active-background: var(--ig-button-active-background, hsla(var(--ig-secondary-300), var(--ig-secondary-a)));
  --active-foreground: var(--ig-button-active-foreground, var(--ig-secondary-300-contrast));
  --border-radius: var(--ig-button-border-radius, clamp(0.75rem, calc(var(--ig-radius-factor, 1) * 1.75rem), 1.75rem));
}
:host([variant=fab]) [part=base] {
  height: var(--size);
}

[part=base] {
  -webkit-transition: all 0.1s ease-in-out;
  transition: all 0.1s ease-in-out;
}

:host(:not([disabled])[variant=flat]) [part=base],
:host(:not(:disabled)[variant=flat]) [part=base] {
  color: var(--foreground);
  background: var(--background);
}
:host(:not([disabled])[variant=flat]) [part=base]:hover,
:host(:not(:disabled)[variant=flat]) [part=base]:hover {
  color: var(--hover-foreground);
  background: var(--hover-background);
}
:host(:not([disabled])[variant=flat]) [part=base]:focus,
:host(:not(:disabled)[variant=flat]) [part=base]:focus {
  color: var(--focus-foreground);
  background: var(--focus-background);
}
:host(:not([disabled])[variant=flat]) [part=base]:active,
:host(:not(:disabled)[variant=flat]) [part=base]:active {
  color: var(--active-foreground);
  background: var(--active-background);
}
:host(:not([disabled])[variant=flat]) [part=base]:focus-visible,
:host(:not(:disabled)[variant=flat]) [part=base]:focus-visible {
  color: var(--focus-visible-foreground);
  background: var(--focus-visible-background);
}
:host(:not([disabled])[variant=flat]) [part=base]:focus-visible:hover,
:host(:not(:disabled)[variant=flat]) [part=base]:focus-visible:hover {
  color: var(--hover-foreground);
  background: var(--hover-foreground);
}
:host(:not([disabled])[variant=flat]) [part=base]:focus-visible:active,
:host(:not(:disabled)[variant=flat]) [part=base]:focus-visible:active {
  color: var(--active-foreground);
  background: var(--active-background);
}

:host(:not([disabled])[variant=outlined]) [part=base],
:host(:not(:disabled)[variant=outlined]) [part=base] {
  color: var(--foreground);
  background: var(--background);
  border: 0.0625rem solid var(--border-color);
}
:host(:not([disabled])[variant=outlined]) [part=base]:hover,
:host(:not(:disabled)[variant=outlined]) [part=base]:hover {
  color: var(--hover-foreground);
  background: var(--hover-background);
  border-color: var(--hover-border-color);
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus,
:host(:not(:disabled)[variant=outlined]) [part=base]:focus {
  color: var(--focus-foreground);
  background: var(--focus-background);
  border-color: var(--focus-border-color);
}
:host(:not([disabled])[variant=outlined]) [part=base]:active,
:host(:not(:disabled)[variant=outlined]) [part=base]:active {
  color: var(--active-foreground);
  background: var(--active-background);
  border-color: var(--active-border-color);
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus-visible,
:host(:not(:disabled)[variant=outlined]) [part=base]:focus-visible {
  color: var(--focus-visible-foreground);
  background: var(--focus-visible-background);
  border-color: var(--focus-visible-border-color);
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus-visible:hover,
:host(:not(:disabled)[variant=outlined]) [part=base]:focus-visible:hover {
  color: var(--hover-foreground);
  background: var(--hover-foreground);
  border-color: var(--hover-border-color);
}
:host(:not([disabled])[variant=outlined]) [part=base]:focus-visible:active,
:host(:not(:disabled)[variant=outlined]) [part=base]:focus-visible:active {
  color: var(--active-foreground);
  background: var(--active-background);
  border-color: var(--active-border-color);
}

:host(:not([disabled])[variant=contained]) [part=base],
:host(:not(:disabled)[variant=contained]) [part=base] {
  color: var(--foreground);
  background: var(--background);
  -webkit-box-shadow: var(--ig-elevation-2);
          box-shadow: var(--ig-elevation-2);
}
:host(:not([disabled])[variant=contained]) [part=base]:hover,
:host(:not(:disabled)[variant=contained]) [part=base]:hover {
  color: var(--hover-foreground);
  background: var(--hover-background);
  -webkit-box-shadow: var(--ig-elevation-4);
          box-shadow: var(--ig-elevation-4);
}
:host(:not([disabled])[variant=contained]) [part=base]:focus,
:host(:not(:disabled)[variant=contained]) [part=base]:focus {
  color: var(--focus-foreground);
  background: var(--focus-background);
}
:host(:not([disabled])[variant=contained]) [part=base]:active,
:host(:not(:disabled)[variant=contained]) [part=base]:active {
  color: var(--active-foreground);
  background: var(--active-background);
}
:host(:not([disabled])[variant=contained]) [part=base]:focus, :host(:not([disabled])[variant=contained]) [part=base]:active,
:host(:not(:disabled)[variant=contained]) [part=base]:focus,
:host(:not(:disabled)[variant=contained]) [part=base]:active {
  -webkit-box-shadow: var(--ig-elevation-8);
          box-shadow: var(--ig-elevation-8);
}
:host(:not([disabled])[variant=contained]) [part=base]:focus-visible,
:host(:not(:disabled)[variant=contained]) [part=base]:focus-visible {
  color: var(--focus-foreground);
  background: var(--focus-background);
}
:host(:not([disabled])[variant=contained]) [part=base]:focus-visible:hover,
:host(:not(:disabled)[variant=contained]) [part=base]:focus-visible:hover {
  color: var(--hover-foreground);
  background: var(--hover-background);
}
:host(:not([disabled])[variant=contained]) [part=base]:focus-visible:active,
:host(:not(:disabled)[variant=contained]) [part=base]:focus-visible:active {
  color: var(--active-foreground);
  background: var(--active-background);
}

:host(:not([disabled])[variant=fab]) [part=base],
:host(:not(:disabled)[variant=fab]) [part=base] {
  color: var(--foreground);
  background: var(--background);
  -webkit-box-shadow: var(--ig-elevation-6);
          box-shadow: var(--ig-elevation-6);
}
:host(:not([disabled])[variant=fab]) [part=base]:hover,
:host(:not(:disabled)[variant=fab]) [part=base]:hover {
  color: var(--hover-foreground);
  background: var(--hover-background);
  -webkit-box-shadow: var(--ig-elevation-4);
          box-shadow: var(--ig-elevation-4);
}
:host(:not([disabled])[variant=fab]) [part=base]:focus,
:host(:not(:disabled)[variant=fab]) [part=base]:focus {
  color: var(--focus-foreground);
  background: var(--focus-background);
}
:host(:not([disabled])[variant=fab]) [part=base]:active,
:host(:not(:disabled)[variant=fab]) [part=base]:active {
  color: var(--active-foreground);
  background: var(--active-background);
}
:host(:not([disabled])[variant=fab]) [part=base]:focus, :host(:not([disabled])[variant=fab]) [part=base]:active,
:host(:not(:disabled)[variant=fab]) [part=base]:focus,
:host(:not(:disabled)[variant=fab]) [part=base]:active {
  -webkit-box-shadow: var(--ig-elevation-8);
          box-shadow: var(--ig-elevation-8);
}
:host(:not([disabled])[variant=fab]) [part=base]:focus-visible,
:host(:not(:disabled)[variant=fab]) [part=base]:focus-visible {
  color: var(--focus-foreground);
  background: var(--focus-background);
}
:host(:not([disabled])[variant=fab]) [part=base]:focus-visible:hover,
:host(:not(:disabled)[variant=fab]) [part=base]:focus-visible:hover {
  color: var(--hover-foreground);
  background: var(--hover-background);
}
:host(:not([disabled])[variant=fab]) [part=base]:focus-visible:active,
:host(:not(:disabled)[variant=fab]) [part=base]:focus-visible:active {
  color: var(--active-foreground);
  background: var(--active-background);
}
:host(:not([disabled])[variant=fab]) [part=base]:hover, :host(:not([disabled])[variant=fab]) [part=base]:focus, :host(:not([disabled])[variant=fab]) [part=base]:focus-visible, :host(:not([disabled])[variant=fab]) [part=base]:active,
:host(:not(:disabled)[variant=fab]) [part=base]:hover,
:host(:not(:disabled)[variant=fab]) [part=base]:focus,
:host(:not(:disabled)[variant=fab]) [part=base]:focus-visible,
:host(:not(:disabled)[variant=fab]) [part=base]:active {
  -webkit-box-shadow: var(--ig-elevation-12);
          box-shadow: var(--ig-elevation-12);
}
:host([variant=flat]) [part=base] {
  border-radius: var(--border-radius);
}

:host([variant=outlined]) [part=base] {
  border-radius: var(--border-radius);
}

:host([variant=contained]) [part=base] {
  border-radius: var(--border-radius);
}

:host([variant=fab]) [part=base] {
  border-radius: var(--border-radius);
}
:host([disabled][variant=flat]) [part=base],
:host(:disabled[variant=flat]) [part=base] {
  color: var(--disabled-foreground);
  background: var(--disabled-background);
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--disabled-border-color);
          box-shadow: inset 0 0 0 0.0625rem var(--disabled-border-color);
}

:host([disabled][variant=outlined]) [part=base],
:host(:disabled[variant=outlined]) [part=base] {
  color: var(--disabled-foreground);
  background: var(--disabled-background);
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--disabled-border-color);
          box-shadow: inset 0 0 0 0.0625rem var(--disabled-border-color);
  border: 0.0625rem solid var(--disabled-border-color);
}

:host([disabled][variant=contained]) [part=base],
:host(:disabled[variant=contained]) [part=base] {
  color: var(--disabled-foreground);
  background: var(--disabled-background);
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--disabled-border-color);
          box-shadow: inset 0 0 0 0.0625rem var(--disabled-border-color);
}

:host([disabled][variant=fab]) [part=base],
:host(:disabled[variant=fab]) [part=base] {
  color: var(--disabled-foreground);
  background: var(--disabled-background);
  -webkit-box-shadow: inset 0 0 0 0.0625rem var(--disabled-border-color);
          box-shadow: inset 0 0 0 0.0625rem var(--disabled-border-color);
}`;
