// igniteui-webcomponents/components/list/themes/light/list-header.base.css.js
import { css } from 'lit';
export const styles = css `:host {
  --background: var(--ig-list-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --header-background: var(--ig-list-header-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --header-text-color: var(--ig-list-header-text-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --item-background: var(--ig-list-item-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --item-background-hover: var(--ig-list-item-background-hover, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --item-background-active: var(--ig-list-item-background-active, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --item-text-color: var(--ig-list-item-text-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --item-text-color-hover: var(--ig-list-item-text-color-hover, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --item-text-color-active: var(--ig-list-item-text-color-active, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --item-title-color: var(--ig-list-item-title-color, 'currentColor');
  --item-title-color-hover: var(--ig-list-item-title-color-hover, 'currentColor');
  --item-title-color-active: var(--ig-list-item-title-color-active, 'currentColor');
  --item-action-color: var(--ig-list-item-action-color, 'currentColor');
  --item-action-color-hover: var(--ig-list-item-action-color-hover, 'currentColor');
  --item-action-color-active: var(--ig-list-item-action-color-active, 'currentColor');
  --item-subtitle-color: var(--ig-list-item-subtitle-color, 'currentColor');
  --item-subtitle-color-hover: var(--ig-list-item-subtitle-color-hover, 'currentColor');
  --item-subtitle-color-active: var(--ig-list-item-subtitle-color-active, 'currentColor');
  --item-thumbnail-color: var(--ig-list-item-thumbnail-color, 'currentColor');
  --item-thumbnail-color-hover: var(--ig-list-item-thumbnail-color-hover, 'currentColor');
  --item-thumbnail-color-active: var(--ig-list-item-thumbnail-color-active, 'currentColor');
  --border-color: var(--ig-list-border-color, transparent);
  --border-width: var(--ig-list-border-width, 0);
  --border-radius: var(--ig-list-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0) * 1.5rem), 1.5rem));
  --item-border-radius: var(--ig-list-item-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0) * 1.5rem), 1.5rem));
  font-family: var(--ig-overline-font-family, var(--ig-font-family));
  font-size: var(--ig-overline-font-size);
  font-weight: var(--ig-overline-font-weight);
  font-style: var(--ig-overline-font-style);
  line-height: var(--ig-overline-line-height);
  letter-spacing: var(--ig-overline-letter-spacing);
  text-transform: var(--ig-overline-text-transform);
  margin-top: var(--ig-overline-margin-top);
  margin-bottom: var(--ig-overline-margin-bottom);
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: start;
      -ms-flex-pack: start;
          justify-content: flex-start;
  color: var(--header-text-color);
  background: var(--header-background);
  padding-inline: max(var(--is-large, 1) * max(16px, -1 * 16px) * var(--ig-spacing-inline-large, var(--ig-spacing-inline, --ig-spacing)), var(--is-medium, 1) * max(8px, -1 * 8px) * var(--ig-spacing-inline-medium, var(--ig-spacing-inline, --ig-spacing)), var(--is-small, 1) * max(4px, -1 * 4px) * var(--ig-spacing-inline-small, var(--ig-spacing-inline, --ig-spacing)));
  padding-block: max(var(--is-large, 1) * max(8px, -1 * 8px) * var(--ig-spacing-block-large, var(--ig-spacing-block, --ig-spacing)), var(--is-medium, 1) * max(4px, -1 * 4px) * var(--ig-spacing-block-medium, var(--ig-spacing-block, --ig-spacing)), var(--is-small, 1) * max(2px, -1 * 2px) * var(--ig-spacing-block-small, var(--ig-spacing-block, --ig-spacing)));
}

::slotted(*) {
  font: inherit;
  margin: inherit;
}`;
