// igniteui-webcomponents/components/list/themes/light/list.base.css.js
import { css } from 'lit';
export const styles = css `:host {
  --is-large: clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-large, 3), 1);
  --is-medium: min(
      clamp(0, (var(--component-size, 1) + 1) - var(--ig-size-medium, 2), 1),
      clamp(0, var(--ig-size-large, 3) - var(--component-size, 1), 1)
  );
  --is-small: clamp(0, var(--ig-size-medium) - var(--component-size, 1), 1);
  position: relative;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  scrollbar-width: var(--ig-scrollbar-size);
  scrollbar-color: var(--ig-scrollbar-thumb-background) var(--ig-scrollbar-track-background);
}
:host *,
:host *::before,
:host *::after {
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
}
:host ::-webkit-scrollbar {
  width: var(--ig-scrollbar-size);
  height: var(--ig-scrollbar-size);
  background: var(--ig-scrollbar-track-background);
}
:host ::-webkit-scrollbar-thumb {
  background: var(--ig-scrollbar-thumb-background);
}

:host([hidden]),
[hidden] {
  display: none !important;
}
:host {
  --background: var(--ig-list-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --header-background: var(--ig-list-header-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --header-text-color: var(--ig-list-header-text-color, hsla(var(--ig-secondary-500), var(--ig-secondary-a)));
  --item-background: var(--ig-list-item-background, hsla(var(--ig-surface-500), var(--ig-surface-a)));
  --item-background-hover: var(--ig-list-item-background-hover, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --item-background-active: var(--ig-list-item-background-active, hsla(var(--ig-gray-200), var(--ig-gray-a)));
  --item-text-color: var(--ig-list-item-text-color, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --item-text-color-hover: var(--ig-list-item-text-color-hover, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --item-text-color-active: var(--ig-list-item-text-color-active, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --item-title-color: var(--ig-list-item-title-color, 'currentColor');
  --item-title-color-hover: var(--ig-list-item-title-color-hover, 'currentColor');
  --item-title-color-active: var(--ig-list-item-title-color-active, 'currentColor');
  --item-action-color: var(--ig-list-item-action-color, 'currentColor');
  --item-action-color-hover: var(--ig-list-item-action-color-hover, 'currentColor');
  --item-action-color-active: var(--ig-list-item-action-color-active, 'currentColor');
  --item-subtitle-color: var(--ig-list-item-subtitle-color, 'currentColor');
  --item-subtitle-color-hover: var(--ig-list-item-subtitle-color-hover, 'currentColor');
  --item-subtitle-color-active: var(--ig-list-item-subtitle-color-active, 'currentColor');
  --item-thumbnail-color: var(--ig-list-item-thumbnail-color, 'currentColor');
  --item-thumbnail-color-hover: var(--ig-list-item-thumbnail-color-hover, 'currentColor');
  --item-thumbnail-color-active: var(--ig-list-item-thumbnail-color-active, 'currentColor');
  --border-color: var(--ig-list-border-color, transparent);
  --border-width: var(--ig-list-border-width, 0);
  --border-radius: var(--ig-list-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0) * 1.5rem), 1.5rem));
  --item-border-radius: var(--ig-list-item-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0) * 1.5rem), 1.5rem));
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-flow: column nowrap;
          flex-flow: column nowrap;
  font-family: var(--ig-font-family);
  overflow: hidden auto;
  height: auto;
  background: var(--background);
  border-radius: var(--border-radius);
}`;
