// igniteui-webcomponents/components/nav-drawer/themes/light/item/nav-drawer-item.indigo.css.js
import { css } from 'lit';
export const styles = css `:host {
  --elevation: var(--ig-nav-drawer-elevation, var(--ig-elevation-0));
  --background: var(--ig-nav-drawer-background, hsla(var(--ig-gray-700), var(--ig-gray-a)));
  --border-color: var(--ig-nav-drawer-border-color, transparent);
  --item-header-text-color: var(--ig-nav-drawer-item-header-text-color, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --item-text-color: var(--ig-nav-drawer-item-text-color, hsla(var(--ig-gray-300), 0.8));
  --item-icon-color: var(--ig-nav-drawer-item-icon-color, hsla(var(--ig-gray-300), 0.8));
  --item-active-text-color: var(--ig-nav-drawer-item-active-text-color, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --item-active-background: var(--ig-nav-drawer-item-active-background, hsla(var(--ig-gray-600), 0.8));
  --item-hover-background: var(--ig-nav-drawer-item-hover-background, transparent);
  --item-hover-text-color: var(--ig-nav-drawer-item-hover-text-color, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --item-active-icon-color: var(--ig-nav-drawer-item-active-icon-color, hsla(var(--ig-gray-50), var(--ig-gray-a)));
  --item-hover-icon-color: var(--ig-nav-drawer-item-hover-icon-color, hsla(var(--ig-gray-100), var(--ig-gray-a)));
  --border-radius: var(--ig-nav-drawer-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 0) * 2.25rem), 2.25rem));
  --item-border-radius: var(--ig-nav-drawer-item-border-radius, clamp(0rem, calc(var(--ig-radius-factor, 1) * 1.5rem), 1.5rem));
  --background-color: var(--ig-nav-drawer-background-color, hsla(var(--ig-gray-500), 0.54));
}

[part~=base] {
  gap: 1.5rem;
  padding: 0.25rem;
  margin: 0.9375rem 0.5rem;
  border-radius: var(--item-border-radius);
  color: var(--item-text-color);
}
[part~=base]:hover, [part~=base]:focus-within {
  background: var(--item-hover-background);
  color: var(--item-hover-text-color);
}
[part~=base]:hover ::slotted(igc-icon), [part~=base]:focus-within ::slotted(igc-icon) {
  color: var(--item-hover-icon-color);
}
[part~=base] ::slotted(igc-icon) {
  color: var(--item-icon-color);
}

[part=content] {
  font-family: var(--ig-subtitle-2-font-family, var(--ig-font-family));
  font-size: var(--ig-subtitle-2-font-size);
  font-weight: var(--ig-subtitle-2-font-weight);
  font-style: var(--ig-subtitle-2-font-style);
  line-height: var(--ig-subtitle-2-line-height);
  letter-spacing: var(--ig-subtitle-2-letter-spacing);
  text-transform: var(--ig-subtitle-2-text-transform);
  margin-top: var(--ig-subtitle-2-margin-top);
  margin-bottom: var(--ig-subtitle-2-margin-bottom);
  -webkit-padding-start: 0.5rem;
          padding-inline-start: 0.5rem;
}

:host([active]) [part~=base] {
  background: var(--item-active-background);
  color: var(--item-active-text-color);
}
:host([active]) [part~=base] ::slotted(igc-icon) {
  color: var(--item-active-icon-color);
}`;
