// igniteui-webcomponents/components/nav-drawer/themes/light/item/nav-drawer-item.base.css.js
import { css } from 'lit';
export const styles = css `[part~=base] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  gap: 2rem;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-flow: row nowrap;
          flex-flow: row nowrap;
  position: relative;
  max-height: 3rem;
  margin: 0.5rem;
  padding: 0.75rem 0.5rem;
  cursor: pointer;
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  outline: transparent;
  white-space: nowrap;
  text-decoration: none;
  border: none;
  -webkit-box-pack: start;
      -ms-flex-pack: start;
          justify-content: flex-start;
  -webkit-transition: background 0.15s ease-out;
  transition: background 0.15s ease-out;
}
[part~=base] ::slotted(igc-icon) {
  --ig-size: 3;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  font: inherit;
}

[part=content] {
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

[part="base mini"] {
  display: block;
}
[part="base mini"] [part=content] {
  display: none;
}
[part="base mini"] ::slotted(igc-icon) {
  width: 100%;
}

:host([disabled]) {
  -webkit-user-select: none;
     -moz-user-select: none;
      -ms-user-select: none;
          user-select: none;
  cursor: initial;
  pointer-events: none;
}`;
