// igniteui-webcomponents/components/button-group/toggle-button.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { LitElement, html } from 'lit';
import { property, query } from 'lit/decorators.js';
import { themes } from '../../theming/theming-decorator.js';
import { styles } from '../button-group/themes/light/toggle-button/toggle-button.base.css.js';
import { styles as bootstrap } from '../button-group/themes/light/toggle-button/toggle-button.bootstrap.css.js';
import { styles as fluent } from '../button-group/themes/light/toggle-button/toggle-button.fluent.css.js';
import { styles as indigo } from '../button-group/themes/light/toggle-button/toggle-button.indigo.css.js';
import { styles as material } from '../button-group/themes/light/toggle-button/toggle-button.material.css.js';
import { alternateName } from '../common/decorators/alternateName.js';
import { registerComponent } from '../common/definitions/register.js';
let IgcToggleButtonComponent = class IgcToggleButtonComponent extends LitElement {
    constructor() {
        super(...arguments);
        this.selected = false;
        this.disabled = false;
    }
    static register() {
        registerComponent(this);
    }
    focus(options) {
        this.nativeElement.focus(options);
    }
    blur() {
        this.nativeElement.blur();
    }
    click() {
        this.nativeElement.click();
    }
    render() {
        return html `
      <button
        part="toggle"
        type="button"
        .disabled=${this.disabled}
        .ariaLabel=${this.ariaLabel}
        aria-pressed=${this.selected}
        aria-disabled=${this.disabled}
      >
        <slot></slot>
      </button>
    `;
    }
};
IgcToggleButtonComponent.styles = styles;
IgcToggleButtonComponent.tagName = 'igc-toggle-button';
IgcToggleButtonComponent.shadowRootOptions = {
    ...LitElement.shadowRootOptions,
    delegatesFocus: true,
};
__decorate([
    query('[part="toggle"]', true)
], IgcToggleButtonComponent.prototype, "nativeElement", void 0);
__decorate([
    property()
], IgcToggleButtonComponent.prototype, "value", void 0);
__decorate([
    property({ type: Boolean, reflect: true })
], IgcToggleButtonComponent.prototype, "selected", void 0);
__decorate([
    property({ type: Boolean, reflect: true })
], IgcToggleButtonComponent.prototype, "disabled", void 0);
__decorate([
    alternateName('focusComponent')
], IgcToggleButtonComponent.prototype, "focus", null);
__decorate([
    alternateName('blurComponent')
], IgcToggleButtonComponent.prototype, "blur", null);
IgcToggleButtonComponent = __decorate([
    themes({
        light: { bootstrap, material, fluent, indigo },
        dark: { bootstrap, material, fluent, indigo },
    })
], IgcToggleButtonComponent);
export default IgcToggleButtonComponent;
